{
  "nbformat": 4,
  "nbformat_minor": 0,
  "metadata": {
    "colab": {
      "name": "Шейченко",
      "provenance": [],
      "collapsed_sections": []
    },
    "kernelspec": {
      "name": "python3",
      "display_name": "Python 3"
    },
    "language_info": {
      "name": "python"
    }
  },
  "cells": [
    {
      "cell_type": "markdown",
      "source": [
        "# Ссылка на диск с картированными чтениями"
      ],
      "metadata": {
        "id": "ZpPTWEJzkmn5"
      }
    },
    {
      "cell_type": "markdown",
      "source": [
        "\n",
        "| | SRA accession  | Stage    |\n",
        "|---:|:-------------|:-----------|\n",
        "| 1 | SRR5836473  | 8 cell       |\n",
        "| 2 | SRR3824222 | Epiblast    |\n",
        "| 3 | SRR5836475  | ICM       |\n"
      ],
      "metadata": {
        "id": "2pRLQP4Kkt6T"
      }
    },
    {
      "cell_type": "markdown",
      "source": [
        "[Bam-файлы и отчеты](https://drive.google.com/drive/folders/1athn52a93obOwfuHynDCRPRhjAPB80aO?usp=sharing)"
      ],
      "metadata": {
        "id": "KPnL4ZtQkxWr"
      }
    },
    {
      "cell_type": "markdown",
      "source": [
        "# Установка FastQC"
      ],
      "metadata": {
        "id": "raGwEIx7k3KG"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "!wget https://www.bioinformatics.babraham.ac.uk/projects/fastqc/fastqc_v0.11.9.zip\n",
        "!unzip fastqc_v0.11.9.zip\n",
        "!chmod a+x FastQC/fastqc"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "1cPn-1YVk4PC",
        "outputId": "feef63f6-2713-495a-adcc-23d8229b1ef9"
      },
      "execution_count": 6,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "--2022-02-16 07:17:31--  https://www.bioinformatics.babraham.ac.uk/projects/fastqc/fastqc_v0.11.9.zip\n",
            "Resolving www.bioinformatics.babraham.ac.uk (www.bioinformatics.babraham.ac.uk)... 149.155.133.4\n",
            "Connecting to www.bioinformatics.babraham.ac.uk (www.bioinformatics.babraham.ac.uk)|149.155.133.4|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 10249221 (9.8M) [application/zip]\n",
            "Saving to: ‘fastqc_v0.11.9.zip’\n",
            "\n",
            "fastqc_v0.11.9.zip  100%[===================>]   9.77M  2.88MB/s    in 3.4s    \n",
            "\n",
            "2022-02-16 07:17:35 (2.88 MB/s) - ‘fastqc_v0.11.9.zip’ saved [10249221/10249221]\n",
            "\n",
            "Archive:  fastqc_v0.11.9.zip\n",
            "  inflating: FastQC/cisd-jhdf5.jar   \n",
            "   creating: FastQC/Configuration/\n",
            "  inflating: FastQC/Configuration/adapter_list.txt  \n",
            "  inflating: FastQC/Configuration/contaminant_list.txt  \n",
            "  inflating: FastQC/Configuration/limits.txt  \n",
            "  inflating: FastQC/fastqc           \n",
            "  inflating: FastQC/fastqc_icon.ico  \n",
            "   creating: FastQC/Help/\n",
            "   creating: FastQC/Help/1 Introduction/\n",
            "   creating: FastQC/Help/1 Introduction/.svn/\n",
            "  inflating: FastQC/Help/1 Introduction/.svn/entries  \n",
            "   creating: FastQC/Help/1 Introduction/.svn/props/\n",
            "   creating: FastQC/Help/1 Introduction/.svn/text-base/\n",
            "  inflating: FastQC/Help/1 Introduction/.svn/text-base/1.1 What is FastQC.html.svn-base  \n",
            "   creating: FastQC/Help/1 Introduction/.svn/tmp/\n",
            "   creating: FastQC/Help/1 Introduction/.svn/tmp/props/\n",
            "  inflating: FastQC/Help/1 Introduction/1.1 What is FastQC.html  \n",
            "   creating: FastQC/Help/2 Basic Operations/\n",
            "   creating: FastQC/Help/2 Basic Operations/.svn/\n",
            "  inflating: FastQC/Help/2 Basic Operations/.svn/entries  \n",
            "   creating: FastQC/Help/2 Basic Operations/.svn/props/\n",
            "   creating: FastQC/Help/2 Basic Operations/.svn/text-base/\n",
            "  inflating: FastQC/Help/2 Basic Operations/.svn/text-base/2.1 Opening a sequence file.html.svn-base  \n",
            "  inflating: FastQC/Help/2 Basic Operations/.svn/text-base/2.2 Evaluating Results.html.svn-base  \n",
            "  inflating: FastQC/Help/2 Basic Operations/.svn/text-base/2.3 Saving a Report.html.svn-base  \n",
            "   creating: FastQC/Help/2 Basic Operations/.svn/tmp/\n",
            "   creating: FastQC/Help/2 Basic Operations/.svn/tmp/props/\n",
            "  inflating: FastQC/Help/2 Basic Operations/2.1 Opening a sequence file.html  \n",
            "  inflating: FastQC/Help/2 Basic Operations/2.2 Evaluating Results.html  \n",
            "  inflating: FastQC/Help/2 Basic Operations/2.3 Saving a Report.html  \n",
            "   creating: FastQC/Help/3 Analysis Modules/\n",
            "   creating: FastQC/Help/3 Analysis Modules/.svn/\n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/entries  \n",
            "   creating: FastQC/Help/3 Analysis Modules/.svn/prop-base/\n",
            " extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/duplication_levels.png.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/prop-base/kmer_profiles.png.svn-base  \n",
            " extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/per_base_gc_content.png.svn-base  \n",
            " extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/per_base_n_content.png.svn-base  \n",
            " extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/per_base_quality.png.svn-base  \n",
            " extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/per_base_sequence_content.png.svn-base  \n",
            " extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/per_sequence_gc_content.png.svn-base  \n",
            " extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/per_sequence_quality.png.svn-base  \n",
            " extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/per_tile_quality.png.svn-base  \n",
            " extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/sequence_length_distribution.png.svn-base  \n",
            "   creating: FastQC/Help/3 Analysis Modules/.svn/props/\n",
            "   creating: FastQC/Help/3 Analysis Modules/.svn/text-base/\n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/1 Basic Statistics.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/10 Adapter Content.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/11 Kmer Content.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/12 Per Tile Sequence Quality.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/2 Per Base Sequence Quality.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/3 Per Sequence Quality Scores.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/4 Per Base Sequence Content.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/5 Per Sequence GC Content.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/6 Per Base N Content.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/7 Sequence Length Distribution.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/8 Duplicate Sequences.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/9 Overrepresented Sequences.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/duplication_levels.png.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/kmer_profiles.png.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/per_base_gc_content.png.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/per_base_n_content.png.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/per_base_quality.png.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/per_base_sequence_content.png.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/per_sequence_gc_content.png.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/per_sequence_quality.png.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/per_tile_quality.png.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/sequence_length_distribution.png.svn-base  \n",
            "   creating: FastQC/Help/3 Analysis Modules/.svn/tmp/\n",
            "   creating: FastQC/Help/3 Analysis Modules/.svn/tmp/props/\n",
            "  inflating: FastQC/Help/3 Analysis Modules/1 Basic Statistics.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/10 Adapter Content.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/11 Kmer Content.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/12 Per Tile Sequence Quality.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/2 Per Base Sequence Quality.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/3 Per Sequence Quality Scores.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/4 Per Base Sequence Content.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/5 Per Sequence GC Content.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/6 Per Base N Content.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/7 Sequence Length Distribution.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/8 Duplicate Sequences.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/9 Overrepresented Sequences.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/duplication_levels.png  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/kmer_profiles.png  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/per_base_gc_content.png  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/per_base_n_content.png  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/per_base_quality.png  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/per_base_sequence_content.png  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/per_sequence_gc_content.png  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/per_sequence_quality.png  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/per_tile_quality.png  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/sequence_length_distribution.png  \n",
            "  inflating: FastQC/INSTALL.txt      \n",
            "  inflating: FastQC/jbzip2-0.9.jar   \n",
            "  inflating: FastQC/LICENSE          \n",
            "  inflating: FastQC/LICENSE.txt      \n",
            "  inflating: FastQC/LICENSE_JHDF5.txt  \n",
            "   creating: FastQC/net/\n",
            "   creating: FastQC/net/sourceforge/\n",
            "   creating: FastQC/net/sourceforge/iharder/\n",
            "   creating: FastQC/net/sourceforge/iharder/base64/\n",
            "  inflating: FastQC/net/sourceforge/iharder/base64/Base64$1.class  \n",
            "  inflating: FastQC/net/sourceforge/iharder/base64/Base64$InputStream.class  \n",
            "  inflating: FastQC/net/sourceforge/iharder/base64/Base64$OutputStream.class  \n",
            "  inflating: FastQC/net/sourceforge/iharder/base64/Base64.class  \n",
            "   creating: FastQC/org/\n",
            "   creating: FastQC/org/apache/\n",
            "   creating: FastQC/org/apache/commons/\n",
            "   creating: FastQC/org/apache/commons/math3/\n",
            "   creating: FastQC/org/apache/commons/math3/analysis/\n",
            "   creating: FastQC/org/apache/commons/math3/analysis/solvers/\n",
            "  inflating: FastQC/org/apache/commons/math3/analysis/solvers/AbstractUnivariateSolver.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/analysis/solvers/AllowedSolution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/analysis/solvers/BaseAbstractUnivariateSolver.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/analysis/solvers/BaseUnivariateSolver.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/analysis/solvers/BracketedUnivariateSolver.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/analysis/solvers/BrentSolver.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/analysis/solvers/UnivariateSolver.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/analysis/solvers/UnivariateSolverUtils.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/analysis/UnivariateFunction.class  \n",
            "   creating: FastQC/org/apache/commons/math3/distribution/\n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/AbstractIntegerDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/AbstractRealDistribution$1.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/AbstractRealDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/BetaDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/BinomialDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/CauchyDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/ChiSquaredDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/FDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/GammaDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/HypergeometricDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/IntegerDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/NormalDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/PascalDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/PoissonDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/RealDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/SaddlePointExpansion.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/TDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/WeibullDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/ZipfDistribution.class  \n",
            "   creating: FastQC/org/apache/commons/math3/exception/\n",
            "  inflating: FastQC/org/apache/commons/math3/exception/ConvergenceException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/DimensionMismatchException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/MathArithmeticException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/MathIllegalArgumentException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/MathIllegalNumberException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/MathIllegalStateException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/MathInternalError.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/MaxCountExceededException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/NoBracketingException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/NotFiniteNumberException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/NotPositiveException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/NotStrictlyPositiveException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/NullArgumentException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/NumberIsTooLargeException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/NumberIsTooSmallException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/OutOfRangeException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/TooManyEvaluationsException.class  \n",
            "   creating: FastQC/org/apache/commons/math3/exception/util/\n",
            "  inflating: FastQC/org/apache/commons/math3/exception/util/ArgUtils.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/util/ExceptionContext.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/util/ExceptionContextProvider.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/util/Localizable.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/util/LocalizedFormats.class  \n",
            "   creating: FastQC/org/apache/commons/math3/random/\n",
            "  inflating: FastQC/org/apache/commons/math3/random/AbstractWell.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/random/BitsStreamGenerator.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/random/RandomData.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/random/RandomDataImpl.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/random/RandomGenerator.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/random/Well19937c.class  \n",
            "   creating: FastQC/org/apache/commons/math3/special/\n",
            "  inflating: FastQC/org/apache/commons/math3/special/Beta$1.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/special/Beta.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/special/Erf.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/special/Gamma$1.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/special/Gamma.class  \n",
            "   creating: FastQC/org/apache/commons/math3/util/\n",
            "  inflating: FastQC/org/apache/commons/math3/util/ArithmeticUtils.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/ContinuedFraction.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/DoubleArray.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/FastMath$ExpFracTable.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/FastMath$ExpIntTable.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/FastMath$lnMant.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/FastMath.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/FastMathCalc.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/FastMathLiteralArrays.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/Incrementor$1.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/Incrementor$MaxCountExceededCallback.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/Incrementor.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/MathUtils.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/Precision.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/ResizableDoubleArray.class  \n",
            "  inflating: FastQC/README.md        \n",
            "  inflating: FastQC/README.txt       \n",
            "  inflating: FastQC/RELEASE_NOTES.txt  \n",
            "  inflating: FastQC/run_fastqc.bat   \n",
            "  inflating: FastQC/sam-1.103.jar    \n",
            "   creating: FastQC/Templates/\n",
            "  inflating: FastQC/Templates/fastqc2fo.xsl  \n",
            "  inflating: FastQC/Templates/header_template.html  \n",
            "   creating: FastQC/Templates/Icons/\n",
            " extracting: FastQC/Templates/Icons/error.png  \n",
            " extracting: FastQC/Templates/Icons/fastqc_icon.png  \n",
            " extracting: FastQC/Templates/Icons/tick.png  \n",
            " extracting: FastQC/Templates/Icons/warning.png  \n",
            "   creating: FastQC/uk/\n",
            "   creating: FastQC/uk/ac/\n",
            "   creating: FastQC/uk/ac/babraham/\n",
            "   creating: FastQC/uk/ac/babraham/FastQC/\n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Analysis/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Analysis/AnalysisListener.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Analysis/AnalysisQueue.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Analysis/AnalysisRunner.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Analysis/OfflineRunner.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Dialogs/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Dialogs/AboutDialog$1.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Dialogs/AboutDialog.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Dialogs/FastQCTitlePanel$SmoothJLabel.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Dialogs/FastQCTitlePanel.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Dialogs/WelcomePanel.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/FastQCApplication$1.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/FastQCApplication.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/FastQCConfig.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/FastQCMenuBar.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/FileFilters/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/FileFilters/BAMFileFilter.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/FileFilters/CasavaFastQFileFilter.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/FileFilters/FastQFileFilter.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/FileFilters/GobyFileFilter.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/FileFilters/MappedBAMFileFilter.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/FileFilters/SequenceFileFilter.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Graphs/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Graphs/BaseGroup.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Graphs/LineGraph.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Graphs/QualityBoxPlot.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Graphs/TileGraph.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Help/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Help/HelpDialog.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Help/HelpIndexRoot$FileSorter.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Help/HelpIndexRoot.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Help/HelpPage.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Help/HelpPageDisplay$HelpEditor.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Help/HelpPageDisplay.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Help/HelpSearchPanel.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Modules/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/AbstractQCModule.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/AdapterContent$Adapter.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/AdapterContent$ResultsTable.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/AdapterContent.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/BasicStats$ResultsTable.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/BasicStats.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/DuplicationLevel.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Modules/GCModel/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/GCModel/GCModel.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/GCModel/GCModelValue.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/KmerContent$Kmer.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/KmerContent$ResultsTable.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/KmerContent.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/ModuleConfig.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/ModuleFactory.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/NContent.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/OverRepresentedSeqs$OverrepresentedSeq.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/OverRepresentedSeqs$ResultsTable.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/OverRepresentedSeqs.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/PerBaseQualityScores.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/PerBaseSequenceContent.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/PerSequenceGCContent.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/PerSequenceQualityScores.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/PerTileQualityScores.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/QCModule.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/SequenceLengthDistribution.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Report/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Report/HTMLReportArchive.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Report/stylesheet.txt  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Resources/\n",
            " extracting: FastQC/uk/ac/babraham/FastQC/Resources/error.png  \n",
            " extracting: FastQC/uk/ac/babraham/FastQC/Resources/fastqc_icon.png  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Resources/fastqc_icon.svg  \n",
            " extracting: FastQC/uk/ac/babraham/FastQC/Resources/fastqc_icon_100.png  \n",
            " extracting: FastQC/uk/ac/babraham/FastQC/Resources/tick.png  \n",
            " extracting: FastQC/uk/ac/babraham/FastQC/Resources/warning.png  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Results/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Results/ResultsPanel$ModuleRenderer.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Results/ResultsPanel.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Sequence/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/BAMFile.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Sequence/Contaminant/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/Contaminant/Contaminant.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/Contaminant/ContaminantHit.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/Contaminant/ContaminentFinder.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/Fast5File.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/FastQFile.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Sequence/QualityEncoding/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/QualityEncoding/PhredEncoding.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/Sequence.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/SequenceFactory.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/SequenceFile.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/SequenceFileGroup.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/SequenceFormatException.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Statistics/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Statistics/NormalDistribution.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Statistics/PearsonCorrelation.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Utilities/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Utilities/CasavaBasename.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Utilities/HotColdColourGradient.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Utilities/ImageToBase64.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Utilities/MultiMemberGZIPInputStream.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Utilities/NameFormatException.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Utilities/NanoporeBasename.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Utilities/QualityCount.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Utilities/RGB.class  \n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "Можно получать данные секвенировия с ENA. Например, https://www.ebi.ac.uk/ena/browser/view/SRR5836473"
      ],
      "metadata": {
        "id": "sLEmMt1slFZp"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "! wget ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR583/003/SRR5836473/SRR5836473_1.fastq.gz\n",
        "! wget ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR583/003/SRR5836473/SRR5836473_2.fastq.gz"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "EhKlO8TBlKPx",
        "outputId": "11e9c05e-13cc-4d4a-f425-ff36a0e60d10"
      },
      "execution_count": 7,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "--2022-02-16 07:17:38--  ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR583/003/SRR5836473/SRR5836473_1.fastq.gz\n",
            "           => ‘SRR5836473_1.fastq.gz’\n",
            "Resolving ftp.sra.ebi.ac.uk (ftp.sra.ebi.ac.uk)... 193.62.197.74\n",
            "Connecting to ftp.sra.ebi.ac.uk (ftp.sra.ebi.ac.uk)|193.62.197.74|:21... connected.\n",
            "Logging in as anonymous ... Logged in!\n",
            "==> SYST ... done.    ==> PWD ... done.\n",
            "==> TYPE I ... done.  ==> CWD (1) /vol1/fastq/SRR583/003/SRR5836473 ... done.\n",
            "==> SIZE SRR5836473_1.fastq.gz ... 6856386252\n",
            "==> PASV ... done.    ==> RETR SRR5836473_1.fastq.gz ... done.\n",
            "Length: 6856386252 (6.4G) (unauthoritative)\n",
            "\n",
            "SRR5836473_1.fastq. 100%[===================>]   6.38G  26.1MB/s    in 6m 7s   \n",
            "\n",
            "2022-02-16 07:23:46 (17.8 MB/s) - ‘SRR5836473_1.fastq.gz’ saved [6856386252]\n",
            "\n",
            "--2022-02-16 07:23:46--  ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR583/003/SRR5836473/SRR5836473_2.fastq.gz\n",
            "           => ‘SRR5836473_2.fastq.gz’\n",
            "Resolving ftp.sra.ebi.ac.uk (ftp.sra.ebi.ac.uk)... 193.62.193.138\n",
            "Connecting to ftp.sra.ebi.ac.uk (ftp.sra.ebi.ac.uk)|193.62.193.138|:21... connected.\n",
            "Logging in as anonymous ... Logged in!\n",
            "==> SYST ... done.    ==> PWD ... done.\n",
            "==> TYPE I ... done.  ==> CWD (1) /vol1/fastq/SRR583/003/SRR5836473 ... done.\n",
            "==> SIZE SRR5836473_2.fastq.gz ... 8127008133\n",
            "==> PASV ... done.    ==> RETR SRR5836473_2.fastq.gz ... done.\n",
            "Length: 8127008133 (7.6G) (unauthoritative)\n",
            "\n",
            "SRR5836473_2.fastq. 100%[===================>]   7.57G  36.2MB/s    in 4m 4s   \n",
            "\n",
            "2022-02-16 07:27:51 (31.8 MB/s) - ‘SRR5836473_2.fastq.gz’ saved [8127008133]\n",
            "\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "! wget ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR382/002/SRR3824222/SRR3824222_1.fastq.gz\n",
        "! wget ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR382/002/SRR3824222/SRR3824222_2.fastq.gz"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "iBSIE23NtH1N",
        "outputId": "86c1b17d-1ca8-4575-f136-8de6fff466db"
      },
      "execution_count": 8,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "--2022-02-16 07:27:51--  ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR382/002/SRR3824222/SRR3824222_1.fastq.gz\n",
            "           => ‘SRR3824222_1.fastq.gz’\n",
            "Resolving ftp.sra.ebi.ac.uk (ftp.sra.ebi.ac.uk)... 193.62.193.138\n",
            "Connecting to ftp.sra.ebi.ac.uk (ftp.sra.ebi.ac.uk)|193.62.193.138|:21... connected.\n",
            "Logging in as anonymous ... Logged in!\n",
            "==> SYST ... done.    ==> PWD ... done.\n",
            "==> TYPE I ... done.  ==> CWD (1) /vol1/fastq/SRR382/002/SRR3824222 ... done.\n",
            "==> SIZE SRR3824222_1.fastq.gz ... 8096784759\n",
            "==> PASV ... done.    ==> RETR SRR3824222_1.fastq.gz ... done.\n",
            "Length: 8096784759 (7.5G) (unauthoritative)\n",
            "\n",
            "SRR3824222_1.fastq. 100%[===================>]   7.54G  36.1MB/s    in 4m 1s   \n",
            "\n",
            "2022-02-16 07:31:53 (32.1 MB/s) - ‘SRR3824222_1.fastq.gz’ saved [8096784759]\n",
            "\n",
            "--2022-02-16 07:31:53--  ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR382/002/SRR3824222/SRR3824222_2.fastq.gz\n",
            "           => ‘SRR3824222_2.fastq.gz’\n",
            "Resolving ftp.sra.ebi.ac.uk (ftp.sra.ebi.ac.uk)... 193.62.197.74\n",
            "Connecting to ftp.sra.ebi.ac.uk (ftp.sra.ebi.ac.uk)|193.62.197.74|:21... connected.\n",
            "Logging in as anonymous ... Logged in!\n",
            "==> SYST ... done.    ==> PWD ... done.\n",
            "==> TYPE I ... done.  ==> CWD (1) /vol1/fastq/SRR382/002/SRR3824222 ... done.\n",
            "==> SIZE SRR3824222_2.fastq.gz ... 10233274189\n",
            "==> PASV ... done.    ==> RETR SRR3824222_2.fastq.gz ... done.\n",
            "Length: 10233274189 (9.5G) (unauthoritative)\n",
            "\n",
            "SRR3824222_2.fastq. 100%[===================>]   9.53G  22.4MB/s    in 6m 38s  \n",
            "\n",
            "2022-02-16 07:38:34 (24.5 MB/s) - ‘SRR3824222_2.fastq.gz’ saved [10233274189]\n",
            "\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "! wget ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR583/005/SRR5836475/SRR5836475_1.fastq.gz\n",
        "! wget ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR583/005/SRR5836475/SRR5836475_2.fastq.gz"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "HwHzPBnrtd6N",
        "outputId": "ced1579f-f809-4d40-c846-bc2fd3ba0495"
      },
      "execution_count": 9,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "--2022-02-16 07:38:34--  ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR583/005/SRR5836475/SRR5836475_1.fastq.gz\n",
            "           => ‘SRR5836475_1.fastq.gz’\n",
            "Resolving ftp.sra.ebi.ac.uk (ftp.sra.ebi.ac.uk)... 193.62.197.74\n",
            "Connecting to ftp.sra.ebi.ac.uk (ftp.sra.ebi.ac.uk)|193.62.197.74|:21... connected.\n",
            "Logging in as anonymous ... Logged in!\n",
            "==> SYST ... done.    ==> PWD ... done.\n",
            "==> TYPE I ... done.  ==> CWD (1) /vol1/fastq/SRR583/005/SRR5836475 ... done.\n",
            "==> SIZE SRR5836475_1.fastq.gz ... 6139527111\n",
            "==> PASV ... done.    ==> RETR SRR5836475_1.fastq.gz ... done.\n",
            "Length: 6139527111 (5.7G) (unauthoritative)\n",
            "\n",
            "SRR5836475_1.fastq. 100%[===================>]   5.72G  17.0MB/s    in 4m 28s  \n",
            "\n",
            "2022-02-16 07:43:03 (21.8 MB/s) - ‘SRR5836475_1.fastq.gz’ saved [6139527111]\n",
            "\n",
            "--2022-02-16 07:43:04--  ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR583/005/SRR5836475/SRR5836475_2.fastq.gz\n",
            "           => ‘SRR5836475_2.fastq.gz’\n",
            "Resolving ftp.sra.ebi.ac.uk (ftp.sra.ebi.ac.uk)... 193.62.193.138\n",
            "Connecting to ftp.sra.ebi.ac.uk (ftp.sra.ebi.ac.uk)|193.62.193.138|:21... connected.\n",
            "Logging in as anonymous ... Logged in!\n",
            "==> SYST ... done.    ==> PWD ... done.\n",
            "==> TYPE I ... done.  ==> CWD (1) /vol1/fastq/SRR583/005/SRR5836475 ... done.\n",
            "==> SIZE SRR5836475_2.fastq.gz ... 7492692021\n",
            "==> PASV ... done.    ==> RETR SRR5836475_2.fastq.gz ... done.\n",
            "Length: 7492692021 (7.0G) (unauthoritative)\n",
            "\n",
            "SRR5836475_2.fastq. 100%[===================>]   6.98G  36.7MB/s    in 3m 18s  \n",
            "\n",
            "2022-02-16 07:46:23 (36.1 MB/s) - ‘SRR5836475_2.fastq.gz’ saved [7492692021]\n",
            "\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "# Установка Bismark с зависимостями (+bowtie2, samtools)\n"
      ],
      "metadata": {
        "id": "xGgoXS72lnQl"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "! pip install -q condacolab\n",
        "import condacolab\n",
        "condacolab.install()"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "bvBgsNgWloVn",
        "outputId": "ee15dc4f-e558-485b-abbc-a438f61f0a9d"
      },
      "execution_count": 2,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "✨🍰✨ Everything looks OK!\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "import condacolab\n",
        "condacolab.check()"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "OauJoS2MlsNQ",
        "outputId": "794db67e-482c-4844-a6b3-8bd54f6caa7f"
      },
      "execution_count": 3,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "✨🍰✨ Everything looks OK!\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "! conda install -c bioconda bismark"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "iVf9ymjlluRR",
        "outputId": "7779bb38-5b1d-4d01-998c-e98297ae9919"
      },
      "execution_count": 4,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Collecting package metadata (current_repodata.json): - \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\bdone\n",
            "Solving environment: | \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\bdone\n",
            "\n",
            "## Package Plan ##\n",
            "\n",
            "  environment location: /usr/local\n",
            "\n",
            "  added / updated specs:\n",
            "    - bismark\n",
            "\n",
            "\n",
            "The following packages will be downloaded:\n",
            "\n",
            "    package                    |            build\n",
            "    ---------------------------|-----------------\n",
            "    bismark-0.23.1             |       hdfd78af_0         6.6 MB  bioconda\n",
            "    bowtie2-2.2.5              |   py37h22450f8_8        14.9 MB  bioconda\n",
            "    ca-certificates-2021.10.8  |       ha878542_0         139 KB  conda-forge\n",
            "    certifi-2021.10.8          |   py37h89c1867_1         145 KB  conda-forge\n",
            "    conda-4.11.0               |   py37h89c1867_0        16.9 MB  conda-forge\n",
            "    hisat2-2.2.1               |       h1b792b2_3        16.5 MB  bioconda\n",
            "    perl-5.32.1                | 0_h7f98852_perl5        14.5 MB  conda-forge\n",
            "    python_abi-3.7             |          2_cp37m           4 KB  conda-forge\n",
            "    samtools-1.3.1             |                0         1.5 MB  bioconda\n",
            "    ------------------------------------------------------------\n",
            "                                           Total:        71.1 MB\n",
            "\n",
            "The following NEW packages will be INSTALLED:\n",
            "\n",
            "  bismark            bioconda/noarch::bismark-0.23.1-hdfd78af_0\n",
            "  bowtie2            bioconda/linux-64::bowtie2-2.2.5-py37h22450f8_8\n",
            "  hisat2             bioconda/linux-64::hisat2-2.2.1-h1b792b2_3\n",
            "  perl               conda-forge/linux-64::perl-5.32.1-0_h7f98852_perl5\n",
            "  samtools           bioconda/linux-64::samtools-1.3.1-0\n",
            "\n",
            "The following packages will be UPDATED:\n",
            "\n",
            "  ca-certificates                      2020.12.5-ha878542_0 --> 2021.10.8-ha878542_0\n",
            "  certifi                          2020.12.5-py37h89c1867_1 --> 2021.10.8-py37h89c1867_1\n",
            "  conda                                4.9.2-py37h89c1867_0 --> 4.11.0-py37h89c1867_0\n",
            "  python_abi                                    3.7-1_cp37m --> 3.7-2_cp37m\n",
            "\n",
            "\n",
            "\n",
            "Downloading and Extracting Packages\n",
            "python_abi-3.7       | 4 KB      | : 100% 1.0/1 [00:00<00:00, 15.42it/s]\n",
            "ca-certificates-2021 | 139 KB    | : 100% 1.0/1 [00:00<00:00, 20.12it/s]\n",
            "certifi-2021.10.8    | 145 KB    | : 100% 1.0/1 [00:00<00:00, 18.97it/s]\n",
            "bowtie2-2.2.5        | 14.9 MB   | : 100% 1.0/1 [00:02<00:00,  2.48s/it]\n",
            "samtools-1.3.1       | 1.5 MB    | : 100% 1.0/1 [00:00<00:00,  3.62it/s]\n",
            "hisat2-2.2.1         | 16.5 MB   | : 100% 1.0/1 [00:02<00:00,  2.93s/it]               \n",
            "conda-4.11.0         | 16.9 MB   | : 100% 1.0/1 [00:04<00:00,  4.02s/it]\n",
            "bismark-0.23.1       | 6.6 MB    | : 100% 1.0/1 [00:01<00:00,  1.19s/it]\n",
            "perl-5.32.1          | 14.5 MB   | : 100% 1.0/1 [00:03<00:00,  3.52s/it]\n",
            "Preparing transaction: - \b\b\\ \b\bdone\n",
            "Verifying transaction: / \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\bdone\n",
            "Executing transaction: - \b\b\\ \b\b| \b\bdone\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "! samtools view"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "igMfZL3MlwiX",
        "outputId": "25570ffd-188b-4b42-e39c-458b31746990"
      },
      "execution_count": 5,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "\n",
            "Usage: samtools view [options] <in.bam>|<in.sam>|<in.cram> [region ...]\n",
            "\n",
            "Options:\n",
            "  -b       output BAM\n",
            "  -C       output CRAM (requires -T)\n",
            "  -1       use fast BAM compression (implies -b)\n",
            "  -u       uncompressed BAM output (implies -b)\n",
            "  -h       include header in SAM output\n",
            "  -H       print SAM header only (no alignments)\n",
            "  -c       print only the count of matching records\n",
            "  -o FILE  output file name [stdout]\n",
            "  -U FILE  output reads not selected by filters to FILE [null]\n",
            "  -t FILE  FILE listing reference names and lengths (see long help) [null]\n",
            "  -L FILE  only include reads overlapping this BED FILE [null]\n",
            "  -r STR   only include reads in read group STR [null]\n",
            "  -R FILE  only include reads with read group listed in FILE [null]\n",
            "  -q INT   only include reads with mapping quality >= INT [0]\n",
            "  -l STR   only include reads in library STR [null]\n",
            "  -m INT   only include reads with number of CIGAR operations consuming\n",
            "           query sequence >= INT [0]\n",
            "  -f INT   only include reads with all bits set in INT set in FLAG [0]\n",
            "  -F INT   only include reads with none of the bits set in INT set in FLAG [0]\n",
            "  -x STR   read tag to strip (repeatable) [null]\n",
            "  -B       collapse the backward CIGAR operation\n",
            "  -s FLOAT integer part sets seed of random number generator [0];\n",
            "           rest sets fraction of templates to subsample [no subsampling]\n",
            "  -@, --threads INT\n",
            "           number of BAM/CRAM compression threads [0]\n",
            "  -?       print long help, including note about region specification\n",
            "  -S       ignored (input format is auto-detected)\n",
            "      --input-fmt-option OPT[=VAL]\n",
            "               Specify a single input file format option in the form\n",
            "               of OPTION or OPTION=VALUE\n",
            "  -O, --output-fmt FORMAT[,OPT[=VAL]]...\n",
            "               Specify output format (SAM, BAM, CRAM)\n",
            "      --output-fmt-option OPT[=VAL]\n",
            "               Specify a single output file format option in the form\n",
            "               of OPTION or OPTION=VALUE\n",
            "  -T, --reference FILE\n",
            "               Reference sequence FASTA FILE [null]\n",
            "\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "Можно загрузить файл с google drive напрямую:"
      ],
      "metadata": {
        "id": "NH8wkrT-l0aU"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "! pip install -U --no-cache-dir gdown --pre"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "sRfunwOn-zAS",
        "outputId": "e8b7269d-49dc-45c8-9176-a5343dc8bb63"
      },
      "execution_count": 6,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Collecting gdown\n",
            "  Downloading gdown-4.2.1.tar.gz (13 kB)\n",
            "  Installing build dependencies ... \u001b[?25l\u001b[?25hdone\n",
            "  Getting requirements to build wheel ... \u001b[?25l\u001b[?25hdone\n",
            "    Preparing wheel metadata ... \u001b[?25l\u001b[?25hdone\n",
            "Collecting beautifulsoup4\n",
            "  Downloading beautifulsoup4-4.10.0-py3-none-any.whl (97 kB)\n",
            "\u001b[K     |████████████████████████████████| 97 kB 29.8 MB/s \n",
            "\u001b[?25hRequirement already satisfied: requests[socks] in /usr/local/lib/python3.7/site-packages (from gdown) (2.25.1)\n",
            "Requirement already satisfied: tqdm in /usr/local/lib/python3.7/site-packages (from gdown) (4.59.0)\n",
            "Collecting filelock\n",
            "  Downloading filelock-3.5.0-py3-none-any.whl (10.0 kB)\n",
            "Requirement already satisfied: six in /usr/local/lib/python3.7/site-packages (from gdown) (1.15.0)\n",
            "Collecting soupsieve>1.2\n",
            "  Downloading soupsieve-2.3.1-py3-none-any.whl (37 kB)\n",
            "Requirement already satisfied: certifi>=2017.4.17 in /usr/local/lib/python3.7/site-packages (from requests[socks]->gdown) (2021.10.8)\n",
            "Requirement already satisfied: idna<3,>=2.5 in /usr/local/lib/python3.7/site-packages (from requests[socks]->gdown) (2.10)\n",
            "Requirement already satisfied: urllib3<1.27,>=1.21.1 in /usr/local/lib/python3.7/site-packages (from requests[socks]->gdown) (1.26.3)\n",
            "Requirement already satisfied: chardet<5,>=3.0.2 in /usr/local/lib/python3.7/site-packages (from requests[socks]->gdown) (4.0.0)\n",
            "Requirement already satisfied: PySocks!=1.5.7,>=1.5.6 in /usr/local/lib/python3.7/site-packages (from requests[socks]->gdown) (1.7.1)\n",
            "Building wheels for collected packages: gdown\n",
            "  Building wheel for gdown (PEP 517) ... \u001b[?25l\u001b[?25hdone\n",
            "  Created wheel for gdown: filename=gdown-4.2.1-py3-none-any.whl size=14429 sha256=a6c5e8172fa6c198d8f4f2b67226c09e59cf08edac3903d292a7f279f38217fa\n",
            "  Stored in directory: /tmp/pip-ephem-wheel-cache-0gpmp2wl/wheels/c2/33/c3/621b1f6276c588264acc8ca46f7285cb541df1e4620d76e8cd\n",
            "Successfully built gdown\n",
            "Installing collected packages: soupsieve, filelock, beautifulsoup4, gdown\n",
            "Successfully installed beautifulsoup4-4.10.0 filelock-3.5.0 gdown-4.2.1 soupsieve-2.3.1\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!rm ~/.cache/gdown/cookies.json"
      ],
      "metadata": {
        "id": "Ydg0RcV7-17Q"
      },
      "execution_count": 37,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "# bam файл\n",
        "! gdown --id 1SE4qg_4hzE2YkPXZbFMwKC3jaK09ZHDy\n",
        "! gdown --id 1ba0PJTxtL7hAcj-K0cWMA5W25bZLIgvn\n",
        "! gdown --id 1U5wI_ik6HOYnuXmE1ipgWxwK0hcYruw3"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "EEuceeiNl3KL",
        "outputId": "0bc18cc1-71bd-4f48-eaa9-be2c6f7da6d6"
      },
      "execution_count": 38,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Downloading...\n",
            "From: https://drive.google.com/uc?id=1SE4qg_4hzE2YkPXZbFMwKC3jaK09ZHDy\n",
            "To: /content/SRR5836475_1_bismark_bt2_pe.bam\n",
            "100% 960M/960M [00:09<00:00, 99.4MB/s]\n",
            "Access denied with the following error:\n",
            "\n",
            " \tCannot retrieve the public link of the file. You may need to change\n",
            "\tthe permission to 'Anyone with the link', or have had many accesses. \n",
            "\n",
            "You may still be able to access the file from the browser:\n",
            "\n",
            "\t https://drive.google.com/uc?id=1ba0PJTxtL7hAcj-K0cWMA5W25bZLIgvn \n",
            "\n",
            "Downloading...\n",
            "From: https://drive.google.com/uc?id=1U5wI_ik6HOYnuXmE1ipgWxwK0hcYruw3\n",
            "To: /content/SRR3824222_1_bismark_bt2_pe.bam\n",
            "100% 1.56G/1.56G [00:14<00:00, 109MB/s]\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "# отчет о выравнивании тоже нужен\n",
        "! gdown --id 1tnhtwKBO2Fszj-Im30XuuADtO7oIN5-B\n",
        "! gdown --id 1a-USP32jw0n6L9pRqsbUCjt4j8VopPBg\n",
        "! gdown --id 14vyhf4aKfEPXR2yfK1Rk_f3Muzf1hL40"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "_QBbM2VGl-Wu",
        "outputId": "58394fee-1094-493f-8efb-fc74be59fe64"
      },
      "execution_count": 39,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Downloading...\n",
            "From: https://drive.google.com/uc?id=1tnhtwKBO2Fszj-Im30XuuADtO7oIN5-B\n",
            "To: /content/SRR5836473_1_bismark_bt2_PE_report.txt\n",
            "100% 1.90k/1.90k [00:00<00:00, 3.47MB/s]\n",
            "Downloading...\n",
            "From: https://drive.google.com/uc?id=1a-USP32jw0n6L9pRqsbUCjt4j8VopPBg\n",
            "To: /content/SRR3824222_1_bismark_bt2_PE_report.txt\n",
            "100% 1.90k/1.90k [00:00<00:00, 3.57MB/s]\n",
            "Downloading...\n",
            "From: https://drive.google.com/uc?id=14vyhf4aKfEPXR2yfK1Rk_f3Muzf1hL40\n",
            "To: /content/SRR5836475_1_bismark_bt2_PE_report.txt\n",
            "100% 1.89k/1.89k [00:00<00:00, 3.81MB/s]\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "! ls -lh"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "f2oiW57_mA76",
        "outputId": "7e85b8fd-101e-4220-a2cb-9275840be7a7"
      },
      "execution_count": 40,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "total 47G\n",
            "-rw-r--r-- 1 root root  14K Feb 16 07:47 condacolab_install.log\n",
            "drwxr-xr-x 8 root root 4.0K Feb 16 07:17 FastQC\n",
            "-rw-r--r-- 1 root root 9.8M Jan 15  2020 fastqc_v0.11.9.zip\n",
            "drwxr-xr-x 1 root root 4.0K Feb  1 14:32 sample_data\n",
            "-rw------- 1 root root 1.5G Feb 16 07:56 SRR3824222_1_bismark_bt2_pe.bam\n",
            "-rw------- 1 root root 1.9K Feb 16 07:57 SRR3824222_1_bismark_bt2_PE_report.txt\n",
            "-rw-r--r-- 1 root root 7.6G Feb 16 07:31 SRR3824222_1.fastq.gz\n",
            "-rw-r--r-- 1 root root 9.6G Feb 16 07:38 SRR3824222_2.fastq.gz\n",
            "-rw------- 1 root root 636M Feb 16 07:53 SRR5836473_1_bismark_bt2_pe.bam\n",
            "-rw------- 1 root root 1.9K Feb 16 07:57 SRR5836473_1_bismark_bt2_PE_report.txt\n",
            "-rw-r--r-- 1 root root 6.4G Feb 16 07:23 SRR5836473_1.fastq.gz\n",
            "-rw-r--r-- 1 root root 7.6G Feb 16 07:27 SRR5836473_2.fastq.gz\n",
            "-rw------- 1 root root 916M Feb 16 07:56 SRR5836475_1_bismark_bt2_pe.bam\n",
            "-rw------- 1 root root 1.9K Feb 16 07:57 SRR5836475_1_bismark_bt2_PE_report.txt\n",
            "-rw-r--r-- 1 root root 5.8G Feb 16 07:43 SRR5836475_1.fastq.gz\n",
            "-rw-r--r-- 1 root root 7.0G Feb 16 07:46 SRR5836475_2.fastq.gz\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "! samtools view SRR5836473_1_bismark_bt2_pe.bam | head\n",
        "! samtools view SRR3824222_1_bismark_bt2_pe.bam | head\n",
        "! samtools view SRR5836475_1_bismark_bt2_pe.bam | head"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "leeJqaLTmDHY",
        "outputId": "37256bbd-6162-46cc-ada4-5d3724f414dc"
      },
      "execution_count": 41,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "SRR5836473.18_18/1\t99\tchr11\t116830009\t0\t139M\t=\t116830059\t190\tTTTATTTTGTTATTGGATTTAGAGATTATTTGGCGGGAATCGGGTTTTTTTTTTTTTTTTTTATAATAGAGTGTCGTAATAGTAAAATTGAGTTTTGATTAGAGTGATTTTGATTTGGTTTCGTTTTTTTTTTCGTTCG\tFKKKKKKKKKKKKKKKK<FKKKKFKKKKKKKKKKKKKFKKKKKKFAFKKKKKKKKKKKKKKKAKFFFFKFF7FFFKFKK7AKKKKKKKKAFKKKKAFKKAFKFK,F<FAKKKAFAFFF,FKKKKKKKKKKKKFF<A77F\tNM:i:37\tMD:Z:0C1C2C0C2C2C6C0C23C0C0C0C0C1C0C0C0C0C2C0C2C4C0T8G2C12C4T1C7C0C4C4C1C4C3C2C3C2\tXM:Z:h.h..hx..h..x......x.............Z......Z...hhhhh.hhhhh..hh..h....x.......Z....x............h......x.......hh....h....h.xZ...h...h..xZ..xZ.\tXR:Z:CT\tXG:Z:CT\n",
            "SRR5836473.18_18/1\t147\tchr11\t116830059\t0\t140M\t=\t116830009\t-190\tTTTTTTTTTTTTATAATAGAGTGTCGTAATAGTAAAATTGAGTTTTGATTAGAGTGATTTTGATTTGGTTTCGTTTTTTTTTTCGTTCGTTTAGTTTTTTTTTTGTAGTTCGAGTTGTTATTTTAGTTGAATTGTTTATT\tKKF<A,KAFA7A7,K77KF<F<A<F<7,,7,7A7,,7AFFFA<,KKKAA<<,7KAF7KKFAA<AKFKKF7<KKKKFKKKKKFFKKFFFKKKKKKKKKFKKKKKKKKFKKFKKKKKKKKKKKKKKKKKKKKKKKKKKKKKF\tNM:i:39\tMD:Z:0C0C0C0C0C2C0C2C4C0T8G2C12C4T1C7C0C4C4C1C4C3C2C3C3C5C0C1C1C2T0C2C0C7C0C2C9C0C3C3\tXM:Z:hhhhh..hh..h....x.......Z....x............h......x.......hh....h....h.xZ...h...h..xZ..xZ..h.....hh.h.h...x..hxZ......hh..h.........xz...h...\tXR:Z:GA\tXG:Z:CT\n",
            "SRR5836473.30_30/1\t99\tchr11\t24232407\t0\t140M\t=\t24232506\t239\tGGTAAGAAGGATATATGTTTAATTATGTTTATAGTAGTTTTATTTATAATAGTTAGAAGTTGGAAAGAATTTAGATGTTTTTCAATAGAGGAATGGATATTAAAAAATGTGGTATATTTATATAATGGAGTATTATTTAG\tKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKFFKKKKKKKKKKKKKKKKKKKFKKKKKAFKK\tNM:i:29\tMD:Z:0T10C1C3C1C0C1C6C4C2C0C13C0C5C9C0C0C5C0C0C0C4C13C0C13C5C1C12C1C2\tXM:Z:...........h.h...h.h..h......h....x..hh.............hx.....x.........hhx.....hhhh.H..x.............hh.............h.....h.h............h.x..\tXR:Z:CT\tXG:Z:CT\n",
            "SRR5836473.30_30/1\t147\tchr11\t24232506\t0\t140M\t=\t24232407\t-239\tTTAAAAAATGTGGTATATTTATATAATGGAGTATTATTTAGTTATTAAAAAGAATGAATTTATGAAATTTTTAGGTAAATGGATGGATTTGGAGGGTATTATTTTGAGTGAGGTAATTTAATTATAAAAGAATTTATATG\t7AKKKKFFFAA7AKFAKKFKFAKKKKKFA7KKKFAFFAKKKKKKFFKFKKKKKKKKKAKKKKKFKKKKKKKKKKKKKFKKKKKKKKKFKKKKKKKKKKKKKKA<AKFKF<AKKKKKKFKFKKKKKKKKKKKKKKKKKKFF\tNM:i:28\tMD:Z:0C0C13C5C1C12C1C2C27C0C4C0C10C0C7C2C2C0C12C0C0C3C1C7C1C1C1C0A0\tXM:Z:hh.............h.....h.h............h.x..h...........................hh....h...........hx.......h..h..hx............hhh...h.h.......h.h.h.h.\tXR:Z:GA\tXG:Z:CT\n",
            "SRR5836473.39_39/1\t83\tchr11\t105051790\t0\t13M1D127M\t=\t105051789\t-142\tTTAAAAACTACATATACTAATATACTAACAAACAAATAAACCAATAAAATAAAATTAAAAACCCAAAAATAAACCCACACACCTATAATCACTTAATCTTTATCAAAAAAACTAAAACCATCCAATAAAAAAAAAACACC\tKFKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKFKKKKKKKKKKKKFKKKKKFKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKFKKKKKKKKKKKKKKKKKKKKKKK\tNM:i:31\tMD:Z:0A8G3^G0G4G0G6G3G4G2G6G0G4G4G2G5G4G15G0G6G6G0A3G0G0G1G13G1G0G6G3G1\tXM:Z:.........x...h....xh......x...x....h..h......hh....h....h..h.....x....h...............hh......h......h....hhh.h.............x.hh......h.....\tXR:Z:CT\tXG:Z:GA\n",
            "SRR5836473.39_39/1\t163\tchr11\t105051789\t0\t14M1D126M\t=\t105051790\t142\tATTAAAAACTACATATACTAATATACTAACAAACAAATAAACCAATAAAATAAAATTAAAAACCCAAAAATAAACCCACACACCTATAATCACTTAATCTTTATCAAAAAAACTAAAACCATCCAATAAAAAAAAAACAC\tKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKFKKKKK<FKA<FKFFKKKKKKKKKFKFKKFK,,<FA,,7<\tNM:i:32\tMD:Z:0G0A8G3^G0G4G0G6G3G4G2G6G0G4G4G2G5G4G15G0G6G6G0A3G0G0G1G13G1G0G6G3G0\tXM:Z:h.........x...h....xh......x...x....h..h......hh....h....h..h.....x....h...............hh......h......h....hhh.h.............x.hh......h....\tXR:Z:GA\tXG:Z:GA\n",
            "SRR5836473.41_41/1\t99\tchr11\t40185837\t0\t140M\t=\t40185875\t178\tAATTTTTTTGGGGATTGTATTGAATTTGTAGATTGTTTTTGGTAAGAGAGTTATTTTTATTATATTGATTTTGTTAATTTATGAATATGGGAGATTTTTTTATTTTTTGAGATTTTTTTTAATTTTTTTTTTTAGAGATT\tKKKKKKKKKKKKKKKFKKKKKKKKKKKKKKFKKKKKKKKKFKKKKKKKKKFKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKF7<77FF\tNM:i:28\tMD:Z:6G0A9C7C9C6C4T2C0C7C9C0C2C0C3C0C5C9C3C0C2C2C6C2C8C3C2C5C1\tXM:Z:.................h.......x.........h......h.......hh.......h.........hx..hh...hh.....h.........h...hh..h..x......h..h........h...h..x.....h.\tXR:Z:CT\tXG:Z:CT\n",
            "SRR5836473.41_41/1\t147\tchr11\t40185875\t0\t140M\t=\t40185837\t-178\tTTGGTAAGAGAGTTATTTTTATTATATTGATTTTGTTAATTTATGAATATGGGAGATTTTTTTATTTTTTGAGATTTTTTTTAATTTTTTTTTTTAGAGATTTGAAGTTTTTATTATATAGATTTTTTATTTTTTTAGTG\tAKKKKKAAFFFKKKKKKKKKKKKKFFKKKKKKKKKKFAFKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKFKKKKKKKKKKKKFKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKK\tNM:i:31\tMD:Z:4C4T2C0C7C9C0C2C0C3C0C5C9C3C0C2C2C6C2C8C3C2C5C8C4C3A4C3C1C2C6T0\tXM:Z:....h.......hh.......h.........hx..hh...hh.....h.........h...hh..h..x......h..h........h...h..x.....h........h....h........h...h.h..h.......\tXR:Z:GA\tXG:Z:CT\n",
            "SRR5836473.46_46/1\t83\tchr11\t22771774\t8\t140M\t=\t22771594\t-320\tCAAATTAAAAAAAAAAAAATAAAAATATCACTTTTTACAAATAATATAATAATATATATAAATAACCCTAAAAATTCTATCACAAAACTCCTAAACCTAATAAACAACTTCAATAAAATAACTAAATATAAAATTAACTC\tKKKKAAKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKK\tNM:i:26\tMD:Z:6G0G4G2G2G1C15G2G2G4G3G9G1G13C1C2G1G9G3G7G5G1G2G2G2G0G15\tXM:Z:......hh....h..h..h.................h..x..h....h...h.........h.h....................h.........h...x.......x.....x.h..h..h..xh...............\tXR:Z:CT\tXG:Z:GA\n",
            "SRR5836473.46_46/1\t163\tchr11\t22771594\t8\t140M\t=\t22771774\t320\tAGAATAAAAACAATCTACAACAAACCAATAACCAACATCAAAATAAATAATAAAAAACTAAAAACAATCCCACTAAAATCAAAAACTAAACAAAACTACCCACTTTCTCCCTACCTATTCAACATTATACTTAAAATCCT\tFKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKAKKKKKKKKKKKKKKKKKKKKKKKKFFKFKKKKKKFKKKKKKK<AKKKKKKFFF7<A<K\tNM:i:25\tMD:Z:1T0G6G9G7G2G11G5G0G1G1G2G2G0G2G17G0G0G4G4G0G2G28G5G2G4\tXM:Z:..h......h.........x.......x..h...........h.....hh.h.h..h..xh..h.................xhh....h....hh..x............................h.....h..h....\tXR:Z:GA\tXG:Z:GA\n",
            "SRR3824222.40_40/1\t99\tchr11\t8735553\t16\t140M\t=\t8735587\t174\tTTAAAGGAGAATTGTTAATTTTAGATTTAGTAAGGTTTAGGTAAGAATTATGTTATTAGTAATATTACGATTTGTGGTTAAAATAGGAGTTGGAAGTTGTTTTTTTTTATTTTTGTTATTTTTATAGTTTGAGGAATTAA\tKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKK7FKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKK\tNM:i:29\tMD:Z:0C6T5A1C2C0C1C13C0C4C11C8C2C11C11C7C3C0C1C2C2C0C0C8C0C1C2C0C7C3\tXM:Z:h..............h..hh.x.............hh....h...........h........h..h.Z.........h...........x.......z...hh.h..h..hhx........hh.x..hx.......h...\tXR:Z:CT\tXG:Z:CT\n",
            "SRR3824222.40_40/1\t147\tchr11\t8735587\t16\t140M\t=\t8735553\t-174\tGTTTAGGTAAGAATTATGTTATTAGTAATATTACGATTTGTGGTTAAAATAGGAGTTGGAAGTTGTTTTTTTTTATTTTTGTTATTTTTATAGTTTGAGGAATTAAGGATATTTTGTAGTAATAGTAAGGGTGAAAAGTT\tKF<KKKKKKKKKKKKKKKFKKKKKKKKKFFAKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKK\tNM:i:30\tMD:Z:1C0C4C11C8C2C11C11C7C3C0C1C2C2C0C0C8C0C1C2C0C7C6C1C0C3C5C2C12C0C0\tXM:Z:.hh....h...........h........h..h.Z.........h...........x.......z...hh.h..h..hhx........hh.x..hx.......h......h.hh...x.....x..h............hx\tXR:Z:GA\tXG:Z:CT\n",
            "SRR3824222.42_42/1\t99\tchr11\t8735518\t0\t140M\t=\t8735603\t225\tAATTAGTATGTATAGTTTTGAGTATTGTGTAGTTTTTAAAGGAGAATTGTTAATTTTAGATTTAGTAAGGTTTAGGAAAGAATTATGTTATTAGTAATATTACGATTTGTGGTTAAAATAGGAGTTGGAAGTCGTTTTTT\tKKKKKKKKKKKKKKKKKKKKKKFFKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKFKKKKKKKKKKKKKKKKKKKKKKKKKAKKKKKKKKKKKKKKFKKKKKK7AK<FKKKF\tNM:i:28\tMD:Z:2C0C1C4C1C4C6C4C2C1C0C6T5A1C2C0C1C13C0C4C11C8C2C11C11C11C0C1C0\tXM:Z:..hh......h.x....x......x....x..h.hh..............h..hh.x.............hh................h........h..h.Z.........h...........x.......Z...hh.h\tXR:Z:CT\tXG:Z:CT\n",
            "SRR3824222.42_42/1\t147\tchr11\t8735603\t0\t140M\t=\t8735518\t-225\tTGTTATTAGTAATATTACGATTTGTGGTTAAAATAGGAGTTGGAAGTCGTTTTTTTTTATTTTTGTTATTTTTATAGTTTGAGGAATTAAGGATATTTTTTAGTAATAGTAAGGGTGAAAAGTTAGAGGTTATTTAAGGG\t<FAKKKKKKKKKAKKKKKKFKKKKKKFFAAFAFAKKKKKKFKKKKKKKKFKKKKKKKFKKKKKKKKKKKKKKKKKKKKKKKKKKKKFKKKKKKKKKFKKKFKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKK\tNM:i:31\tMD:Z:3C8C2C11C11C11C0C1C2C2C0C0C8C0C1C2C0C7C6C1C0C2G0C5C2C12C0C5C0C1G0C6\tXM:Z:...h........h..h.Z.........h...........x.......Z...hh.h..h..hhx........hh.x..hx.......h......h.hh...x.....x..h............hx.....hx..h......\tXR:Z:GA\tXG:Z:CT\n",
            "SRR3824222.52_52/1\t99\tchr11\t58717111\t23\t140M\t=\t58717131\t160\tTAAAAAATATTTACGGTTTTGTTTTTTTTTTTTAAAAGATATTAAGAGTTATAGGTGTGGGTCGTGATAGTATTTACGGGAGGAATCGGGTTAATGTTTATTTAAGTTAAGGTTAAAAGTTTATTTATTTATGGATGAGA\tKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKAFKKKKKKKKKKKKKFKKKKKKKKKKKKKKKKFKKKKKKKKK,FKKKKKKKKKKKKKKKKKKKKKKKKKKKFKKFKKKKKKKKKKKKKKKFK\tNM:i:39\tMD:Z:0C0C8C2A2C1C2C0C1C0C0C0C2C0C0C8C0C5C0C1C18C1C0C0C14A1C5C0C1C0C0C3C0C11C0C0C1C1C2C11\tXM:Z:h.........h.....h.x..hh.hhhh..hhh........hh.....hh.x..........Z.......h.hhh.Z.........Z....h.....hh.hhh...hh...........hhh.h.h..h...........\tXR:Z:CT\tXG:Z:CT\n",
            "SRR3824222.52_52/1\t147\tchr11\t58717131\t23\t140M\t=\t58717111\t-160\tGTTTTTTTTTTTTAAAAGATATTAAGAGTTATAGGTGTGGGTCGTGATAGTATTTACGGGAGGAATCGGGTTAATGTTTATTTAAGTTAAGGTTAAAAGTTTATTTATTTATGGATGAGAAAATTATTTGATTATTTTAG\t<7<,F<FFKF<<AAAKKF<<KFA,FFFKKFA,KKKKFFKFKFKKKKFKFKFAAAFAAKKKKKKKF,FKKKFKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKFFF\tNM:i:38\tMD:Z:1C0C1C0C0C0C2C0C0C8C0C5C0C1C18C1C0C0C14A1C5C0C1C0C0C3C0C11C0C0C1C1C2C15C7C1C0C1C2\tXM:Z:.hh.hhhh..hhh........hh.....hh.x..........Z.......h.hhh.Z.........Z....h.....hh.hhh...hh...........hhh.h.h..h...............h.......h.hh.x..\tXR:Z:GA\tXG:Z:CT\n",
            "SRR3824222.56_56/1\t99\tchr11\t87478638\t0\t140M\t=\t87478712\t212\tATAGTACGGTTTTTAATTTCTTTTTTTTTTATTAATTAAGTTTAATAAGGTAATGTTAAATTGAGGTGATTAAATGATTTTTTTATTCGTAGATGAGTGGGTTTTTAATTTTGGTTTGGGTGGACATTGATTTGATTTTT\tKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKFKKKKKKKAKKKK7<FKKKKKKKKKFAKKKKAKKFFKKKFFKKKFFAA,A7\tNM:i:32\tMD:Z:1C8C1C0C2C0C3C1C0C0C0C0C0C21C2C1C1T2C9C12C2C0T13C1A4C0C4C15C0C0C4C0C1\tXM:Z:.x....Z...h.hh..hh.H.h.hhhhhh.....................h..z.h....x.........h............h..x..............h......hh....h.........H.....hxz....hh.\tXR:Z:CT\tXG:Z:CT\n",
            "SRR3824222.56_56/1\t147\tchr11\t87478712\t0\t138M\t=\t87478638\t-212\tTGATTTTTTTATTCGTAGATGAGTGGGTTTTTAATTTTGGTTTGGGTGGACATTGATTTGATTTTTTTTGTGGGTGTTGTTACGATTTATATTTGTGGTTTTTGGTATTTTTTGGGGAGGGGAGGGAGAGTTATAGAT\tAFAFKFAKKKKKFKKAKKKKKKFKAAKF<FA<AFFKKKKKKAKKKKKKKKKKKKKKKKKKKKKKKKKKKFAFKF7FKKKKKKKKK7KKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKFF<FKA,F,KKKKKKKKKKKF\tNM:i:31\tMD:Z:9C2C0T13C1A4C0C4C15C0C0C4C0C1C0C0C7C3C4C0C0C1C1C0C5C1C7C16C4C0C0T5\tXM:Z:.........h..x..............h......hh....h.........H.....hxz....hh.hxz.......x...h.Z..hhh.h.hx.....h.h.......h.....................hh......\tXR:Z:GA\tXG:Z:CT\n",
            "SRR3824222.65_65/1\t83\tchr11\t28372568\t0\t140M\t=\t28372555\t-153\tCAACTCAATATTATTCTTAATAAAATATATTTCAAATTTATTCAATCAAACTTAACCAAAATTCCAACGTCCTACTTAAAATTCAACTAAAAAATTACCTATTCAACACAAATTAACATTATAAAATCAAAACTTTCTAC\tKKFKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKFKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKK\tNM:i:23\tMD:Z:7G1G14G1G12G4G9G3G0G0G8C19G0G2G11G3T5G9G3G1G1C1A2G1\tXM:Z:.......x.h..............h.h............h....x.........h...xhh.......Z....................hh..h...........x.........h.........h...x.h......x.\tXR:Z:CT\tXG:Z:GA\n",
            "SRR3824222.65_65/1\t163\tchr11\t28372555\t0\t140M\t=\t28372568\t153\tTACGAAAAAATATCAACTCAATATTATTCTTAATAAAATATATTTCAAATTTATTCAATCAAACTTAACCAAAATTCCAACGTCCTACTTAAAATTCAACTAAAAAATTACCTATTCAACACAAATTAACATTATAAAAT\tKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKFKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKFFKKKFKKKFKKKKKKKKKKKKA\tNM:i:23\tMD:Z:1G0T1G1G0G12G1G14G1G12G4G9G3G0G0G8C19G0G2G11G3T5G9G1\tXM:Z:.h.Hh.hh............x.h..............h.h............h....x.........h...xhh.......Z....................hh..h...........x.........h.........h.\tXR:Z:GA\tXG:Z:GA\n",
            "SRR5836475.1_1/1\t83\tchr11\t27610910\t0\t140M\t=\t27610897\t-153\tTTTACAAAAAATTTAAAACAAATAAAATAAAAACAAAATAAACCTTTAATAACCTTAAAAAAAATAACTACCCCTCTTTTATTTTAAAAAACCAATTCTTTAAAATACAATAAACACATTCAACTATTCCTTATCCCATA\tFFKKAKAKKKKKKKKKKKKKKKKKKFKKKKKKKKKFAKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKK\tNM:i:39\tMD:Z:1G1G3G7G0G8G0G1G2G0G3G0G1G0G0G5G0G1G5G0G0G0G0G1T3G2G10G1C5G1G12G0G1G1G2G1G3G14G6G0\tXM:Z:...h...h.......hh........hh.h..hh...hh.hhh.....hh.h.....hhhhh.....h..x..........h.......h.h............hh.h.z..h.h...z..............h......h\tXR:Z:CT\tXG:Z:GA\n",
            "SRR5836475.1_1/1\t163\tchr11\t27610897\t0\t140M\t=\t27610910\t153\tCCTTTAATATCAATTTACAAAAAATTTAAAACAAATAAAATAAAAACAAAATAAACCTTTAATAACCTTAAAAAAAATAACTACCCCTCTTTTATTTTAAAAAACCAATTCTTTAAAATACAATAAACACATTCAACTAT\tFFKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKFKKKKKKKKKKKFKKKKKKKKKKKKAKKKKFKKKFFKKKKKFKKAAKKKAFAAFKKKF<AKKK\tNM:i:41\tMD:Z:2C2G5G0G1G1G3G7G0G8G0G1G2G0G3G0G1G0G0G5G0G1G5G0G0G0G0G1T3G2G10G1C5G1G12G0G1G1G2G1G3G9\tXM:Z:.....h.....zx...h...h.......hh........hh.h..hh...hh.hhh.....hh.h.....hhhhh.....h..x..........h.......h.h............hh.h.z..h.h...z.........\tXR:Z:GA\tXG:Z:GA\n",
            "SRR5836475.28_28/1\t83\tchr11\t58716819\t0\t140M\t=\t58716800\t-159\tAATAACAAAAATAAAAAAAAACAACTTCCAACTCCTATTTTAACCACAAATCATAATATTACTAATAACATAATTCTTACCTAAACCTTACTAAATCTAAAATTAACAATTCTCCTTTAAAAACTACACAATACTCAAAA\tKKKKKKKKKKKFKKKKKKKKKFAKKKKKKFAKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKK\tNM:i:28\tMD:Z:7G1G2G1G0G1G0G3G7G11G11G0G1G8G11G0T3G0G4G8G2G2G14G0G1G2G4G6G2\tXM:Z:.......x.h..h.hh.hh...z.......x...........h...........hh.h........h...........h....hh....h........x..h..h..............hh.h..x....x......x..\tXR:Z:CT\tXG:Z:GA\n",
            "SRR5836475.28_28/1\t163\tchr11\t58716800\t0\t140M\t=\t58716819\t159\tAAGATCCTCAAACTATAAAAATAACAAAAATAAAAAAAAACAACTTCCAACTCCTATTTTAACCACAAATCATAATATTACTAATAACATAATTCTTACCTAAACCTTACTAAATCTAAAATTAACAATTCTCCTTTAAA\tKKKFKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKFKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKFKKKFFKAKKK7FFFKKKKKKFAA<FKAFKKKFKFF<AAFKKKKKKKFA<\tNM:i:32\tMD:Z:0T2T6G0G2G1G0G0G7G1G2G1G0G1G0G3G7G11G11G0G1G8G11G0T3G0G4G8G2G2G14G0G0\tXM:Z:..H.......xh..x.hhh.......x.h..h.hh.hh...z.......x...........h...........hh.h........h...........h....hh....h........x..h..h..............hh\tXR:Z:GA\tXG:Z:GA\n",
            "SRR5836475.64_64/1\t99\tchr11\t43166397\t0\t140M\t=\t43166396\t141\tATAAAGTTTGTAAGATTAATGGGTTTTTTTTTTTATTGATGGTTGATTAGGTTATTTTTTGATATATATGTAGTTAGAGATTTGAGTTTTGGGGGGTATTGGTTAGTTTATAATGTTGTTTTATTTATAGGGTTGTAGAT\tKKKKKKFKKKKKKKKKKKKKKKKFKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKFKKKKKAKKKKKKKKKKKKKKKKKFKKKKKKKKK,<FFKKFKFFKFKKKKKFFKFFFKKKFKKKAKKFKAF7,7AAFK\tNM:i:33\tMD:Z:10C1C2C0C6C0C1C1C3C0C1C6C0C2C4C0C2C2C5C5C2C6C0A4C1C9C9C3T7C0C1C0C10C4\tXM:Z:..........h....hh......hh.h.h...hh.x......xz..h....hh..h..x.....h.....x..h......h.....h.x.........x.........h...........hh.hh..........x....\tXR:Z:CT\tXG:Z:CT\n",
            "SRR5836475.64_64/1\t147\tchr11\t43166396\t0\t140M\t=\t43166397\t-141\tTATAAAGTTTGTAAGATTAATGGGTTTTTTTTTTTATTGATGGTTGATTAGGTTATTTTTTGATATATATGTAGTTAGAGATTTGAGTTTTGGGGGGTATTGGTTAGTTTATAATGTTGTTTTATTTATAGGGTTGTAGA\tKKAAFFKF<<A7FKKKKKKKFAKKKKFKKKKKKKKKKKKKKKKKFKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKK\tNM:i:33\tMD:Z:11C1C2C0C6C0C1C1C3C0C1C6C0C2C4C0C2C2C5C5C2C6C0A4C1C9C9C3T7C0C1C0C10C3\tXM:Z:...........h....hh......hh.h.h...hh.x......xz..h....hh..h..x.....h.....x..h......h.....h.x.........x.........h...........hh.hh..........x...\tXR:Z:GA\tXG:Z:CT\n",
            "SRR5836475.86_86/1\t83\tchr11\t71167245\t0\t140M\t=\t71167203\t-182\tCTTCTTACCAATTAAAACATCTTCTAAATATATACCCAAAAAAAATATTACTAAATCCTCCAATAATACTATATCCAATTTTCTAAAAAAATACCAAAATAATTTAAAAAATAATTATACAAACTTACAATCCCACCAAC\t<FAAA7KKKKKKFA,FAF<FAFKAKKKKKKKFKFFAFKKKKKKKKKKKKFFKKKKKKKKKKKKKKFKKKKKKKKKFKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKFAKKKKK\tNM:i:36\tMD:Z:10G2G0G0G0G8G0G6G4G0G1G1G0G4G2G0G8G2G6G11G1G0G2C1G3G1C1G4C0C1G1G1G0G2G5G3G13\tXM:Z:..........x..hhhh........xh......h....xh.h.hh....h..xh........x..h......h...........x.hh....x...x...x.......x.h.hh..h.....h...h.............\tXR:Z:CT\tXG:Z:GA\n",
            "SRR5836475.86_86/1\t163\tchr11\t71167203\t0\t140M\t=\t71167245\t182\tTATTATAAATAAAACTACTATAAACATAATTAAACATATATCCTTCTTACCAATTAAAACATCTTCTAAATATATACCCAAAAAAAATATTACTAAATCCTCCAATAATACTATATCCAATTTTCTAAAAAAATACCACA\tKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKAKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKFKFFKKKKFKKKFFFFKFKKFF<FF<FFKA<A7AF,AA77AFFF,A<<<AF<A7,<<AKA7<7,,7\tNM:i:36\tMD:Z:9A2G0G2G4G6G1G0G1G3G1G12G2G0G0G0G8G0G6G4G0G1G1G0G4G2G0G8G2G6G11G1G0G2C1G3G1\tXM:Z:............hh..x....h......h..h.h...h.h............x..hhhh........xh......h....xh.h.hh....h..xh........x..h......h...........x.hh....x.....\tXR:Z:GA\tXG:Z:GA\n",
            "SRR5836475.89_89/1\t99\tchr11\t7397034\t0\t140M\t=\t7397045\t151\tTTGTTGTTATTTGGAATTTTGATTTTAGTTATTTTGATTGGTGTGTGATGGAATTTTAGGGTTGTTTTGATTTGTATTTTTTTGATGATTAAGGATGTTGAATATTTTTTTAGGTGTTTTTTAGATATTTGGTATTTTTT\tKKKKKKKKKKKKKKKKKKKKKKKKKKKKAFKKKKKKKKFKKKKKKKKKKKKFKKKKKKKKF7FFAFKKKKKKFF<KKKKKKKKKKKKK,KKAKKKKKKKKKKKKKKKKKKKKKKKKAAKKKKKKFKKKKKKKKKKKKKKK\tNM:i:30\tMD:Z:0C2C3C1C0C2A1T7C4C0C3C3C7A8C1C17C4C0C0C15C4C7C8C1C2C0C3C6C0C1C0\tXM:Z:x..x...h.hx............h....hh...x...x................h.x.................h....hhx...............x....h.......x........h.x...h...z......hh.x\tXR:Z:CT\tXG:Z:CT\n",
            "SRR5836475.89_89/1\t147\tchr11\t7397045\t0\t140M\t=\t7397034\t-151\tTGGAATTTTGATTTTAGTTATTTTGATTGGTGTGTGATGGAATTTTAGGGTTGTTTTGATTTGTATTTTTTTGATGATTAAGGATGTTGAATATTTTTTTAGGTGTTTTTTAGATATTTGGTATTTTTTAGGTGAGAACC\tF<AAFKA7KF<KKA<7A<,AFKFKKKKFKKAKKKFKKKKKKK<FAKKFFKKKFKKAAFAFAKFF<7KF7KKKKKKKKKKKKKKKKKKAKAKK<KFFKKKKKAKFFKKKKKKKF,KKFKKKKKKFFKKF,<AFAA,F,F,,\tNM:i:27\tMD:Z:2A1T7C4C0C3C3C7A8C1C17C4C0C0C15C4C7C8C1C2C0C3C6C0C1C9T0T0\tXM:Z:............h....hh...x...x................h.x.................h....hhx...............x....h.......x........h.x...h...z......hh.x...........\tXR:Z:GA\tXG:Z:CT\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "! samtools sort SRR5836473_1_bismark_bt2_pe.bam -o sorted_8cell.bam\n",
        "! samtools sort SRR3824222_1_bismark_bt2_pe.bam -o sorted_epiblast.bam\n",
        "! samtools sort SRR5836475_1_bismark_bt2_pe.bam -o sorted_icm.bam"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "aSL4aOfPmGs2",
        "outputId": "aec90a30-953a-4141-f2ef-0959d78423d9"
      },
      "execution_count": 42,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "[bam_sort_core] merging from 5 files...\n",
            "[bam_sort_core] merging from 11 files...\n",
            "[bam_sort_core] merging from 7 files...\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "! samtools index  sorted_8cell.bam\n",
        "! samtools index  sorted_epiblast.bam\n",
        "! samtools index  sorted_icm.bam"
      ],
      "metadata": {
        "id": "WiMeNuC-mSf8"
      },
      "execution_count": 43,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "source": [
        "Сколько ридов закартировалось на регион 11347700-11367700; 40185800-40195800 во всех образцах?"
      ],
      "metadata": {
        "id": "_EG8HLl4mdJu"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "! samtools view sorted_8cell.bam chr11:11347700-11367700 | cut -f1 | sort -u | wc -l\n",
        "! samtools view sorted_8cell.bam chr11:40185800-40195800 | cut -f1 | sort -u | wc -l"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "SLWHwSdHmZ0H",
        "outputId": "afffa0be-94b0-4d10-9a14-245cfa21aacb"
      },
      "execution_count": 44,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "1090\n",
            "464\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "! samtools view sorted_epiblast.bam chr11:11347700-11367700 | cut -f1 | sort -u | wc -l\n",
        "! samtools view sorted_epiblast.bam chr11:40185800-40195800 | cut -f1 | sort -u | wc -l"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "37Cxc5l2mhBo",
        "outputId": "ed3b8507-462b-4846-e04d-3a4326b34c17"
      },
      "execution_count": 45,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "2328\n",
            "1062\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "! samtools view sorted_icm.bam chr11:11347700-11367700 | cut -f1 | sort -u | wc -l\n",
        "! samtools view sorted_icm.bam chr11:40185800-40195800 | cut -f1 | sort -u | wc -l"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "Z_pj8v9hmjma",
        "outputId": "f63260d5-c3e8-423e-87e3-a69e19718e0d"
      },
      "execution_count": 46,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "1456\n",
            "630\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "import pandas as pd\n",
        "df = pd.DataFrame({\n",
        "        'Образцы': ['8cell', 'epiblast', 'ICM'],\n",
        "        'chr11:11347700-11367700': [1090, 2328, 1456],\n",
        "        'chr11:40185800-40195800': [464, 1062, 630]\n",
        "     })\n",
        "df"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 142
        },
        "id": "cn8Vb7VLmnMR",
        "outputId": "0a3e6e6a-bdb4-496b-8f64-e97088702cc9"
      },
      "execution_count": 47,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/html": [
              "\n",
              "  <div id=\"df-02a747f6-2954-4411-a126-bd9253ff0fd6\">\n",
              "    <div class=\"colab-df-container\">\n",
              "      <div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>Образцы</th>\n",
              "      <th>chr11:11347700-11367700</th>\n",
              "      <th>chr11:40185800-40195800</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>8cell</td>\n",
              "      <td>1090</td>\n",
              "      <td>464</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>epiblast</td>\n",
              "      <td>2328</td>\n",
              "      <td>1062</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>ICM</td>\n",
              "      <td>1456</td>\n",
              "      <td>630</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "</div>\n",
              "      <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-02a747f6-2954-4411-a126-bd9253ff0fd6')\"\n",
              "              title=\"Convert this dataframe to an interactive table.\"\n",
              "              style=\"display:none;\">\n",
              "        \n",
              "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
              "       width=\"24px\">\n",
              "    <path d=\"M0 0h24v24H0V0z\" fill=\"none\"/>\n",
              "    <path d=\"M18.56 5.44l.94 2.06.94-2.06 2.06-.94-2.06-.94-.94-2.06-.94 2.06-2.06.94zm-11 1L8.5 8.5l.94-2.06 2.06-.94-2.06-.94L8.5 2.5l-.94 2.06-2.06.94zm10 10l.94 2.06.94-2.06 2.06-.94-2.06-.94-.94-2.06-.94 2.06-2.06.94z\"/><path d=\"M17.41 7.96l-1.37-1.37c-.4-.4-.92-.59-1.43-.59-.52 0-1.04.2-1.43.59L10.3 9.45l-7.72 7.72c-.78.78-.78 2.05 0 2.83L4 21.41c.39.39.9.59 1.41.59.51 0 1.02-.2 1.41-.59l7.78-7.78 2.81-2.81c.8-.78.8-2.07 0-2.86zM5.41 20L4 18.59l7.72-7.72 1.47 1.35L5.41 20z\"/>\n",
              "  </svg>\n",
              "      </button>\n",
              "      \n",
              "  <style>\n",
              "    .colab-df-container {\n",
              "      display:flex;\n",
              "      flex-wrap:wrap;\n",
              "      gap: 12px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert {\n",
              "      background-color: #E8F0FE;\n",
              "      border: none;\n",
              "      border-radius: 50%;\n",
              "      cursor: pointer;\n",
              "      display: none;\n",
              "      fill: #1967D2;\n",
              "      height: 32px;\n",
              "      padding: 0 0 0 0;\n",
              "      width: 32px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert:hover {\n",
              "      background-color: #E2EBFA;\n",
              "      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "      fill: #174EA6;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert {\n",
              "      background-color: #3B4455;\n",
              "      fill: #D2E3FC;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert:hover {\n",
              "      background-color: #434B5C;\n",
              "      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
              "      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
              "      fill: #FFFFFF;\n",
              "    }\n",
              "  </style>\n",
              "\n",
              "      <script>\n",
              "        const buttonEl =\n",
              "          document.querySelector('#df-02a747f6-2954-4411-a126-bd9253ff0fd6 button.colab-df-convert');\n",
              "        buttonEl.style.display =\n",
              "          google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "\n",
              "        async function convertToInteractive(key) {\n",
              "          const element = document.querySelector('#df-02a747f6-2954-4411-a126-bd9253ff0fd6');\n",
              "          const dataTable =\n",
              "            await google.colab.kernel.invokeFunction('convertToInteractive',\n",
              "                                                     [key], {});\n",
              "          if (!dataTable) return;\n",
              "\n",
              "          const docLinkHtml = 'Like what you see? Visit the ' +\n",
              "            '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n",
              "            + ' to learn more about interactive tables.';\n",
              "          element.innerHTML = '';\n",
              "          dataTable['output_type'] = 'display_data';\n",
              "          await google.colab.output.renderOutput(dataTable, element);\n",
              "          const docLink = document.createElement('div');\n",
              "          docLink.innerHTML = docLinkHtml;\n",
              "          element.appendChild(docLink);\n",
              "        }\n",
              "      </script>\n",
              "    </div>\n",
              "  </div>\n",
              "  "
            ],
            "text/plain": [
              "    Образцы  chr11:11347700-11367700  chr11:40185800-40195800\n",
              "0     8cell                     1090                      464\n",
              "1  epiblast                     2328                     1062\n",
              "2       ICM                     1456                      630"
            ]
          },
          "metadata": {},
          "execution_count": 47
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "Дедупликация"
      ],
      "metadata": {
        "id": "mSNi9v_UmqRy"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "! deduplicate_bismark  --bam  --paired  -o s_8_cell  SRR5836473_1_bismark_bt2_pe.bam"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "IvOQZ7oEmspx",
        "outputId": "05b366aa-a094-465d-b175-e591cbf60697"
      },
      "execution_count": 48,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Output filename was given as: s_8_cell\n",
            "Processing paired-end Bismark output file(s) (SAM format):\n",
            "SRR5836473_1_bismark_bt2_pe.bam\n",
            "\n",
            "\n",
            "If there are several alignments to a single position in the genome the first alignment will be chosen. Since the input files are not in any way sorted this is a near-enough random selection of reads.\n",
            "\n",
            "Checking file >>SRR5836473_1_bismark_bt2_pe.bam<< for signs of file truncation...\n",
            "\n",
            "\n",
            "\n",
            "Now testing Bismark result file SRR5836473_1_bismark_bt2_pe.bam for positional sorting (which would be bad...)\t...passed!\n",
            "Output file is: s_8_cell.deduplicated.bam\n",
            "\n",
            "skipping header line:\t@HD\tVN:1.0\tSO:unsorted\n",
            "skipping header line:\t@SQ\tSN:chr11\tLN:122082543\n",
            "skipping header line:\t@PG\tID:Bismark\tVN:v0.23.1dev\tCL:\"bismark -p 2 --gzip mouse_chr -1 data/SRR5836473_1.fastq.gz -2 data/SRR5836473_2.fastq.gz -o cell8/\"\n",
            "\n",
            "Total number of alignments analysed in SRR5836473_1_bismark_bt2_pe.bam:\t2850231\n",
            "Total number duplicated alignments removed:\t521904 (18.31%)\n",
            "Duplicated alignments were found at:\t230432 different position(s)\n",
            "\n",
            "Total count of deduplicated leftover sequences: 2328327 (81.69% of total)\n",
            "\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "! deduplicate_bismark  --bam  --paired  -o s_epiblast  SRR3824222_1_bismark_bt2_pe.bam"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "qnkqjRf_qHLd",
        "outputId": "285825b4-2322-4f11-dd20-ee169904e787"
      },
      "execution_count": 50,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Output filename was given as: s_epiblast\n",
            "Processing paired-end Bismark output file(s) (SAM format):\n",
            "SRR3824222_1_bismark_bt2_pe.bam\n",
            "\n",
            "\n",
            "If there are several alignments to a single position in the genome the first alignment will be chosen. Since the input files are not in any way sorted this is a near-enough random selection of reads.\n",
            "\n",
            "Checking file >>SRR3824222_1_bismark_bt2_pe.bam<< for signs of file truncation...\n",
            "\n",
            "\n",
            "\n",
            "Now testing Bismark result file SRR3824222_1_bismark_bt2_pe.bam for positional sorting (which would be bad...)\t...passed!\n",
            "Output file is: s_epiblast.deduplicated.bam\n",
            "\n",
            "skipping header line:\t@HD\tVN:1.0\tSO:unsorted\n",
            "skipping header line:\t@SQ\tSN:chr11\tLN:122082543\n",
            "skipping header line:\t@PG\tID:Bismark\tVN:v0.23.1dev\tCL:\"bismark -p 2 --gzip mouse_chr -1 data/SRR3824222_1.fastq.gz -2 data/SRR3824222_2.fastq.gz -o epiblast/\"\n",
            "\n",
            "Total number of alignments analysed in SRR3824222_1_bismark_bt2_pe.bam:\t7039016\n",
            "Total number duplicated alignments removed:\t205258 (2.92%)\n",
            "Duplicated alignments were found at:\t164796 different position(s)\n",
            "\n",
            "Total count of deduplicated leftover sequences: 6833758 (97.08% of total)\n",
            "\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "! deduplicate_bismark  --bam  --paired  -o s_icm  SRR5836475_1_bismark_bt2_pe.bam"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "Ap_aJ9iArMuG",
        "outputId": "237384d2-3e41-44f5-e1e5-06636ea3f9eb"
      },
      "execution_count": 52,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Output filename was given as: s_icm\n",
            "Processing paired-end Bismark output file(s) (SAM format):\n",
            "SRR5836475_1_bismark_bt2_pe.bam\n",
            "\n",
            "\n",
            "If there are several alignments to a single position in the genome the first alignment will be chosen. Since the input files are not in any way sorted this is a near-enough random selection of reads.\n",
            "\n",
            "Checking file >>SRR5836475_1_bismark_bt2_pe.bam<< for signs of file truncation...\n",
            "\n",
            "\n",
            "\n",
            "Now testing Bismark result file SRR5836475_1_bismark_bt2_pe.bam for positional sorting (which would be bad...)\t...passed!\n",
            "Output file is: s_icm.deduplicated.bam\n",
            "\n",
            "skipping header line:\t@HD\tVN:1.0\tSO:unsorted\n",
            "skipping header line:\t@SQ\tSN:chr11\tLN:122082543\n",
            "skipping header line:\t@PG\tID:Bismark\tVN:v0.23.1dev\tCL:\"bismark --gzip mouse_chr -1 data/SRR5836475_1.fastq.gz -2 data/SRR5836475_2.fastq.gz -o ICM/\"\n",
            "\n",
            "Total number of alignments analysed in SRR5836475_1_bismark_bt2_pe.bam:\t4159998\n",
            "Total number duplicated alignments removed:\t377882 (9.08%)\n",
            "Duplicated alignments were found at:\t229833 different position(s)\n",
            "\n",
            "Total count of deduplicated leftover sequences: 3782116 (90.92% of total)\n",
            "\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "! ls"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "5DyuJfN8sI3Y",
        "outputId": "63506e51-96c6-403d-f37a-fb6d55e9bc2e"
      },
      "execution_count": 53,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "condacolab_install.log\n",
            "FastQC\n",
            "fastqc_v0.11.9.zip\n",
            "s_8_cell.deduplicated.bam\n",
            "sample_data\n",
            "s_epiblast.deduplicated.bam\n",
            "s_icm.deduplicated.bam\n",
            "sorted_8cell.bam\n",
            "sorted_8cell.bam.bai\n",
            "sorted_epiblast.bam\n",
            "sorted_epiblast.bam.bai\n",
            "sorted_icm.bam\n",
            "sorted_icm.bam.bai\n",
            "SRR3824222_1_bismark_bt2_pe.bam\n",
            "SRR3824222_1_bismark_bt2_pe.deduplication_report.txt\n",
            "SRR3824222_1_bismark_bt2_PE_report.txt\n",
            "SRR3824222_1.fastq.gz\n",
            "SRR3824222_2.fastq.gz\n",
            "SRR5836473_1_bismark_bt2_pe.bam\n",
            "SRR5836473_1_bismark_bt2_pe.deduplication_report.txt\n",
            "SRR5836473_1_bismark_bt2_PE_report.txt\n",
            "SRR5836473_1.fastq.gz\n",
            "SRR5836473_2.fastq.gz\n",
            "SRR5836475_1_bismark_bt2_pe.bam\n",
            "SRR5836475_1_bismark_bt2_pe.deduplication_report.txt\n",
            "SRR5836475_1_bismark_bt2_PE_report.txt\n",
            "SRR5836475_1.fastq.gz\n",
            "SRR5836475_2.fastq.gz\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "df2 = pd.DataFrame({\n",
        "        'Образцы': ['8cell', 'epiblast', 'ICM'],\n",
        "        'Процент прочтений дуплицированно': [81.69, 97.08, 90.92],\n",
        "     })\n",
        "df2"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 142
        },
        "id": "RGgoAw8LsJJt",
        "outputId": "d0286332-bbf5-4c2f-91fd-db09144700e0"
      },
      "execution_count": 54,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/html": [
              "\n",
              "  <div id=\"df-15cdcfb1-9bdb-4068-b548-b3678b902995\">\n",
              "    <div class=\"colab-df-container\">\n",
              "      <div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>Образцы</th>\n",
              "      <th>Процент прочтений дуплицированно</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>8cell</td>\n",
              "      <td>81.69</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>epiblast</td>\n",
              "      <td>97.08</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>ICM</td>\n",
              "      <td>90.92</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "</div>\n",
              "      <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-15cdcfb1-9bdb-4068-b548-b3678b902995')\"\n",
              "              title=\"Convert this dataframe to an interactive table.\"\n",
              "              style=\"display:none;\">\n",
              "        \n",
              "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
              "       width=\"24px\">\n",
              "    <path d=\"M0 0h24v24H0V0z\" fill=\"none\"/>\n",
              "    <path d=\"M18.56 5.44l.94 2.06.94-2.06 2.06-.94-2.06-.94-.94-2.06-.94 2.06-2.06.94zm-11 1L8.5 8.5l.94-2.06 2.06-.94-2.06-.94L8.5 2.5l-.94 2.06-2.06.94zm10 10l.94 2.06.94-2.06 2.06-.94-2.06-.94-.94-2.06-.94 2.06-2.06.94z\"/><path d=\"M17.41 7.96l-1.37-1.37c-.4-.4-.92-.59-1.43-.59-.52 0-1.04.2-1.43.59L10.3 9.45l-7.72 7.72c-.78.78-.78 2.05 0 2.83L4 21.41c.39.39.9.59 1.41.59.51 0 1.02-.2 1.41-.59l7.78-7.78 2.81-2.81c.8-.78.8-2.07 0-2.86zM5.41 20L4 18.59l7.72-7.72 1.47 1.35L5.41 20z\"/>\n",
              "  </svg>\n",
              "      </button>\n",
              "      \n",
              "  <style>\n",
              "    .colab-df-container {\n",
              "      display:flex;\n",
              "      flex-wrap:wrap;\n",
              "      gap: 12px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert {\n",
              "      background-color: #E8F0FE;\n",
              "      border: none;\n",
              "      border-radius: 50%;\n",
              "      cursor: pointer;\n",
              "      display: none;\n",
              "      fill: #1967D2;\n",
              "      height: 32px;\n",
              "      padding: 0 0 0 0;\n",
              "      width: 32px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert:hover {\n",
              "      background-color: #E2EBFA;\n",
              "      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "      fill: #174EA6;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert {\n",
              "      background-color: #3B4455;\n",
              "      fill: #D2E3FC;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert:hover {\n",
              "      background-color: #434B5C;\n",
              "      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
              "      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
              "      fill: #FFFFFF;\n",
              "    }\n",
              "  </style>\n",
              "\n",
              "      <script>\n",
              "        const buttonEl =\n",
              "          document.querySelector('#df-15cdcfb1-9bdb-4068-b548-b3678b902995 button.colab-df-convert');\n",
              "        buttonEl.style.display =\n",
              "          google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "\n",
              "        async function convertToInteractive(key) {\n",
              "          const element = document.querySelector('#df-15cdcfb1-9bdb-4068-b548-b3678b902995');\n",
              "          const dataTable =\n",
              "            await google.colab.kernel.invokeFunction('convertToInteractive',\n",
              "                                                     [key], {});\n",
              "          if (!dataTable) return;\n",
              "\n",
              "          const docLinkHtml = 'Like what you see? Visit the ' +\n",
              "            '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n",
              "            + ' to learn more about interactive tables.';\n",
              "          element.innerHTML = '';\n",
              "          dataTable['output_type'] = 'display_data';\n",
              "          await google.colab.output.renderOutput(dataTable, element);\n",
              "          const docLink = document.createElement('div');\n",
              "          docLink.innerHTML = docLinkHtml;\n",
              "          element.appendChild(docLink);\n",
              "        }\n",
              "      </script>\n",
              "    </div>\n",
              "  </div>\n",
              "  "
            ],
            "text/plain": [
              "    Образцы  Процент прочтений дуплицированно\n",
              "0     8cell                             81.69\n",
              "1  epiblast                             97.08\n",
              "2       ICM                             90.92"
            ]
          },
          "metadata": {},
          "execution_count": 54
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "! gdown --id 1Gh6YYO8zmCiUfZ4ShVcvFp3J_gXjnoOF"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "6zb1bbCkm1_L",
        "outputId": "128438c6-c66d-4345-990a-bc941670cee9"
      },
      "execution_count": 55,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Downloading...\n",
            "From: https://drive.google.com/uc?id=1Gh6YYO8zmCiUfZ4ShVcvFp3J_gXjnoOF\n",
            "To: /content/mouse_chr.zip\n",
            "100% 366M/366M [00:05<00:00, 62.3MB/s]\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "! unzip mouse_chr.zip"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "Jouuy9xYm4Vg",
        "outputId": "c7aac388-8abe-43b7-e86e-b6d4d5fe6f94"
      },
      "execution_count": 56,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Archive:  mouse_chr.zip\n",
            "   creating: mouse_chr/\n",
            "   creating: mouse_chr/Bisulfite_Genome/\n",
            "   creating: mouse_chr/Bisulfite_Genome/CT_conversion/\n",
            "  inflating: mouse_chr/Bisulfite_Genome/CT_conversion/genome_mfa.CT_conversion.fa  \n",
            "  inflating: mouse_chr/Bisulfite_Genome/CT_conversion/BS_CT.4.bt2  \n",
            "  inflating: mouse_chr/Bisulfite_Genome/CT_conversion/BS_CT.1.bt2  \n",
            "  inflating: mouse_chr/Bisulfite_Genome/CT_conversion/BS_CT.3.bt2  \n",
            "  inflating: mouse_chr/Bisulfite_Genome/CT_conversion/BS_CT.2.bt2  \n",
            "  inflating: mouse_chr/Bisulfite_Genome/CT_conversion/BS_CT.rev.2.bt2  \n",
            "  inflating: mouse_chr/Bisulfite_Genome/CT_conversion/BS_CT.rev.1.bt2  \n",
            "   creating: mouse_chr/Bisulfite_Genome/GA_conversion/\n",
            "  inflating: mouse_chr/Bisulfite_Genome/GA_conversion/BS_GA.rev.1.bt2  \n",
            "  inflating: mouse_chr/Bisulfite_Genome/GA_conversion/BS_GA.rev.2.bt2  \n",
            "  inflating: mouse_chr/Bisulfite_Genome/GA_conversion/genome_mfa.GA_conversion.fa  \n",
            "  inflating: mouse_chr/Bisulfite_Genome/GA_conversion/BS_GA.4.bt2  \n",
            "  inflating: mouse_chr/Bisulfite_Genome/GA_conversion/BS_GA.1.bt2  \n",
            "  inflating: mouse_chr/Bisulfite_Genome/GA_conversion/BS_GA.2.bt2  \n",
            "  inflating: mouse_chr/Bisulfite_Genome/GA_conversion/BS_GA.3.bt2  \n",
            "  inflating: mouse_chr/chr11.fa      \n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "Время работы: ±30 мин"
      ],
      "metadata": {
        "id": "F5htgnWVm85A"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "! head s_8_cell.deduplicated.bedGraph"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "qcSdoUrvsdr_",
        "outputId": "f0c2e51e-ae2a-4f05-a488-708a8103879b"
      },
      "execution_count": 57,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "head: cannot open 's_8_cell.deduplicated.bedGraph' for reading: No such file or directory\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "! head s_icm.deduplicated.bedGraph"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "PcXEwogysi8w",
        "outputId": "b69d4198-8f91-4dfc-92a0-0326d2952bfa"
      },
      "execution_count": 58,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "head: cannot open 's_icm.deduplicated.bedGraph' for reading: No such file or directory\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "! head s_epiblast.deduplicated.bedGraph"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "OtcQMo5OsjLf",
        "outputId": "cfd724f0-4b4a-4fd4-868a-93b3a45bd058"
      },
      "execution_count": 59,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "head: cannot open 's_epiblast.deduplicated.bedGraph' for reading: No such file or directory\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "Коллинг метилирования"
      ],
      "metadata": {
        "id": "mOIOiIxJnAmv"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "! bismark_methylation_extractor --gzip --bedGraph --cytosine_report --genome_folder ./mouse_chr s_8_cell.deduplicated.bam"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "WZaslZlWs1oI",
        "outputId": "8eba9f88-8bab-451c-9e72-fb4a6e5527cb"
      },
      "execution_count": 60,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "\n",
            " *** Bismark methylation extractor version v0.23.1 ***\n",
            "\n",
            "Trying to determine the type of mapping from the SAM header line of file s_8_cell.deduplicated.bam\n",
            "Treating file(s) as paired-end data (as extracted from @PG line)\n",
            "\n",
            "Setting option '--no_overlap' since this is (normally) the right thing to do for paired-end data\n",
            "\n",
            "\n",
            "Setting core usage to single-threaded (default). Consider using --multicore <int> to speed up the extraction process.\n",
            "\n",
            "Summarising Bismark methylation extractor parameters:\n",
            "===============================================================\n",
            "Bismark paired-end SAM format specified (default)\n",
            "Number of cores to be used: 1\n",
            "Output will be written to the current directory ('/content')\n",
            "\n",
            "\n",
            "Summarising bedGraph parameters:\n",
            "===============================================================\n",
            "Generating additional output in bedGraph and coverage format\n",
            "bedGraph format:\t<Chromosome> <Start Position> <End Position> <Methylation Percentage>\n",
            "coverage format:\t<Chromosome> <Start Position> <End Position> <Methylation Percentage> <count methylated> <count non-methylated>\n",
            "\n",
            "Using a cutoff of 1 read(s) to report cytosine positions\n",
            "Reporting and sorting cytosine methylation information in CpG context only (default)\n",
            "The bedGraph UNIX sort command will use the following memory setting:\t'2G'. Temporary directory used for sorting is the output directory\n",
            "\n",
            "\n",
            "Summarising genome-wide cytosine methylation report parameters:\n",
            "===============================================================\n",
            "Generating comprehensive genome-wide cytosine report\n",
            "(output format: <Chromosome> <Position> <Strand> <count methylated> <count non-methylated>  <C-context>  <trinucleotide context> )\n",
            "Reporting cytosine methylation in CpG context only (default)\n",
            "Using 1-based genomic coordinates (default)\n",
            "Genome folder was specified as ./mouse_chr/\n",
            "\n",
            "Checking file >>s_8_cell.deduplicated.bam<< for signs of file truncation...\n",
            "\n",
            "\n",
            "Now testing Bismark result file >s_8_cell.deduplicated.bam< for positional sorting (which would be bad...)\t...passed!\n",
            "Writing result file containing methylation information for C in CpG context from the original top strand to CpG_OT_s_8_cell.deduplicated.txt.gz\n",
            "Writing result file containing methylation information for C in CpG context from the complementary to original top strand to CpG_CTOT_s_8_cell.deduplicated.txt.gz\n",
            "Writing result file containing methylation information for C in CpG context from the complementary to original bottom strand to CpG_CTOB_s_8_cell.deduplicated.txt.gz\n",
            "Writing result file containing methylation information for C in CpG context from the original bottom strand to CpG_OB_s_8_cell.deduplicated.txt.gz\n",
            "\n",
            "Writing result file containing methylation information for C in CHG context from the original top strand to CHG_OT_s_8_cell.deduplicated.txt.gz\n",
            "Writing result file containing methylation information for C in CHG context from the complementary to original top strand to CHG_CTOT_s_8_cell.deduplicated.txt.gz\n",
            "Writing result file containing methylation information for C in CHG context from the complementary to original bottom strand to CHG_CTOB_s_8_cell.deduplicated.txt.gz\n",
            "Writing result file containing methylation information for C in CHG context from the original bottom strand to CHG_OB_s_8_cell.deduplicated.txt.gz\n",
            "\n",
            "Writing result file containing methylation information for C in CHH context from the original top strand to CHH_OT_s_8_cell.deduplicated.txt.gz\n",
            "Writing result file containing methylation information for C in CHH context from the complementary to original top strand to CHH_CTOT_s_8_cell.deduplicated.txt.gz\n",
            "Writing result file containing methylation information for C in CHH context from the complementary to original bottom strand to CHH_CTOB_s_8_cell.deduplicated.txt.gz\n",
            "Writing result file containing methylation information for C in CHH context from the original bottom strand to CHH_OB_s_8_cell.deduplicated.txt.gz\n",
            "\n",
            "Now reading in Bismark result file s_8_cell.deduplicated.bam\n",
            "skipping SAM header line:\t@HD\tVN:1.0\tSO:unsorted\n",
            "skipping SAM header line:\t@SQ\tSN:chr11\tLN:122082543\n",
            "skipping SAM header line:\t@PG\tID:Bismark\tVN:v0.23.1dev\tCL:\"bismark -p 2 --gzip mouse_chr -1 data/SRR5836473_1.fastq.gz -2 data/SRR5836473_2.fastq.gz -o cell8/\"\n",
            "Processed lines: 500000\n",
            "Processed lines: 1000000\n",
            "Processed lines: 1500000\n",
            "Processed lines: 2000000\n",
            "\n",
            "Processed 2328327 lines in total\n",
            "Total number of methylation call strings processed: 4656654\n",
            "\n",
            "Final Cytosine Methylation Report\n",
            "=================================\n",
            "Total number of C's analysed:\t104224209\n",
            "\n",
            "Total methylated C's in CpG context:\t2101489\n",
            "Total methylated C's in CHG context:\t269204\n",
            "Total methylated C's in CHH context:\t944695\n",
            "\n",
            "Total C to T conversions in CpG context:\t2731177\n",
            "Total C to T conversions in CHG context:\t22964928\n",
            "Total C to T conversions in CHH context:\t75212716\n",
            "\n",
            "C methylated in CpG context:\t43.5%\n",
            "C methylated in CHG context:\t1.2%\n",
            "C methylated in CHH context:\t1.2%\n",
            "\n",
            "\n",
            "Determining maximum read lengths for M-Bias plots\n",
            "Maximum read length of Read 1: 140\n",
            "Maximum read length of Read 2: 140\n",
            "\n",
            "Perl module GD::Graph::lines is not installed, skipping drawing M-bias plots (only writing out M-bias plot table)\n",
            "Determining maximum read lengths for M-Bias plots\n",
            "Maximum read length of Read 1: 140\n",
            "Maximum read length of Read 2: 140\n",
            "\n",
            "Perl module GD::Graph::lines is not installed, skipping drawing M-bias plots (only writing out M-bias plot table)\n",
            "Deleting unused files ...\n",
            "\n",
            "CpG_OT_s_8_cell.deduplicated.txt.gz contains data ->\tkept\n",
            "CpG_CTOT_s_8_cell.deduplicated.txt.gz was empty ->\tdeleted\n",
            "CpG_CTOB_s_8_cell.deduplicated.txt.gz was empty ->\tdeleted\n",
            "CpG_OB_s_8_cell.deduplicated.txt.gz contains data ->\tkept\n",
            "CHG_OT_s_8_cell.deduplicated.txt.gz contains data ->\tkept\n",
            "CHG_CTOT_s_8_cell.deduplicated.txt.gz was empty ->\tdeleted\n",
            "CHG_CTOB_s_8_cell.deduplicated.txt.gz was empty ->\tdeleted\n",
            "CHG_OB_s_8_cell.deduplicated.txt.gz contains data ->\tkept\n",
            "CHH_OT_s_8_cell.deduplicated.txt.gz contains data ->\tkept\n",
            "CHH_CTOT_s_8_cell.deduplicated.txt.gz was empty ->\tdeleted\n",
            "CHH_CTOB_s_8_cell.deduplicated.txt.gz was empty ->\tdeleted\n",
            "CHH_OB_s_8_cell.deduplicated.txt.gz contains data ->\tkept\n",
            "\n",
            "\n",
            "Using these input files: CpG_OT_s_8_cell.deduplicated.txt.gz CpG_OB_s_8_cell.deduplicated.txt.gz CHG_OT_s_8_cell.deduplicated.txt.gz CHG_OB_s_8_cell.deduplicated.txt.gz CHH_OT_s_8_cell.deduplicated.txt.gz CHH_OB_s_8_cell.deduplicated.txt.gz\n",
            "\n",
            "Summary of parameters for bismark2bedGraph conversion:\n",
            "======================================================\n",
            "bedGraph output:\t\ts_8_cell.deduplicated.bedGraph.gz\n",
            "output directory:\t\t><\n",
            "remove whitespaces:\t\tno\n",
            "CX context:\t\t\tno (CpG context only, default)\n",
            "No-header selected:\t\tno\n",
            "Sorting method:\t\t\tUnix sort-based (smaller memory footprint, but slower)\n",
            "Sort buffer size:\t\t2G\n",
            "Coverage threshold:\t\t1\n",
            "=============================================================================\n",
            "Methylation information will now be written into a bedGraph and coverage file\n",
            "=============================================================================\n",
            "\n",
            "Using the following files as Input:\n",
            "/content/CpG_OT_s_8_cell.deduplicated.txt.gz\t/content/CpG_OB_s_8_cell.deduplicated.txt.gz\n",
            "\n",
            "Writing bedGraph to file: s_8_cell.deduplicated.bedGraph.gz\n",
            "Also writing out a coverage file including counts methylated and unmethylated residues to file: s_8_cell.deduplicated.bismark.cov.gz\n",
            "\n",
            "Now writing methylation information for file >>CpG_OT_s_8_cell.deduplicated.txt.gz<< to individual files for each chromosome\n",
            "Finished writing out individual chromosome files for CpG_OT_s_8_cell.deduplicated.txt.gz\n",
            "Now writing methylation information for file >>CpG_OB_s_8_cell.deduplicated.txt.gz<< to individual files for each chromosome\n",
            "Finished writing out individual chromosome files for CpG_OB_s_8_cell.deduplicated.txt.gz\n",
            "\n",
            "Collecting temporary chromosome file information... Processing the following input file(s):\n",
            "CpG_OT_s_8_cell.deduplicated.txt.gz.chrchr11.methXtractor.temp\n",
            "\n",
            "Sorting input file CpG_OT_s_8_cell.deduplicated.txt.gz.chrchr11.methXtractor.temp by positions (using -S of 2G)\n",
            "tcmalloc: large alloc 2147491840 bytes == 0x5615ca076000 @  0x7fce3a6c41e7 0x5615c93e9718 0x5615c93e85a1 0x7fce3a0a2bf7 0x5615c93e902a\n",
            "Successfully deleted the temporary input file CpG_OT_s_8_cell.deduplicated.txt.gz.chrchr11.methXtractor.temp\n",
            "\n",
            "Finished BedGraph conversion ...\n",
            "\n",
            "Summary of parameters for genome-wide cytosine report:\n",
            "==============================================================================\n",
            "Coverage infile:\t\t\t\ts_8_cell.deduplicated.bismark.cov.gz\n",
            "Output directory:\t\t\t\t><\n",
            "Parent directory:\t\t\t\t>/content/<\n",
            "Genome directory:\t\t\t\t>/content/mouse_chr/<\n",
            "Coverage threshold\t\t\t\t0 (all positions will be reported, default)\n",
            "CX context:\t\t\t\t\tno (CpG context only, default)\n",
            "Genome coordinates used:\t\t\t1-based (default)\n",
            "GZIP compression:\t\t\t\tyes\n",
            "Split by chromosome:\t\t\t\tno\n",
            "\n",
            "\n",
            "Now reading in and storing sequence information of the genome specified in: /content/mouse_chr/\n",
            "\n",
            "chr chr11 (122082543 bp)\n",
            "\n",
            "Stored sequence information of 1 chromosomes/scaffolds in total\n",
            "\n",
            "==============================================================================\n",
            "Methylation information will now be written into a genome-wide cytosine report\n",
            "==============================================================================\n",
            "\n",
            "Adding context-specific methylation summaries\n",
            "\n",
            ">>> Writing genome-wide cytosine report to: s_8_cell.deduplicated.CpG_report.txt.gz <<<\n",
            "\n",
            ">>> Writing all cytosine context summary file to: s_8_cell.deduplicated.cytosine_context_summary.txt <<<\n",
            "\n",
            "Storing all covered cytosine positions for chromosome: chr11\n",
            "Writing cytosine report for last chromosome chr11 (stored 1736778 different covered positions)\n",
            "Finished writing out cytosine report for covered chromosomes (processed 1 chromosomes/scaffolds in total)\n",
            "\n",
            "Now processing chromosomes that were not covered by any methylation calls in the coverage file...\n",
            "All chromosomes in the genome were covered by at least some reads. coverage2cytosine processing complete.\n",
            "\n",
            "\n",
            "\n",
            "Finished generating genome-wide cytosine report\n",
            "\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "! ls -lh"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "TCQypXIKtj5S",
        "outputId": "93fdb845-3f39-4494-a94b-cc9b73d81470"
      },
      "execution_count": 61,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "total 53G\n",
            "-rw-r--r-- 1 root root  47M Feb 16 08:55 CHG_OB_s_8_cell.deduplicated.txt.gz\n",
            "-rw-r--r-- 1 root root  47M Feb 16 08:55 CHG_OT_s_8_cell.deduplicated.txt.gz\n",
            "-rw-r--r-- 1 root root 127M Feb 16 08:55 CHH_OB_s_8_cell.deduplicated.txt.gz\n",
            "-rw-r--r-- 1 root root 126M Feb 16 08:55 CHH_OT_s_8_cell.deduplicated.txt.gz\n",
            "-rw-r--r-- 1 root root  14K Feb 16 07:47 condacolab_install.log\n",
            "-rw-r--r-- 1 root root  15M Feb 16 08:55 CpG_OB_s_8_cell.deduplicated.txt.gz\n",
            "-rw-r--r-- 1 root root  15M Feb 16 08:55 CpG_OT_s_8_cell.deduplicated.txt.gz\n",
            "drwxr-xr-x 8 root root 4.0K Feb 16 07:17 FastQC\n",
            "-rw-r--r-- 1 root root 9.8M Jan 15  2020 fastqc_v0.11.9.zip\n",
            "drwxr-xr-x 3 root root 4.0K Feb  3 06:58 mouse_chr\n",
            "-rw------- 1 root root 349M Feb 16 08:36 mouse_chr.zip\n",
            "-rw-r--r-- 1 root root 528M Feb 16 08:21 s_8_cell.deduplicated.bam\n",
            "-rw-r--r-- 1 root root 9.8M Feb 16 08:56 s_8_cell.deduplicated.bedGraph.gz\n",
            "-rw-r--r-- 1 root root 9.4M Feb 16 08:56 s_8_cell.deduplicated.bismark.cov.gz\n",
            "-rw-r--r-- 1 root root 9.9M Feb 16 08:59 s_8_cell.deduplicated.CpG_report.txt.gz\n",
            "-rw-r--r-- 1 root root 1.6K Feb 16 08:59 s_8_cell.deduplicated.cytosine_context_summary.txt\n",
            "-rw-r--r-- 1 root root  22K Feb 16 08:56 s_8_cell.deduplicated.M-bias.txt\n",
            "-rw-r--r-- 1 root root  840 Feb 16 08:56 s_8_cell.deduplicated_splitting_report.txt\n",
            "drwxr-xr-x 1 root root 4.0K Feb  1 14:32 sample_data\n",
            "-rw-r--r-- 1 root root 1.5G Feb 16 08:30 s_epiblast.deduplicated.bam\n",
            "-rw-r--r-- 1 root root 836M Feb 16 08:35 s_icm.deduplicated.bam\n",
            "-rw-r--r-- 1 root root 524M Feb 16 08:01 sorted_8cell.bam\n",
            "-rw-r--r-- 1 root root 199K Feb 16 08:17 sorted_8cell.bam.bai\n",
            "-rw-r--r-- 1 root root 1.3G Feb 16 08:09 sorted_epiblast.bam\n",
            "-rw-r--r-- 1 root root 349K Feb 16 08:17 sorted_epiblast.bam.bai\n",
            "-rw-r--r-- 1 root root 770M Feb 16 08:14 sorted_icm.bam\n",
            "-rw-r--r-- 1 root root 339K Feb 16 08:18 sorted_icm.bam.bai\n",
            "-rw------- 1 root root 1.5G Feb 16 07:56 SRR3824222_1_bismark_bt2_pe.bam\n",
            "-rw-r--r-- 1 root root  282 Feb 16 08:30 SRR3824222_1_bismark_bt2_pe.deduplication_report.txt\n",
            "-rw------- 1 root root 1.9K Feb 16 07:57 SRR3824222_1_bismark_bt2_PE_report.txt\n",
            "-rw-r--r-- 1 root root 7.6G Feb 16 07:31 SRR3824222_1.fastq.gz\n",
            "-rw-r--r-- 1 root root 9.6G Feb 16 07:38 SRR3824222_2.fastq.gz\n",
            "-rw------- 1 root root 636M Feb 16 07:53 SRR5836473_1_bismark_bt2_pe.bam\n",
            "-rw-r--r-- 1 root root  283 Feb 16 08:21 SRR5836473_1_bismark_bt2_pe.deduplication_report.txt\n",
            "-rw------- 1 root root 1.9K Feb 16 07:57 SRR5836473_1_bismark_bt2_PE_report.txt\n",
            "-rw-r--r-- 1 root root 6.4G Feb 16 07:23 SRR5836473_1.fastq.gz\n",
            "-rw-r--r-- 1 root root 7.6G Feb 16 07:27 SRR5836473_2.fastq.gz\n",
            "-rw------- 1 root root 916M Feb 16 07:56 SRR5836475_1_bismark_bt2_pe.bam\n",
            "-rw-r--r-- 1 root root  282 Feb 16 08:35 SRR5836475_1_bismark_bt2_pe.deduplication_report.txt\n",
            "-rw------- 1 root root 1.9K Feb 16 07:57 SRR5836475_1_bismark_bt2_PE_report.txt\n",
            "-rw-r--r-- 1 root root 5.8G Feb 16 07:43 SRR5836475_1.fastq.gz\n",
            "-rw-r--r-- 1 root root 7.0G Feb 16 07:46 SRR5836475_2.fastq.gz\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "! bismark_methylation_extractor --gzip --bedGraph --cytosine_report --genome_folder ./mouse_chr s_icm.deduplicated.bam"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "ndiK8MhMtGop",
        "outputId": "de59c416-2799-4743-dd91-a7130ff54c26"
      },
      "execution_count": 62,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "\n",
            " *** Bismark methylation extractor version v0.23.1 ***\n",
            "\n",
            "Trying to determine the type of mapping from the SAM header line of file s_icm.deduplicated.bam\n",
            "Treating file(s) as paired-end data (as extracted from @PG line)\n",
            "\n",
            "Setting option '--no_overlap' since this is (normally) the right thing to do for paired-end data\n",
            "\n",
            "\n",
            "Setting core usage to single-threaded (default). Consider using --multicore <int> to speed up the extraction process.\n",
            "\n",
            "Summarising Bismark methylation extractor parameters:\n",
            "===============================================================\n",
            "Bismark paired-end SAM format specified (default)\n",
            "Number of cores to be used: 1\n",
            "Output will be written to the current directory ('/content')\n",
            "\n",
            "\n",
            "Summarising bedGraph parameters:\n",
            "===============================================================\n",
            "Generating additional output in bedGraph and coverage format\n",
            "bedGraph format:\t<Chromosome> <Start Position> <End Position> <Methylation Percentage>\n",
            "coverage format:\t<Chromosome> <Start Position> <End Position> <Methylation Percentage> <count methylated> <count non-methylated>\n",
            "\n",
            "Using a cutoff of 1 read(s) to report cytosine positions\n",
            "Reporting and sorting cytosine methylation information in CpG context only (default)\n",
            "The bedGraph UNIX sort command will use the following memory setting:\t'2G'. Temporary directory used for sorting is the output directory\n",
            "\n",
            "\n",
            "Summarising genome-wide cytosine methylation report parameters:\n",
            "===============================================================\n",
            "Generating comprehensive genome-wide cytosine report\n",
            "(output format: <Chromosome> <Position> <Strand> <count methylated> <count non-methylated>  <C-context>  <trinucleotide context> )\n",
            "Reporting cytosine methylation in CpG context only (default)\n",
            "Using 1-based genomic coordinates (default)\n",
            "Genome folder was specified as ./mouse_chr/\n",
            "\n",
            "Checking file >>s_icm.deduplicated.bam<< for signs of file truncation...\n",
            "\n",
            "\n",
            "Now testing Bismark result file >s_icm.deduplicated.bam< for positional sorting (which would be bad...)\t...passed!\n",
            "Writing result file containing methylation information for C in CpG context from the original top strand to CpG_OT_s_icm.deduplicated.txt.gz\n",
            "Writing result file containing methylation information for C in CpG context from the complementary to original top strand to CpG_CTOT_s_icm.deduplicated.txt.gz\n",
            "Writing result file containing methylation information for C in CpG context from the complementary to original bottom strand to CpG_CTOB_s_icm.deduplicated.txt.gz\n",
            "Writing result file containing methylation information for C in CpG context from the original bottom strand to CpG_OB_s_icm.deduplicated.txt.gz\n",
            "\n",
            "Writing result file containing methylation information for C in CHG context from the original top strand to CHG_OT_s_icm.deduplicated.txt.gz\n",
            "Writing result file containing methylation information for C in CHG context from the complementary to original top strand to CHG_CTOT_s_icm.deduplicated.txt.gz\n",
            "Writing result file containing methylation information for C in CHG context from the complementary to original bottom strand to CHG_CTOB_s_icm.deduplicated.txt.gz\n",
            "Writing result file containing methylation information for C in CHG context from the original bottom strand to CHG_OB_s_icm.deduplicated.txt.gz\n",
            "\n",
            "Writing result file containing methylation information for C in CHH context from the original top strand to CHH_OT_s_icm.deduplicated.txt.gz\n",
            "Writing result file containing methylation information for C in CHH context from the complementary to original top strand to CHH_CTOT_s_icm.deduplicated.txt.gz\n",
            "Writing result file containing methylation information for C in CHH context from the complementary to original bottom strand to CHH_CTOB_s_icm.deduplicated.txt.gz\n",
            "Writing result file containing methylation information for C in CHH context from the original bottom strand to CHH_OB_s_icm.deduplicated.txt.gz\n",
            "\n",
            "Now reading in Bismark result file s_icm.deduplicated.bam\n",
            "skipping SAM header line:\t@HD\tVN:1.0\tSO:unsorted\n",
            "skipping SAM header line:\t@SQ\tSN:chr11\tLN:122082543\n",
            "skipping SAM header line:\t@PG\tID:Bismark\tVN:v0.23.1dev\tCL:\"bismark --gzip mouse_chr -1 data/SRR5836475_1.fastq.gz -2 data/SRR5836475_2.fastq.gz -o ICM/\"\n",
            "Processed lines: 500000\n",
            "Processed lines: 1000000\n",
            "Processed lines: 1500000\n",
            "Processed lines: 2000000\n",
            "Processed lines: 2500000\n",
            "Processed lines: 3000000\n",
            "Processed lines: 3500000\n",
            "\n",
            "Processed 3782116 lines in total\n",
            "Total number of methylation call strings processed: 7564232\n",
            "\n",
            "Final Cytosine Methylation Report\n",
            "=================================\n",
            "Total number of C's analysed:\t177157754\n",
            "\n",
            "Total methylated C's in CpG context:\t1914374\n",
            "Total methylated C's in CHG context:\t297421\n",
            "Total methylated C's in CHH context:\t1066806\n",
            "\n",
            "Total C to T conversions in CpG context:\t6262904\n",
            "Total C to T conversions in CHG context:\t39194423\n",
            "Total C to T conversions in CHH context:\t128421826\n",
            "\n",
            "C methylated in CpG context:\t23.4%\n",
            "C methylated in CHG context:\t0.8%\n",
            "C methylated in CHH context:\t0.8%\n",
            "\n",
            "\n",
            "Determining maximum read lengths for M-Bias plots\n",
            "Maximum read length of Read 1: 140\n",
            "Maximum read length of Read 2: 140\n",
            "\n",
            "Perl module GD::Graph::lines is not installed, skipping drawing M-bias plots (only writing out M-bias plot table)\n",
            "Determining maximum read lengths for M-Bias plots\n",
            "Maximum read length of Read 1: 140\n",
            "Maximum read length of Read 2: 140\n",
            "\n",
            "Perl module GD::Graph::lines is not installed, skipping drawing M-bias plots (only writing out M-bias plot table)\n",
            "Deleting unused files ...\n",
            "\n",
            "CpG_OT_s_icm.deduplicated.txt.gz contains data ->\tkept\n",
            "CpG_CTOT_s_icm.deduplicated.txt.gz was empty ->\tdeleted\n",
            "CpG_CTOB_s_icm.deduplicated.txt.gz was empty ->\tdeleted\n",
            "CpG_OB_s_icm.deduplicated.txt.gz contains data ->\tkept\n",
            "CHG_OT_s_icm.deduplicated.txt.gz contains data ->\tkept\n",
            "CHG_CTOT_s_icm.deduplicated.txt.gz was empty ->\tdeleted\n",
            "CHG_CTOB_s_icm.deduplicated.txt.gz was empty ->\tdeleted\n",
            "CHG_OB_s_icm.deduplicated.txt.gz contains data ->\tkept\n",
            "CHH_OT_s_icm.deduplicated.txt.gz contains data ->\tkept\n",
            "CHH_CTOT_s_icm.deduplicated.txt.gz was empty ->\tdeleted\n",
            "CHH_CTOB_s_icm.deduplicated.txt.gz was empty ->\tdeleted\n",
            "CHH_OB_s_icm.deduplicated.txt.gz contains data ->\tkept\n",
            "\n",
            "\n",
            "Using these input files: CpG_OT_s_icm.deduplicated.txt.gz CpG_OB_s_icm.deduplicated.txt.gz CHG_OT_s_icm.deduplicated.txt.gz CHG_OB_s_icm.deduplicated.txt.gz CHH_OT_s_icm.deduplicated.txt.gz CHH_OB_s_icm.deduplicated.txt.gz\n",
            "\n",
            "Summary of parameters for bismark2bedGraph conversion:\n",
            "======================================================\n",
            "bedGraph output:\t\ts_icm.deduplicated.bedGraph.gz\n",
            "output directory:\t\t><\n",
            "remove whitespaces:\t\tno\n",
            "CX context:\t\t\tno (CpG context only, default)\n",
            "No-header selected:\t\tno\n",
            "Sorting method:\t\t\tUnix sort-based (smaller memory footprint, but slower)\n",
            "Sort buffer size:\t\t2G\n",
            "Coverage threshold:\t\t1\n",
            "=============================================================================\n",
            "Methylation information will now be written into a bedGraph and coverage file\n",
            "=============================================================================\n",
            "\n",
            "Using the following files as Input:\n",
            "/content/CpG_OT_s_icm.deduplicated.txt.gz\t/content/CpG_OB_s_icm.deduplicated.txt.gz\n",
            "\n",
            "Writing bedGraph to file: s_icm.deduplicated.bedGraph.gz\n",
            "Also writing out a coverage file including counts methylated and unmethylated residues to file: s_icm.deduplicated.bismark.cov.gz\n",
            "\n",
            "Now writing methylation information for file >>CpG_OT_s_icm.deduplicated.txt.gz<< to individual files for each chromosome\n",
            "Finished writing out individual chromosome files for CpG_OT_s_icm.deduplicated.txt.gz\n",
            "Now writing methylation information for file >>CpG_OB_s_icm.deduplicated.txt.gz<< to individual files for each chromosome\n",
            "Finished writing out individual chromosome files for CpG_OB_s_icm.deduplicated.txt.gz\n",
            "\n",
            "Collecting temporary chromosome file information... Processing the following input file(s):\n",
            "CpG_OT_s_icm.deduplicated.txt.gz.chrchr11.methXtractor.temp\n",
            "\n",
            "Sorting input file CpG_OT_s_icm.deduplicated.txt.gz.chrchr11.methXtractor.temp by positions (using -S of 2G)\n",
            "tcmalloc: large alloc 2147491840 bytes == 0x55d2eb222000 @  0x7f09cf2251e7 0x55d2e8b97718 0x55d2e8b965a1 0x7f09cec03bf7 0x55d2e8b9702a\n",
            "Successfully deleted the temporary input file CpG_OT_s_icm.deduplicated.txt.gz.chrchr11.methXtractor.temp\n",
            "\n",
            "Finished BedGraph conversion ...\n",
            "\n",
            "Summary of parameters for genome-wide cytosine report:\n",
            "==============================================================================\n",
            "Coverage infile:\t\t\t\ts_icm.deduplicated.bismark.cov.gz\n",
            "Output directory:\t\t\t\t><\n",
            "Parent directory:\t\t\t\t>/content/<\n",
            "Genome directory:\t\t\t\t>/content/mouse_chr/<\n",
            "Coverage threshold\t\t\t\t0 (all positions will be reported, default)\n",
            "CX context:\t\t\t\t\tno (CpG context only, default)\n",
            "Genome coordinates used:\t\t\t1-based (default)\n",
            "GZIP compression:\t\t\t\tyes\n",
            "Split by chromosome:\t\t\t\tno\n",
            "\n",
            "\n",
            "Now reading in and storing sequence information of the genome specified in: /content/mouse_chr/\n",
            "\n",
            "chr chr11 (122082543 bp)\n",
            "\n",
            "Stored sequence information of 1 chromosomes/scaffolds in total\n",
            "\n",
            "==============================================================================\n",
            "Methylation information will now be written into a genome-wide cytosine report\n",
            "==============================================================================\n",
            "\n",
            "Adding context-specific methylation summaries\n",
            "\n",
            ">>> Writing genome-wide cytosine report to: s_icm.deduplicated.CpG_report.txt.gz <<<\n",
            "\n",
            ">>> Writing all cytosine context summary file to: s_icm.deduplicated.cytosine_context_summary.txt <<<\n",
            "\n",
            "Storing all covered cytosine positions for chromosome: chr11\n",
            "Writing cytosine report for last chromosome chr11 (stored 2053827 different covered positions)\n",
            "Finished writing out cytosine report for covered chromosomes (processed 1 chromosomes/scaffolds in total)\n",
            "\n",
            "Now processing chromosomes that were not covered by any methylation calls in the coverage file...\n",
            "All chromosomes in the genome were covered by at least some reads. coverage2cytosine processing complete.\n",
            "\n",
            "\n",
            "\n",
            "Finished generating genome-wide cytosine report\n",
            "\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "! ls -lh"
      ],
      "metadata": {
        "id": "ZyO30DILtkpe",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "ab29f697-3eba-4021-b501-d35f5f3e3dec"
      },
      "execution_count": 63,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "total 54G\n",
            "-rw-r--r-- 1 root root  47M Feb 16 08:55 CHG_OB_s_8_cell.deduplicated.txt.gz\n",
            "-rw-r--r-- 1 root root  77M Feb 16 09:30 CHG_OB_s_icm.deduplicated.txt.gz\n",
            "-rw-r--r-- 1 root root  47M Feb 16 08:55 CHG_OT_s_8_cell.deduplicated.txt.gz\n",
            "-rw-r--r-- 1 root root  77M Feb 16 09:30 CHG_OT_s_icm.deduplicated.txt.gz\n",
            "-rw-r--r-- 1 root root 127M Feb 16 08:55 CHH_OB_s_8_cell.deduplicated.txt.gz\n",
            "-rw-r--r-- 1 root root 211M Feb 16 09:30 CHH_OB_s_icm.deduplicated.txt.gz\n",
            "-rw-r--r-- 1 root root 126M Feb 16 08:55 CHH_OT_s_8_cell.deduplicated.txt.gz\n",
            "-rw-r--r-- 1 root root 212M Feb 16 09:30 CHH_OT_s_icm.deduplicated.txt.gz\n",
            "-rw-r--r-- 1 root root  14K Feb 16 07:47 condacolab_install.log\n",
            "-rw-r--r-- 1 root root  15M Feb 16 08:55 CpG_OB_s_8_cell.deduplicated.txt.gz\n",
            "-rw-r--r-- 1 root root  24M Feb 16 09:30 CpG_OB_s_icm.deduplicated.txt.gz\n",
            "-rw-r--r-- 1 root root  15M Feb 16 08:55 CpG_OT_s_8_cell.deduplicated.txt.gz\n",
            "-rw-r--r-- 1 root root  23M Feb 16 09:30 CpG_OT_s_icm.deduplicated.txt.gz\n",
            "drwxr-xr-x 8 root root 4.0K Feb 16 07:17 FastQC\n",
            "-rw-r--r-- 1 root root 9.8M Jan 15  2020 fastqc_v0.11.9.zip\n",
            "drwxr-xr-x 3 root root 4.0K Feb  3 06:58 mouse_chr\n",
            "-rw------- 1 root root 349M Feb 16 08:36 mouse_chr.zip\n",
            "-rw-r--r-- 1 root root 528M Feb 16 08:21 s_8_cell.deduplicated.bam\n",
            "-rw-r--r-- 1 root root 9.8M Feb 16 08:56 s_8_cell.deduplicated.bedGraph.gz\n",
            "-rw-r--r-- 1 root root 9.4M Feb 16 08:56 s_8_cell.deduplicated.bismark.cov.gz\n",
            "-rw-r--r-- 1 root root 9.9M Feb 16 08:59 s_8_cell.deduplicated.CpG_report.txt.gz\n",
            "-rw-r--r-- 1 root root 1.6K Feb 16 08:59 s_8_cell.deduplicated.cytosine_context_summary.txt\n",
            "-rw-r--r-- 1 root root  22K Feb 16 08:56 s_8_cell.deduplicated.M-bias.txt\n",
            "-rw-r--r-- 1 root root  840 Feb 16 08:56 s_8_cell.deduplicated_splitting_report.txt\n",
            "drwxr-xr-x 1 root root 4.0K Feb  1 14:32 sample_data\n",
            "-rw-r--r-- 1 root root 1.5G Feb 16 08:30 s_epiblast.deduplicated.bam\n",
            "-rw-r--r-- 1 root root 836M Feb 16 08:35 s_icm.deduplicated.bam\n",
            "-rw-r--r-- 1 root root  12M Feb 16 09:31 s_icm.deduplicated.bedGraph.gz\n",
            "-rw-r--r-- 1 root root  12M Feb 16 09:31 s_icm.deduplicated.bismark.cov.gz\n",
            "-rw-r--r-- 1 root root  11M Feb 16 09:35 s_icm.deduplicated.CpG_report.txt.gz\n",
            "-rw-r--r-- 1 root root 1.7K Feb 16 09:35 s_icm.deduplicated.cytosine_context_summary.txt\n",
            "-rw-r--r-- 1 root root  23K Feb 16 09:30 s_icm.deduplicated.M-bias.txt\n",
            "-rw-r--r-- 1 root root  839 Feb 16 09:30 s_icm.deduplicated_splitting_report.txt\n",
            "-rw-r--r-- 1 root root 524M Feb 16 08:01 sorted_8cell.bam\n",
            "-rw-r--r-- 1 root root 199K Feb 16 08:17 sorted_8cell.bam.bai\n",
            "-rw-r--r-- 1 root root 1.3G Feb 16 08:09 sorted_epiblast.bam\n",
            "-rw-r--r-- 1 root root 349K Feb 16 08:17 sorted_epiblast.bam.bai\n",
            "-rw-r--r-- 1 root root 770M Feb 16 08:14 sorted_icm.bam\n",
            "-rw-r--r-- 1 root root 339K Feb 16 08:18 sorted_icm.bam.bai\n",
            "-rw------- 1 root root 1.5G Feb 16 07:56 SRR3824222_1_bismark_bt2_pe.bam\n",
            "-rw-r--r-- 1 root root  282 Feb 16 08:30 SRR3824222_1_bismark_bt2_pe.deduplication_report.txt\n",
            "-rw------- 1 root root 1.9K Feb 16 07:57 SRR3824222_1_bismark_bt2_PE_report.txt\n",
            "-rw-r--r-- 1 root root 7.6G Feb 16 07:31 SRR3824222_1.fastq.gz\n",
            "-rw-r--r-- 1 root root 9.6G Feb 16 07:38 SRR3824222_2.fastq.gz\n",
            "-rw------- 1 root root 636M Feb 16 07:53 SRR5836473_1_bismark_bt2_pe.bam\n",
            "-rw-r--r-- 1 root root  283 Feb 16 08:21 SRR5836473_1_bismark_bt2_pe.deduplication_report.txt\n",
            "-rw------- 1 root root 1.9K Feb 16 07:57 SRR5836473_1_bismark_bt2_PE_report.txt\n",
            "-rw-r--r-- 1 root root 6.4G Feb 16 07:23 SRR5836473_1.fastq.gz\n",
            "-rw-r--r-- 1 root root 7.6G Feb 16 07:27 SRR5836473_2.fastq.gz\n",
            "-rw------- 1 root root 916M Feb 16 07:56 SRR5836475_1_bismark_bt2_pe.bam\n",
            "-rw-r--r-- 1 root root  282 Feb 16 08:35 SRR5836475_1_bismark_bt2_pe.deduplication_report.txt\n",
            "-rw------- 1 root root 1.9K Feb 16 07:57 SRR5836475_1_bismark_bt2_PE_report.txt\n",
            "-rw-r--r-- 1 root root 5.8G Feb 16 07:43 SRR5836475_1.fastq.gz\n",
            "-rw-r--r-- 1 root root 7.0G Feb 16 07:46 SRR5836475_2.fastq.gz\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "! bismark_methylation_extractor --gzip --bedGraph --cytosine_report --genome_folder ./mouse_chr s_epiblast.deduplicated.bam"
      ],
      "metadata": {
        "id": "ircKNlP_tRbL",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "4451cb4f-a01e-4ca2-e36c-605c733a1b07"
      },
      "execution_count": 64,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "\n",
            " *** Bismark methylation extractor version v0.23.1 ***\n",
            "\n",
            "Trying to determine the type of mapping from the SAM header line of file s_epiblast.deduplicated.bam\n",
            "Treating file(s) as paired-end data (as extracted from @PG line)\n",
            "\n",
            "Setting option '--no_overlap' since this is (normally) the right thing to do for paired-end data\n",
            "\n",
            "\n",
            "Setting core usage to single-threaded (default). Consider using --multicore <int> to speed up the extraction process.\n",
            "\n",
            "Summarising Bismark methylation extractor parameters:\n",
            "===============================================================\n",
            "Bismark paired-end SAM format specified (default)\n",
            "Number of cores to be used: 1\n",
            "Output will be written to the current directory ('/content')\n",
            "\n",
            "\n",
            "Summarising bedGraph parameters:\n",
            "===============================================================\n",
            "Generating additional output in bedGraph and coverage format\n",
            "bedGraph format:\t<Chromosome> <Start Position> <End Position> <Methylation Percentage>\n",
            "coverage format:\t<Chromosome> <Start Position> <End Position> <Methylation Percentage> <count methylated> <count non-methylated>\n",
            "\n",
            "Using a cutoff of 1 read(s) to report cytosine positions\n",
            "Reporting and sorting cytosine methylation information in CpG context only (default)\n",
            "The bedGraph UNIX sort command will use the following memory setting:\t'2G'. Temporary directory used for sorting is the output directory\n",
            "\n",
            "\n",
            "Summarising genome-wide cytosine methylation report parameters:\n",
            "===============================================================\n",
            "Generating comprehensive genome-wide cytosine report\n",
            "(output format: <Chromosome> <Position> <Strand> <count methylated> <count non-methylated>  <C-context>  <trinucleotide context> )\n",
            "Reporting cytosine methylation in CpG context only (default)\n",
            "Using 1-based genomic coordinates (default)\n",
            "Genome folder was specified as ./mouse_chr/\n",
            "\n",
            "Checking file >>s_epiblast.deduplicated.bam<< for signs of file truncation...\n",
            "\n",
            "\n",
            "Now testing Bismark result file >s_epiblast.deduplicated.bam< for positional sorting (which would be bad...)\t...passed!\n",
            "Writing result file containing methylation information for C in CpG context from the original top strand to CpG_OT_s_epiblast.deduplicated.txt.gz\n",
            "Writing result file containing methylation information for C in CpG context from the complementary to original top strand to CpG_CTOT_s_epiblast.deduplicated.txt.gz\n",
            "Writing result file containing methylation information for C in CpG context from the complementary to original bottom strand to CpG_CTOB_s_epiblast.deduplicated.txt.gz\n",
            "Writing result file containing methylation information for C in CpG context from the original bottom strand to CpG_OB_s_epiblast.deduplicated.txt.gz\n",
            "\n",
            "Writing result file containing methylation information for C in CHG context from the original top strand to CHG_OT_s_epiblast.deduplicated.txt.gz\n",
            "Writing result file containing methylation information for C in CHG context from the complementary to original top strand to CHG_CTOT_s_epiblast.deduplicated.txt.gz\n",
            "Writing result file containing methylation information for C in CHG context from the complementary to original bottom strand to CHG_CTOB_s_epiblast.deduplicated.txt.gz\n",
            "Writing result file containing methylation information for C in CHG context from the original bottom strand to CHG_OB_s_epiblast.deduplicated.txt.gz\n",
            "\n",
            "Writing result file containing methylation information for C in CHH context from the original top strand to CHH_OT_s_epiblast.deduplicated.txt.gz\n",
            "Writing result file containing methylation information for C in CHH context from the complementary to original top strand to CHH_CTOT_s_epiblast.deduplicated.txt.gz\n",
            "Writing result file containing methylation information for C in CHH context from the complementary to original bottom strand to CHH_CTOB_s_epiblast.deduplicated.txt.gz\n",
            "Writing result file containing methylation information for C in CHH context from the original bottom strand to CHH_OB_s_epiblast.deduplicated.txt.gz\n",
            "\n",
            "Now reading in Bismark result file s_epiblast.deduplicated.bam\n",
            "skipping SAM header line:\t@HD\tVN:1.0\tSO:unsorted\n",
            "skipping SAM header line:\t@SQ\tSN:chr11\tLN:122082543\n",
            "skipping SAM header line:\t@PG\tID:Bismark\tVN:v0.23.1dev\tCL:\"bismark -p 2 --gzip mouse_chr -1 data/SRR3824222_1.fastq.gz -2 data/SRR3824222_2.fastq.gz -o epiblast/\"\n",
            "Processed lines: 500000\n",
            "Processed lines: 1000000\n",
            "Processed lines: 1500000\n",
            "Processed lines: 2000000\n",
            "Processed lines: 2500000\n",
            "Processed lines: 3000000\n",
            "Processed lines: 3500000\n",
            "Processed lines: 4000000\n",
            "Processed lines: 4500000\n",
            "Processed lines: 5000000\n",
            "Processed lines: 5500000\n",
            "Processed lines: 6000000\n",
            "Processed lines: 6500000\n",
            "\n",
            "Processed 6833758 lines in total\n",
            "Total number of methylation call strings processed: 13667516\n",
            "\n",
            "Final Cytosine Methylation Report\n",
            "=================================\n",
            "Total number of C's analysed:\t330808061\n",
            "\n",
            "Total methylated C's in CpG context:\t11544557\n",
            "Total methylated C's in CHG context:\t1572954\n",
            "Total methylated C's in CHH context:\t3695988\n",
            "\n",
            "Total C to T conversions in CpG context:\t3465254\n",
            "Total C to T conversions in CHG context:\t71995231\n",
            "Total C to T conversions in CHH context:\t238534077\n",
            "\n",
            "C methylated in CpG context:\t76.9%\n",
            "C methylated in CHG context:\t2.1%\n",
            "C methylated in CHH context:\t1.5%\n",
            "\n",
            "\n",
            "Determining maximum read lengths for M-Bias plots\n",
            "Maximum read length of Read 1: 140\n",
            "Maximum read length of Read 2: 140\n",
            "\n",
            "Perl module GD::Graph::lines is not installed, skipping drawing M-bias plots (only writing out M-bias plot table)\n",
            "Determining maximum read lengths for M-Bias plots\n",
            "Maximum read length of Read 1: 140\n",
            "Maximum read length of Read 2: 140\n",
            "\n",
            "Perl module GD::Graph::lines is not installed, skipping drawing M-bias plots (only writing out M-bias plot table)\n",
            "Deleting unused files ...\n",
            "\n",
            "CpG_OT_s_epiblast.deduplicated.txt.gz contains data ->\tkept\n",
            "CpG_CTOT_s_epiblast.deduplicated.txt.gz was empty ->\tdeleted\n",
            "CpG_CTOB_s_epiblast.deduplicated.txt.gz was empty ->\tdeleted\n",
            "CpG_OB_s_epiblast.deduplicated.txt.gz contains data ->\tkept\n",
            "CHG_OT_s_epiblast.deduplicated.txt.gz contains data ->\tkept\n",
            "CHG_CTOT_s_epiblast.deduplicated.txt.gz was empty ->\tdeleted\n",
            "CHG_CTOB_s_epiblast.deduplicated.txt.gz was empty ->\tdeleted\n",
            "CHG_OB_s_epiblast.deduplicated.txt.gz contains data ->\tkept\n",
            "CHH_OT_s_epiblast.deduplicated.txt.gz contains data ->\tkept\n",
            "CHH_CTOT_s_epiblast.deduplicated.txt.gz was empty ->\tdeleted\n",
            "CHH_CTOB_s_epiblast.deduplicated.txt.gz was empty ->\tdeleted\n",
            "CHH_OB_s_epiblast.deduplicated.txt.gz contains data ->\tkept\n",
            "\n",
            "\n",
            "Using these input files: CpG_OT_s_epiblast.deduplicated.txt.gz CpG_OB_s_epiblast.deduplicated.txt.gz CHG_OT_s_epiblast.deduplicated.txt.gz CHG_OB_s_epiblast.deduplicated.txt.gz CHH_OT_s_epiblast.deduplicated.txt.gz CHH_OB_s_epiblast.deduplicated.txt.gz\n",
            "\n",
            "Summary of parameters for bismark2bedGraph conversion:\n",
            "======================================================\n",
            "bedGraph output:\t\ts_epiblast.deduplicated.bedGraph.gz\n",
            "output directory:\t\t><\n",
            "remove whitespaces:\t\tno\n",
            "CX context:\t\t\tno (CpG context only, default)\n",
            "No-header selected:\t\tno\n",
            "Sorting method:\t\t\tUnix sort-based (smaller memory footprint, but slower)\n",
            "Sort buffer size:\t\t2G\n",
            "Coverage threshold:\t\t1\n",
            "=============================================================================\n",
            "Methylation information will now be written into a bedGraph and coverage file\n",
            "=============================================================================\n",
            "\n",
            "Using the following files as Input:\n",
            "/content/CpG_OT_s_epiblast.deduplicated.txt.gz\t/content/CpG_OB_s_epiblast.deduplicated.txt.gz\n",
            "\n",
            "Writing bedGraph to file: s_epiblast.deduplicated.bedGraph.gz\n",
            "Also writing out a coverage file including counts methylated and unmethylated residues to file: s_epiblast.deduplicated.bismark.cov.gz\n",
            "\n",
            "Now writing methylation information for file >>CpG_OT_s_epiblast.deduplicated.txt.gz<< to individual files for each chromosome\n",
            "Finished writing out individual chromosome files for CpG_OT_s_epiblast.deduplicated.txt.gz\n",
            "Now writing methylation information for file >>CpG_OB_s_epiblast.deduplicated.txt.gz<< to individual files for each chromosome\n",
            "Finished writing out individual chromosome files for CpG_OB_s_epiblast.deduplicated.txt.gz\n",
            "\n",
            "Collecting temporary chromosome file information... Processing the following input file(s):\n",
            "CpG_OT_s_epiblast.deduplicated.txt.gz.chrchr11.methXtractor.temp\n",
            "\n",
            "Sorting input file CpG_OT_s_epiblast.deduplicated.txt.gz.chrchr11.methXtractor.temp by positions (using -S of 2G)\n",
            "tcmalloc: large alloc 2147491840 bytes == 0x5601827c4000 @  0x7f0dd5ba21e7 0x56018010c718 0x56018010b5a1 0x7f0dd5580bf7 0x56018010c02a\n",
            "Successfully deleted the temporary input file CpG_OT_s_epiblast.deduplicated.txt.gz.chrchr11.methXtractor.temp\n",
            "\n",
            "Finished BedGraph conversion ...\n",
            "\n",
            "Summary of parameters for genome-wide cytosine report:\n",
            "==============================================================================\n",
            "Coverage infile:\t\t\t\ts_epiblast.deduplicated.bismark.cov.gz\n",
            "Output directory:\t\t\t\t><\n",
            "Parent directory:\t\t\t\t>/content/<\n",
            "Genome directory:\t\t\t\t>/content/mouse_chr/<\n",
            "Coverage threshold\t\t\t\t0 (all positions will be reported, default)\n",
            "CX context:\t\t\t\t\tno (CpG context only, default)\n",
            "Genome coordinates used:\t\t\t1-based (default)\n",
            "GZIP compression:\t\t\t\tyes\n",
            "Split by chromosome:\t\t\t\tno\n",
            "\n",
            "\n",
            "Now reading in and storing sequence information of the genome specified in: /content/mouse_chr/\n",
            "\n",
            "chr chr11 (122082543 bp)\n",
            "\n",
            "Stored sequence information of 1 chromosomes/scaffolds in total\n",
            "\n",
            "==============================================================================\n",
            "Methylation information will now be written into a genome-wide cytosine report\n",
            "==============================================================================\n",
            "\n",
            "Adding context-specific methylation summaries\n",
            "\n",
            ">>> Writing genome-wide cytosine report to: s_epiblast.deduplicated.CpG_report.txt.gz <<<\n",
            "\n",
            ">>> Writing all cytosine context summary file to: s_epiblast.deduplicated.cytosine_context_summary.txt <<<\n",
            "\n",
            "Storing all covered cytosine positions for chromosome: chr11\n",
            "Writing cytosine report for last chromosome chr11 (stored 2249610 different covered positions)\n",
            "Finished writing out cytosine report for covered chromosomes (processed 1 chromosomes/scaffolds in total)\n",
            "\n",
            "Now processing chromosomes that were not covered by any methylation calls in the coverage file...\n",
            "All chromosomes in the genome were covered by at least some reads. coverage2cytosine processing complete.\n",
            "\n",
            "\n",
            "\n",
            "Finished generating genome-wide cytosine report\n",
            "\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "! ls -lh"
      ],
      "metadata": {
        "id": "ZOoa1deNtlZ3",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "99d96a1d-cf82-4193-bc87-7f6fac6b6db7"
      },
      "execution_count": 65,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "total 55G\n",
            "-rw-r--r-- 1 root root  47M Feb 16 08:55 CHG_OB_s_8_cell.deduplicated.txt.gz\n",
            "-rw-r--r-- 1 root root 142M Feb 16 10:32 CHG_OB_s_epiblast.deduplicated.txt.gz\n",
            "-rw-r--r-- 1 root root  77M Feb 16 09:30 CHG_OB_s_icm.deduplicated.txt.gz\n",
            "-rw-r--r-- 1 root root  47M Feb 16 08:55 CHG_OT_s_8_cell.deduplicated.txt.gz\n",
            "-rw-r--r-- 1 root root 140M Feb 16 10:32 CHG_OT_s_epiblast.deduplicated.txt.gz\n",
            "-rw-r--r-- 1 root root  77M Feb 16 09:30 CHG_OT_s_icm.deduplicated.txt.gz\n",
            "-rw-r--r-- 1 root root 127M Feb 16 08:55 CHH_OB_s_8_cell.deduplicated.txt.gz\n",
            "-rw-r--r-- 1 root root 395M Feb 16 10:32 CHH_OB_s_epiblast.deduplicated.txt.gz\n",
            "-rw-r--r-- 1 root root 211M Feb 16 09:30 CHH_OB_s_icm.deduplicated.txt.gz\n",
            "-rw-r--r-- 1 root root 126M Feb 16 08:55 CHH_OT_s_8_cell.deduplicated.txt.gz\n",
            "-rw-r--r-- 1 root root 395M Feb 16 10:32 CHH_OT_s_epiblast.deduplicated.txt.gz\n",
            "-rw-r--r-- 1 root root 212M Feb 16 09:30 CHH_OT_s_icm.deduplicated.txt.gz\n",
            "-rw-r--r-- 1 root root  14K Feb 16 07:47 condacolab_install.log\n",
            "-rw-r--r-- 1 root root  15M Feb 16 08:55 CpG_OB_s_8_cell.deduplicated.txt.gz\n",
            "-rw-r--r-- 1 root root  41M Feb 16 10:32 CpG_OB_s_epiblast.deduplicated.txt.gz\n",
            "-rw-r--r-- 1 root root  24M Feb 16 09:30 CpG_OB_s_icm.deduplicated.txt.gz\n",
            "-rw-r--r-- 1 root root  15M Feb 16 08:55 CpG_OT_s_8_cell.deduplicated.txt.gz\n",
            "-rw-r--r-- 1 root root  41M Feb 16 10:32 CpG_OT_s_epiblast.deduplicated.txt.gz\n",
            "-rw-r--r-- 1 root root  23M Feb 16 09:30 CpG_OT_s_icm.deduplicated.txt.gz\n",
            "drwxr-xr-x 8 root root 4.0K Feb 16 07:17 FastQC\n",
            "-rw-r--r-- 1 root root 9.8M Jan 15  2020 fastqc_v0.11.9.zip\n",
            "drwxr-xr-x 3 root root 4.0K Feb  3 06:58 mouse_chr\n",
            "-rw------- 1 root root 349M Feb 16 08:36 mouse_chr.zip\n",
            "-rw-r--r-- 1 root root 528M Feb 16 08:21 s_8_cell.deduplicated.bam\n",
            "-rw-r--r-- 1 root root 9.8M Feb 16 08:56 s_8_cell.deduplicated.bedGraph.gz\n",
            "-rw-r--r-- 1 root root 9.4M Feb 16 08:56 s_8_cell.deduplicated.bismark.cov.gz\n",
            "-rw-r--r-- 1 root root 9.9M Feb 16 08:59 s_8_cell.deduplicated.CpG_report.txt.gz\n",
            "-rw-r--r-- 1 root root 1.6K Feb 16 08:59 s_8_cell.deduplicated.cytosine_context_summary.txt\n",
            "-rw-r--r-- 1 root root  22K Feb 16 08:56 s_8_cell.deduplicated.M-bias.txt\n",
            "-rw-r--r-- 1 root root  840 Feb 16 08:56 s_8_cell.deduplicated_splitting_report.txt\n",
            "drwxr-xr-x 1 root root 4.0K Feb  1 14:32 sample_data\n",
            "-rw-r--r-- 1 root root 1.5G Feb 16 08:30 s_epiblast.deduplicated.bam\n",
            "-rw-r--r-- 1 root root  14M Feb 16 10:33 s_epiblast.deduplicated.bedGraph.gz\n",
            "-rw-r--r-- 1 root root  14M Feb 16 10:33 s_epiblast.deduplicated.bismark.cov.gz\n",
            "-rw-r--r-- 1 root root  11M Feb 16 10:37 s_epiblast.deduplicated.CpG_report.txt.gz\n",
            "-rw-r--r-- 1 root root 1.7K Feb 16 10:37 s_epiblast.deduplicated.cytosine_context_summary.txt\n",
            "-rw-r--r-- 1 root root  24K Feb 16 10:32 s_epiblast.deduplicated.M-bias.txt\n",
            "-rw-r--r-- 1 root root  847 Feb 16 10:32 s_epiblast.deduplicated_splitting_report.txt\n",
            "-rw-r--r-- 1 root root 836M Feb 16 08:35 s_icm.deduplicated.bam\n",
            "-rw-r--r-- 1 root root  12M Feb 16 09:31 s_icm.deduplicated.bedGraph.gz\n",
            "-rw-r--r-- 1 root root  12M Feb 16 09:31 s_icm.deduplicated.bismark.cov.gz\n",
            "-rw-r--r-- 1 root root  11M Feb 16 09:35 s_icm.deduplicated.CpG_report.txt.gz\n",
            "-rw-r--r-- 1 root root 1.7K Feb 16 09:35 s_icm.deduplicated.cytosine_context_summary.txt\n",
            "-rw-r--r-- 1 root root  23K Feb 16 09:30 s_icm.deduplicated.M-bias.txt\n",
            "-rw-r--r-- 1 root root  839 Feb 16 09:30 s_icm.deduplicated_splitting_report.txt\n",
            "-rw-r--r-- 1 root root 524M Feb 16 08:01 sorted_8cell.bam\n",
            "-rw-r--r-- 1 root root 199K Feb 16 08:17 sorted_8cell.bam.bai\n",
            "-rw-r--r-- 1 root root 1.3G Feb 16 08:09 sorted_epiblast.bam\n",
            "-rw-r--r-- 1 root root 349K Feb 16 08:17 sorted_epiblast.bam.bai\n",
            "-rw-r--r-- 1 root root 770M Feb 16 08:14 sorted_icm.bam\n",
            "-rw-r--r-- 1 root root 339K Feb 16 08:18 sorted_icm.bam.bai\n",
            "-rw------- 1 root root 1.5G Feb 16 07:56 SRR3824222_1_bismark_bt2_pe.bam\n",
            "-rw-r--r-- 1 root root  282 Feb 16 08:30 SRR3824222_1_bismark_bt2_pe.deduplication_report.txt\n",
            "-rw------- 1 root root 1.9K Feb 16 07:57 SRR3824222_1_bismark_bt2_PE_report.txt\n",
            "-rw-r--r-- 1 root root 7.6G Feb 16 07:31 SRR3824222_1.fastq.gz\n",
            "-rw-r--r-- 1 root root 9.6G Feb 16 07:38 SRR3824222_2.fastq.gz\n",
            "-rw------- 1 root root 636M Feb 16 07:53 SRR5836473_1_bismark_bt2_pe.bam\n",
            "-rw-r--r-- 1 root root  283 Feb 16 08:21 SRR5836473_1_bismark_bt2_pe.deduplication_report.txt\n",
            "-rw------- 1 root root 1.9K Feb 16 07:57 SRR5836473_1_bismark_bt2_PE_report.txt\n",
            "-rw-r--r-- 1 root root 6.4G Feb 16 07:23 SRR5836473_1.fastq.gz\n",
            "-rw-r--r-- 1 root root 7.6G Feb 16 07:27 SRR5836473_2.fastq.gz\n",
            "-rw------- 1 root root 916M Feb 16 07:56 SRR5836475_1_bismark_bt2_pe.bam\n",
            "-rw-r--r-- 1 root root  282 Feb 16 08:35 SRR5836475_1_bismark_bt2_pe.deduplication_report.txt\n",
            "-rw------- 1 root root 1.9K Feb 16 07:57 SRR5836475_1_bismark_bt2_PE_report.txt\n",
            "-rw-r--r-- 1 root root 5.8G Feb 16 07:43 SRR5836475_1.fastq.gz\n",
            "-rw-r--r-- 1 root root 7.0G Feb 16 07:46 SRR5836475_2.fastq.gz\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "! bismark2report -mbias_report s_8_cell.deduplicated.M-bias.txt"
      ],
      "metadata": {
        "id": "whtfKA-attbo",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "77bab476-28cd-4c9b-b4af-97ca82d01469"
      },
      "execution_count": 66,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Found 3 alignment reports in current directory. Now trying to figure out whether there are corresponding optional reports\n",
            "\n",
            "Writing Bismark HTML report to >> SRR3824222_1_bismark_bt2_PE_report.html <<\n",
            "\n",
            "==============================================================================================================\n",
            "Using the following alignment report:\t\t> SRR3824222_1_bismark_bt2_PE_report.txt <\n",
            "Processing alignment report SRR3824222_1_bismark_bt2_PE_report.txt ...\n",
            "Complete\n",
            "\n",
            "Using the following deduplication report:\t> SRR3824222_1_bismark_bt2_pe.deduplication_report.txt <\n",
            "Processing deduplication report SRR3824222_1_bismark_bt2_pe.deduplication_report.txt ...\n",
            "Complete\n",
            "\n",
            "No splitting report file specified, skipping this step\n",
            "Using the following M-bias report:\t\t> s_8_cell.deduplicated.M-bias.txt <\n",
            "Processing M-bias report s_8_cell.deduplicated.M-bias.txt ...\n",
            "Complete\n",
            "\n",
            "No nucleotide coverage report file specified, skipping this step\n",
            "==============================================================================================================\n",
            "\n",
            "\n",
            "\n",
            "Writing Bismark HTML report to >> SRR5836473_1_bismark_bt2_PE_report.html <<\n",
            "\n",
            "==============================================================================================================\n",
            "Using the following alignment report:\t\t> SRR5836473_1_bismark_bt2_PE_report.txt <\n",
            "Processing alignment report SRR5836473_1_bismark_bt2_PE_report.txt ...\n",
            "Complete\n",
            "\n",
            "Using the following deduplication report:\t> SRR5836473_1_bismark_bt2_pe.deduplication_report.txt <\n",
            "Processing deduplication report SRR5836473_1_bismark_bt2_pe.deduplication_report.txt ...\n",
            "Complete\n",
            "\n",
            "No splitting report file specified, skipping this step\n",
            "No M-bias report file specified, skipping this step\n",
            "No nucleotide coverage report file specified, skipping this step\n",
            "==============================================================================================================\n",
            "\n",
            "\n",
            "\n",
            "Writing Bismark HTML report to >> SRR5836475_1_bismark_bt2_PE_report.html <<\n",
            "\n",
            "==============================================================================================================\n",
            "Using the following alignment report:\t\t> SRR5836475_1_bismark_bt2_PE_report.txt <\n",
            "Processing alignment report SRR5836475_1_bismark_bt2_PE_report.txt ...\n",
            "Complete\n",
            "\n",
            "Using the following deduplication report:\t> SRR5836475_1_bismark_bt2_pe.deduplication_report.txt <\n",
            "Processing deduplication report SRR5836475_1_bismark_bt2_pe.deduplication_report.txt ...\n",
            "Complete\n",
            "\n",
            "No splitting report file specified, skipping this step\n",
            "No M-bias report file specified, skipping this step\n",
            "No nucleotide coverage report file specified, skipping this step\n",
            "==============================================================================================================\n",
            "\n",
            "\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "! bismark2report -mbias_report s_icm.deduplicated.M-bias.txt"
      ],
      "metadata": {
        "id": "AlTHcy5etuVr",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "0fd52f63-541b-42a9-ce90-f5375f8a89c2"
      },
      "execution_count": 67,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Found 3 alignment reports in current directory. Now trying to figure out whether there are corresponding optional reports\n",
            "\n",
            "Writing Bismark HTML report to >> SRR3824222_1_bismark_bt2_PE_report.html <<\n",
            "\n",
            "==============================================================================================================\n",
            "Using the following alignment report:\t\t> SRR3824222_1_bismark_bt2_PE_report.txt <\n",
            "Processing alignment report SRR3824222_1_bismark_bt2_PE_report.txt ...\n",
            "Complete\n",
            "\n",
            "Using the following deduplication report:\t> SRR3824222_1_bismark_bt2_pe.deduplication_report.txt <\n",
            "Processing deduplication report SRR3824222_1_bismark_bt2_pe.deduplication_report.txt ...\n",
            "Complete\n",
            "\n",
            "No splitting report file specified, skipping this step\n",
            "Using the following M-bias report:\t\t> s_icm.deduplicated.M-bias.txt <\n",
            "Processing M-bias report s_icm.deduplicated.M-bias.txt ...\n",
            "Complete\n",
            "\n",
            "No nucleotide coverage report file specified, skipping this step\n",
            "==============================================================================================================\n",
            "\n",
            "\n",
            "\n",
            "Writing Bismark HTML report to >> SRR5836473_1_bismark_bt2_PE_report.html <<\n",
            "\n",
            "==============================================================================================================\n",
            "Using the following alignment report:\t\t> SRR5836473_1_bismark_bt2_PE_report.txt <\n",
            "Processing alignment report SRR5836473_1_bismark_bt2_PE_report.txt ...\n",
            "Complete\n",
            "\n",
            "Using the following deduplication report:\t> SRR5836473_1_bismark_bt2_pe.deduplication_report.txt <\n",
            "Processing deduplication report SRR5836473_1_bismark_bt2_pe.deduplication_report.txt ...\n",
            "Complete\n",
            "\n",
            "No splitting report file specified, skipping this step\n",
            "No M-bias report file specified, skipping this step\n",
            "No nucleotide coverage report file specified, skipping this step\n",
            "==============================================================================================================\n",
            "\n",
            "\n",
            "\n",
            "Writing Bismark HTML report to >> SRR5836475_1_bismark_bt2_PE_report.html <<\n",
            "\n",
            "==============================================================================================================\n",
            "Using the following alignment report:\t\t> SRR5836475_1_bismark_bt2_PE_report.txt <\n",
            "Processing alignment report SRR5836475_1_bismark_bt2_PE_report.txt ...\n",
            "Complete\n",
            "\n",
            "Using the following deduplication report:\t> SRR5836475_1_bismark_bt2_pe.deduplication_report.txt <\n",
            "Processing deduplication report SRR5836475_1_bismark_bt2_pe.deduplication_report.txt ...\n",
            "Complete\n",
            "\n",
            "No splitting report file specified, skipping this step\n",
            "No M-bias report file specified, skipping this step\n",
            "No nucleotide coverage report file specified, skipping this step\n",
            "==============================================================================================================\n",
            "\n",
            "\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "! bismark2report -mbias_report s_epiblast.deduplicated.M-bias.txt"
      ],
      "metadata": {
        "id": "DoeLlJ2atugN",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "25914bff-bb39-463e-e9ad-a9bcd8cbf113"
      },
      "execution_count": 68,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Found 3 alignment reports in current directory. Now trying to figure out whether there are corresponding optional reports\n",
            "\n",
            "Writing Bismark HTML report to >> SRR3824222_1_bismark_bt2_PE_report.html <<\n",
            "\n",
            "==============================================================================================================\n",
            "Using the following alignment report:\t\t> SRR3824222_1_bismark_bt2_PE_report.txt <\n",
            "Processing alignment report SRR3824222_1_bismark_bt2_PE_report.txt ...\n",
            "Complete\n",
            "\n",
            "Using the following deduplication report:\t> SRR3824222_1_bismark_bt2_pe.deduplication_report.txt <\n",
            "Processing deduplication report SRR3824222_1_bismark_bt2_pe.deduplication_report.txt ...\n",
            "Complete\n",
            "\n",
            "No splitting report file specified, skipping this step\n",
            "Using the following M-bias report:\t\t> s_epiblast.deduplicated.M-bias.txt <\n",
            "Processing M-bias report s_epiblast.deduplicated.M-bias.txt ...\n",
            "Complete\n",
            "\n",
            "No nucleotide coverage report file specified, skipping this step\n",
            "==============================================================================================================\n",
            "\n",
            "\n",
            "\n",
            "Writing Bismark HTML report to >> SRR5836473_1_bismark_bt2_PE_report.html <<\n",
            "\n",
            "==============================================================================================================\n",
            "Using the following alignment report:\t\t> SRR5836473_1_bismark_bt2_PE_report.txt <\n",
            "Processing alignment report SRR5836473_1_bismark_bt2_PE_report.txt ...\n",
            "Complete\n",
            "\n",
            "Using the following deduplication report:\t> SRR5836473_1_bismark_bt2_pe.deduplication_report.txt <\n",
            "Processing deduplication report SRR5836473_1_bismark_bt2_pe.deduplication_report.txt ...\n",
            "Complete\n",
            "\n",
            "No splitting report file specified, skipping this step\n",
            "No M-bias report file specified, skipping this step\n",
            "No nucleotide coverage report file specified, skipping this step\n",
            "==============================================================================================================\n",
            "\n",
            "\n",
            "\n",
            "Writing Bismark HTML report to >> SRR5836475_1_bismark_bt2_PE_report.html <<\n",
            "\n",
            "==============================================================================================================\n",
            "Using the following alignment report:\t\t> SRR5836475_1_bismark_bt2_PE_report.txt <\n",
            "Processing alignment report SRR5836475_1_bismark_bt2_PE_report.txt ...\n",
            "Complete\n",
            "\n",
            "Using the following deduplication report:\t> SRR5836475_1_bismark_bt2_pe.deduplication_report.txt <\n",
            "Processing deduplication report SRR5836475_1_bismark_bt2_pe.deduplication_report.txt ...\n",
            "Complete\n",
            "\n",
            "No splitting report file specified, skipping this step\n",
            "No M-bias report file specified, skipping this step\n",
            "No nucleotide coverage report file specified, skipping this step\n",
            "==============================================================================================================\n",
            "\n",
            "\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "! ls *bedGraph.gz | xargs -tI{} gzip -d {}"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "UhhxqPMbViP8",
        "outputId": "04177f5b-98a7-4f57-f607-70e51c3a583f"
      },
      "execution_count": 97,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "gzip -d s_8_cell.deduplicated.bedGraph.gz \n",
            "gzip -d s_epiblast.deduplicated.bedGraph.gz \n",
            "gzip -d s_icm.deduplicated.bedGraph.gz \n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "! head s_8_cell.deduplicated.bedGraph"
      ],
      "metadata": {
        "id": "aOqrAfgft6S4",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "308471d5-83b1-41b8-929f-b28ca670d8d1"
      },
      "execution_count": 98,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "track type=bedGraph\n",
            "chr11\t3100030\t3100031\t50\n",
            "chr11\t3100047\t3100048\t66.6666666666667\n",
            "chr11\t3100048\t3100049\t50\n",
            "chr11\t3100223\t3100224\t87.5\n",
            "chr11\t3100224\t3100225\t53.3333333333333\n",
            "chr11\t3100285\t3100286\t66.6666666666667\n",
            "chr11\t3100286\t3100287\t33.3333333333333\n",
            "chr11\t3100309\t3100310\t62.5\n",
            "chr11\t3100310\t3100311\t50\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "! head s_icm.deduplicated.bedGraph"
      ],
      "metadata": {
        "id": "4j7tQZOnt_o2",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "97ea5b43-ab75-47d8-cd40-b14373dedf7b"
      },
      "execution_count": 99,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "track type=bedGraph\n",
            "chr11\t3100030\t3100031\t33.3333333333333\n",
            "chr11\t3100031\t3100032\t50\n",
            "chr11\t3100047\t3100048\t50\n",
            "chr11\t3100048\t3100049\t57.1428571428571\n",
            "chr11\t3100223\t3100224\t36.3636363636364\n",
            "chr11\t3100224\t3100225\t22.7272727272727\n",
            "chr11\t3100285\t3100286\t25\n",
            "chr11\t3100286\t3100287\t28.5714285714286\n",
            "chr11\t3100309\t3100310\t22.2222222222222\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "! head s_epiblast.deduplicated.bedGraph"
      ],
      "metadata": {
        "id": "_tIxn_RGt_vE",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "be347ceb-e702-4beb-ced2-6827e85241bd"
      },
      "execution_count": 100,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "track type=bedGraph\n",
            "chr11\t3100030\t3100031\t75\n",
            "chr11\t3100031\t3100032\t100\n",
            "chr11\t3100047\t3100048\t100\n",
            "chr11\t3100048\t3100049\t100\n",
            "chr11\t3100223\t3100224\t96.875\n",
            "chr11\t3100224\t3100225\t93.75\n",
            "chr11\t3100285\t3100286\t93.3333333333333\n",
            "chr11\t3100286\t3100287\t92.3076923076923\n",
            "chr11\t3100309\t3100310\t96.7741935483871\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "import pandas as pd\n",
        "\n",
        "df = pd.read_csv(\"s_8_cell.deduplicated.bedGraph\", delimiter='\\t', skiprows=1, header=None)\n",
        "df.head()"
      ],
      "metadata": {
        "id": "J-zVuSzt5Rw3",
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 204
        },
        "outputId": "b3e24710-bbdf-4768-f390-e97609f9f111"
      },
      "execution_count": 101,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/html": [
              "\n",
              "  <div id=\"df-16d9a163-b630-4585-95e7-436436a97b7d\">\n",
              "    <div class=\"colab-df-container\">\n",
              "      <div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>0</th>\n",
              "      <th>1</th>\n",
              "      <th>2</th>\n",
              "      <th>3</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>chr11</td>\n",
              "      <td>3100030</td>\n",
              "      <td>3100031</td>\n",
              "      <td>50.000000</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>chr11</td>\n",
              "      <td>3100047</td>\n",
              "      <td>3100048</td>\n",
              "      <td>66.666667</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>chr11</td>\n",
              "      <td>3100048</td>\n",
              "      <td>3100049</td>\n",
              "      <td>50.000000</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3</th>\n",
              "      <td>chr11</td>\n",
              "      <td>3100223</td>\n",
              "      <td>3100224</td>\n",
              "      <td>87.500000</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4</th>\n",
              "      <td>chr11</td>\n",
              "      <td>3100224</td>\n",
              "      <td>3100225</td>\n",
              "      <td>53.333333</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "</div>\n",
              "      <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-16d9a163-b630-4585-95e7-436436a97b7d')\"\n",
              "              title=\"Convert this dataframe to an interactive table.\"\n",
              "              style=\"display:none;\">\n",
              "        \n",
              "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
              "       width=\"24px\">\n",
              "    <path d=\"M0 0h24v24H0V0z\" fill=\"none\"/>\n",
              "    <path d=\"M18.56 5.44l.94 2.06.94-2.06 2.06-.94-2.06-.94-.94-2.06-.94 2.06-2.06.94zm-11 1L8.5 8.5l.94-2.06 2.06-.94-2.06-.94L8.5 2.5l-.94 2.06-2.06.94zm10 10l.94 2.06.94-2.06 2.06-.94-2.06-.94-.94-2.06-.94 2.06-2.06.94z\"/><path d=\"M17.41 7.96l-1.37-1.37c-.4-.4-.92-.59-1.43-.59-.52 0-1.04.2-1.43.59L10.3 9.45l-7.72 7.72c-.78.78-.78 2.05 0 2.83L4 21.41c.39.39.9.59 1.41.59.51 0 1.02-.2 1.41-.59l7.78-7.78 2.81-2.81c.8-.78.8-2.07 0-2.86zM5.41 20L4 18.59l7.72-7.72 1.47 1.35L5.41 20z\"/>\n",
              "  </svg>\n",
              "      </button>\n",
              "      \n",
              "  <style>\n",
              "    .colab-df-container {\n",
              "      display:flex;\n",
              "      flex-wrap:wrap;\n",
              "      gap: 12px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert {\n",
              "      background-color: #E8F0FE;\n",
              "      border: none;\n",
              "      border-radius: 50%;\n",
              "      cursor: pointer;\n",
              "      display: none;\n",
              "      fill: #1967D2;\n",
              "      height: 32px;\n",
              "      padding: 0 0 0 0;\n",
              "      width: 32px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert:hover {\n",
              "      background-color: #E2EBFA;\n",
              "      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "      fill: #174EA6;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert {\n",
              "      background-color: #3B4455;\n",
              "      fill: #D2E3FC;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert:hover {\n",
              "      background-color: #434B5C;\n",
              "      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
              "      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
              "      fill: #FFFFFF;\n",
              "    }\n",
              "  </style>\n",
              "\n",
              "      <script>\n",
              "        const buttonEl =\n",
              "          document.querySelector('#df-16d9a163-b630-4585-95e7-436436a97b7d button.colab-df-convert');\n",
              "        buttonEl.style.display =\n",
              "          google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "\n",
              "        async function convertToInteractive(key) {\n",
              "          const element = document.querySelector('#df-16d9a163-b630-4585-95e7-436436a97b7d');\n",
              "          const dataTable =\n",
              "            await google.colab.kernel.invokeFunction('convertToInteractive',\n",
              "                                                     [key], {});\n",
              "          if (!dataTable) return;\n",
              "\n",
              "          const docLinkHtml = 'Like what you see? Visit the ' +\n",
              "            '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n",
              "            + ' to learn more about interactive tables.';\n",
              "          element.innerHTML = '';\n",
              "          dataTable['output_type'] = 'display_data';\n",
              "          await google.colab.output.renderOutput(dataTable, element);\n",
              "          const docLink = document.createElement('div');\n",
              "          docLink.innerHTML = docLinkHtml;\n",
              "          element.appendChild(docLink);\n",
              "        }\n",
              "      </script>\n",
              "    </div>\n",
              "  </div>\n",
              "  "
            ],
            "text/plain": [
              "       0        1        2          3\n",
              "0  chr11  3100030  3100031  50.000000\n",
              "1  chr11  3100047  3100048  66.666667\n",
              "2  chr11  3100048  3100049  50.000000\n",
              "3  chr11  3100223  3100224  87.500000\n",
              "4  chr11  3100224  3100225  53.333333"
            ]
          },
          "metadata": {},
          "execution_count": 101
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "import seaborn as sns"
      ],
      "metadata": {
        "id": "IaD31Aj45Xy7"
      },
      "execution_count": 102,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "sns.set_style(\"darkgrid\")"
      ],
      "metadata": {
        "id": "FJCgPlGi5Zel"
      },
      "execution_count": 103,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "import matplotlib.pyplot as plt\n",
        "\n",
        "plt.figure(figsize=[10, 6])\n",
        "plt.title(\"Распределение метилирования 8cell\", fontsize=16)\n",
        "plt.hist(df[3], bins=100, density=True)\n",
        "plt.show()"
      ],
      "metadata": {
        "id": "4STdxlbj5bWV",
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 392
        },
        "outputId": "a8d1652b-7435-47c7-96e0-e28122df3a87"
      },
      "execution_count": 104,
      "outputs": [
        {
          "output_type": "display_data",
          "data": {
            "image/png": "iVBORw0KGgoAAAANSUhEUgAAAlMAAAF3CAYAAACBlM5VAAAABHNCSVQICAgIfAhkiAAAAAlwSFlzAAALEgAACxIB0t1+/AAAADh0RVh0U29mdHdhcmUAbWF0cGxvdGxpYiB2ZXJzaW9uMy4yLjIsIGh0dHA6Ly9tYXRwbG90bGliLm9yZy+WH4yJAAAgAElEQVR4nO3de3zMd97//2fODZHEISZFcDnUqqQObYkt0oZIieNi99KWonQva8XiapfqpStKtFXHVTYbhyq1VqsssbSixC6X1YMrVw+76urGhspQzaHRkGSS7x9+mV+nOUx4J5mRPO63W263zOfznvf7NZ/3fDLPfD6fmfEoLS0tFQAAAG6Lp6sLAAAAuJMRpgAAAAwQpgAAAAwQpgAAAAwQpgAAAAwQpgAAAAwQplAndu/erS5duth/evbsqREjRmjbtm0qLi52dXkA7hA2m01btmzRsGHD1KNHD/Xr108zZszQ3//+91obs0uXLlq7dq399tq1a9WlS5daGw93Hm9XF4CGZfXq1QoNDVV+fr4OHjyoxYsX6+rVq5o1a5arSwNwB1i9erV+//vf6+mnn1ZkZKSys7O1YcMGPfnkk9q7d69CQ0NdXSIaIMIU6lTXrl3Vrl07SVK/fv10/vx5bd26lTAFoFp2796toUOHavbs2fZlXbp00dChQ3X06FH9+7//uwurQ0PFaT64VEREhPLz83X16lVJUkpKiiZOnKjIyEj17NlTo0aN0jvvvFPufsXFxUpKStLQoUMVERGhyMhIPfXUU/q///s/SdKpU6ccTiv+8GfXrl32vubNm6cBAwboo48+0pgxYxQREaHo6Gi98cYb5cbNzMzU3LlzFRkZqfDwcI0cOVLvvfdehY9twIABFY59/vz52+6zssfzfQUFBXrllVcUHR2t8PBwRUdHa/369SopKbG3Kds+J06ccLjvhQsXKtw+EyZMcGj3wQcfqEuXLoqOjr7lsStSNm6XLl106tQph3Uff/xxpdtu586dGjFihCIiItSnTx8999xzysnJcbq9yn527959y4+xrH1V/UnShAkTNG/ePIf77dmzR126dHEYq2wuDh06pHnz5unBBx9Ur169NHfuXGVnZzvcPz8/XwkJCerXr5/Cw8MVGxurLVu26PtfZPHD53737t01cuRIHThwwKGvv/zlL5o2bZr69eun7t27a9iwYdq0aZNsNptDu+joaP3nf/5nuW0wYcIEjR8/vty4Fy5csC8rKirSkCFDys1rUVGRVqxYoUGDBik8PLzK/eOHioqK1LhxY4dlgYGBklTuefa3v/1NkydP1v33368ePXpoxIgRDs9tyflzCKgOjkzBpS5cuCAvLy81atRI0s1gERsbq6efflqenp46ffq0nn/+eV2/ft3hD/ecOXN0+PBhPfnkk+rbt68KCwt1+vRpXblyRR07drS3e/755xUREWG/feXKFf3yl78sV0d+fr5mz56tadOmqW3btjpw4IBefPFFNW7cWD/5yU8kSZcuXdJPf/pTNW/eXPPnz1ezZs104MABzZw5U+vWrdPAgQPL9RsVFaVf/OIXkqSjR49q/fr1Dutvp8+xY8dq3LhxkqRdu3bprbfesq8rLi62h8rp06erS5cuOnPmjF577TXl5uaWe3G/HTabTYsWLZKXl5fD8poYOzg4WDt27FCfPn3sy3bs2KHg4OByL3DLly/X5s2bNWHCBD377LOyWq1atWqVvvjiC/3hD3+Ql5eXdu7caW+/a9cuHTt2TL/97W/ty9q2bXtLj/H7QkJC7H1V9rz6vvz8fC1fvrzSPpcuXaof//jHevXVV3X+/HmtWLFCly9ftof6kpISPf300/rss88UHx+ve+65R0ePHlViYqK++eYbzZkzx6G/hQsXqlu3bvr222+1adMmzZ07V506ddI999wj6ea+1rdvXz3xxBPy8/PTJ598orVr1+qbb76pMDzdji1btlQYjpKTk5WUlKQZM2YoMjJSPj4+Fe4fFXnssce0ZcsWDRgwwH6a76WXXlJoaKiGDBlib3f48GHFx8erV69eSkhIUNOmTfXFF1/oq6++srepznMIqA7CFOqUzWZTcXGxrl27pj//+c9677339Mgjj8jf31+S9B//8R/2tiUlJerdu7euXLmiHTt22MPUyZMndejQIS1YsEATJ060tx80aFC58Tp27KgePXrYb3//v+bvu3btmhYvXqy4uDhJN48qWa1WrV27VqNHj5aHh4fWrl2r0tJSvfHGG2ratKkkqX///srKytKaNWvKBZ+ioiK1aNHCPv6XX35Zbtxb6bOwsFCS1Lp1a3ufx48fd+hv//79+vDDD7Vt2zY9+OCDkqS+fftKktatW6dp06apefPmFW6D6tq2bZsuXLig0aNH6+TJkzU69qhRo7R9+3Z9/fXXatGihbKzs3Xw4EE99thj2rx5s73dhQsXtHHjRs2YMcMhxLRv316PPfaY3n//fQ0aNMhh7o8fPy5fX1+HZbf6GMsUFRXJ39/f3ldlz6vvW7NmjTw9PTVo0KByR5wkqVOnTkpMTLTfDgoK0jPPPKOTJ0+qb9++OnbsmD788EMlJibaA36/fv1UUFCgTZs2adKkSWrWrJlDf2X1hYSEaOTIkTp79qw9TH3/n5PS0lI98MADKioq0qZNmzRnzhx5epqduLBarXrttdf02GOPlTvKe+bMGXXu3FkzZ860L6to/6jIrFmz5OPjo5kzZ9qPRLVv395hHyotLdXSpUv1ox/9SFu3brU/lh//+Mf2fqr7HAKqg9N8qFNDhgxRt27d1Lt3by1atEjDhw/X0qVL7eszMjI0Z84c9e/fX926dVO3bt20a9cu/fOf/7S3+etf/yoPDw/99Kc/rbG6vLy8NHjwYIdlQ4cO1VdffSWr1Srp5otxVFSUmjRpouLiYvtPv3799Pe//135+fkO9y8oKLCHxMrcSp83btyQJPn5+VXZX+vWrdWzZ0+H/h566CEVFRXpzJkzDu1LSkoc2jk7Hff1119r7dq1mj59erkLfW917Ip06tRJPXv2tJ+K2b17t+699157AChz4sQJlZSUaMSIEQ5jde/eXY0bN9bp06edjnU7j7HM9evXq5yHH/rHP/6h7du3a968efajsD/0/aMqkvToo4/K09NTH3/8sSTp9OnT8vT01LBhwxzajRgxosq5zcnJ0a5du+Tn56fu3bvb11++fFkLFy7UI488ovDwcHXr1k2rVq1SXl6e/bR7mdLSUoftXFxc7HBqsSKJiYlq06ZNudOnkhQWFqaMjAy9//77KigoqNZzr8ybb76pDRs2aPr06dq6datWr16txo0ba8qUKfZ99csvv9TFixc1bty4SkNhbT6H0PBwZAp1at26dbJYLGrcuLFat27t8IJ07do1TZkyRXfddZfmzp2rtm3bysfHRzt27NDbb79tb5eTk6OgoCDdddddNVZXYGCgfHx8HJa1aNFC0s3/sENDQ/XNN99oz5492rNnT4V9ZGdnKyAgQJKUl5engoIChYSEVDnurfRZdporODi4yv4uXryobt26Vbj+h6fKnnrqqSrr+6GXX35ZLVq00KRJk/S73/3OaOzKjB8/Xq+88oqmTZumnTt3asaMGeWu4yl7sY+JiTEaqyJVPcbv91/VPPzQ4sWL9cADD2jo0KFKS0ursE3Z862Mr6+vAgMD7QEhNzdXQUFB8vX1rfB+ubm5DssnTZpk//2uu+7Sf/3XfyksLEzSzaA1ffp0Xb58WTNnzlSHDh3k5+enw4cPa8OGDfbgXmb//v3av39/uZp79epV4WM5efKkDh48qDfeeKPCU2UzZszQxYsX9Ytf/KLaIUq6ud0TExP11FNPKT4+3r48MjJS0dHR2rhxo8M1T1W9s682n0NoeAhTqFOdO3e2v5vvh86cOaOLFy9q+/bteuCBB+zLt23b5tCuadOmys3N1fXr12ssUOXl5amoqMghUH399deSJIvFIulmiLn//vs1bdq0CvsoaydJZ8+elXTz8VblVvrMyMiQVPl1PmX9tWnTRqtWrapwfevWrR1uL1y4UPfdd5/99pUrVzR9+vQK7/vBBx/oT3/6kzZt2lTuBf12xq5MTEyMlixZoqVLlyo3N1dDhgwp90JeFmQ2bdpkv/i4ovW3ytljLHP+/Hn179+/Wn3+6U9/0pkzZ7R3794q25U938oUFhYqLy/P/hwICgpSbm6uCgsLHWoru19QUJDD/RctWqRu3bqpqKhIH374oRYtWiQfHx+NGjVK//rXv/TJJ5/o5Zdf1siRI+33ef/99yusbcCAAQ7hRZJeeOGFCtsWFxdr8eLFGjZsmB588MEKT4E2bdpUy5cv1/Dhw9W9e3dNmTJFR48edbierSIZGRkqLCx0uA5Sujnfbdu2tb8Bpex0X1kQrUhtPYfQMBGm4DYKCgokySHQ5ObmKjU11aHdQw89pKSkJO3atavCUwi3w2az6d1337VfMyVJBw4cUKtWrewvZv3799fHH3+szp07Ow1xR48elb+/v8OF1BW5nT7Dw8Or7O/dd99Vo0aNHC7Er8y//du/ObwwVXbtj81m0+LFizV48GCH605Mxq6Mj4+Pxo4dqw0bNmjKlCkVhpqHHnpInp6e+uqrr/TQQw/d9ljfV53HKEmffPKJrly5Yr8urCrXrl3Tyy+/rAkTJjjdJn/+8581duxY++2DBw+qpKREPXv2lCT17t1bGzdu1MGDBzVixAh7u3379snHx6fctWDfn9tevXopJSVFKSkpGjVqlK5fvy7JcV8rKirSvn37KqwtKCioXIBp3LhxhR+4u3XrVlmtVm3ZsqXKx7ts2TIVFxdr0aJFCgwM1BdffFFle+n/PwqXnp7ucD1hTk6Ozp8/r3vvvVfSzcfeunVr7dq1Sz/72c/k4eFRrq/aeA6h4SJMwW306tVLAQEBWrRokeLj4/Xdd99p/fr1atq0qb799lt7u8jISMXGxmrZsmW6dOmSIiMjVVxcrNOnT+vhhx92GmAq0rhxY73yyivKzs5Wu3btlJKSohMnTmjZsmX2P8Tx8fEaN26cHn/8cT3xxBNq3bq18vLydPbsWWVmZioxMVGFhYU6fPiwtm3bpt69e+vcuXP2Mf71r39Jkj7//HM1b95cAQEB1eozKytLW7Zs0fbt2zV58uQqr8MaPny4du/erUmTJmnKlCn60Y9+pMLCQmVmZurIkSNat26d0+u4KnLmzBn5+flpw4YNdTL2lClT1KdPH3Xt2rXC9W3bttW0adO0ePFi/fOf/1Tv3r3l5+enS5cu6a9//avGjRunyMjIGn+MO3fu1IYNG9S6dWs9+uijTvtMTU1VixYtnL7TT5LOnTun+fPna+jQocrIyNDKlSvVu3dv+0X8AwYM0P33368XXnhB33zzjTp37qxjx45p165d+vnPf+5w8XlZf35+fvYjU2fPnrUfTevQoYNat26tlStXytPTU97e3nr99ded1lgdf/jDH/Tss8+qZcuWlbYpqzs5ObnCo0KVadOmjR555BFt3LhRnp6e6t27t7Kzs5WcnKyioiL7RfUeHh567rnnNHPmTE2cOFHjx49X06ZN9eWXX+rq1auKj4+vlecQGi7CFNxGs2bN9Nvf/lYvvfSS4uPj1bJlS02cOFG5ubnlDv+vWLFCv//977Vnzx5t3bpVAQEBioiIsH9kwK0KCAjQihUrtGTJEp09e1YtWrTQggULNHr0aHubVq1a6e2339batWu1YsUKZWdnKzg4WJ07d9aoUaMk3byot+zDBI8dO6Zjx46VG2vWrFnaunWr+vTpU60+T5w4ob/85S+aN2+ew7sXK+Lj46ONGzcqKSlJO3fu1IULF9SoUSOFhYXp4YcfLnddWHXZbDZNnz5dd999d52MHRQUVOXRIenmx2N06NBBb775pt588015eHgoNDRUffv2Vfv27as9Vhlnj7G4uFhr1qxR//79NXv27EovJP9hn7/+9a/LfS5SRRYsWKAjR45o9uzZstlsio6O1oIFC+zrPT09lZSUpBUrVig5OVk5OTlq3bq15s+fryeffLJcfwkJCZJuzovFYtHkyZPt757z9fXVunXrlJCQoF//+tcKCgrSmDFj1KpVKz3//PNOa61Khw4d9MQTT1S6PicnRwsWLNDPfvYz9evX75b7X7lypTZt2qSUlBRt3rxZAQEBuvfee/Wb3/zG4ejZoEGDtGnTJr322mv27RgWFuawrWr6OYSGy6PU2VsygHpu3rx5OnHiRKUXBt+KCxcuaODAgfrHP/5RaZuYmBi9+OKLt3UEDfXPqVOnNHHiRG3evNlpgATgnvhoBKAG+fr6Orz9vCJdu3a1v0MPAHDn4zQfUINatmypP/7xj1W2WbNmTR1VAwCoC5zmAwAAMMBpPgAAAAOEKQAAAAMuu2aqpKRENlvtnmH08vKo9TFw65gX98OcuCfmxf0wJ+6pLubFx6f8VyOVcVmYstlKlZPzXa2OERzcqNbHwK1jXtwPc+KemBf3w5y4p7qYl5CQJpWu4zQfAACAAcIUAACAAcIUAACAAcIUAACAAcIUAACAAcIUAACAAcIUAACAAcIUAACAAcIUAACAAcIUAACAAcIUAACAAcIUAACAAcIUAACAAW9XF1CbbKr4W54LbhQrP6+g7gsCAAD1Tr0OU3f5eKn9vJRyyzOWxSnfBfUAAID6h9N8AAAABghTAAAABghTAAAABghTAAAABghTAAAABghTAAAABghTAAAABghTAAAABghTAAAABghTAAAABghTAAAABghTAAAABghTAAAABghTAAAABghTAAAABghTAAAABghTAAAABghTAAAABghTAAAABghTAAAABghTAAAABghTAAAABghTAAAABghTAAAABghTAAAABghTAAAABghTAAAABghTAAAABghTAAAABqoVptLS0hQbG6uYmBglJSVV2u7QoUPq0qWL/vd//7fGCgQAAHBnTsOUzWZTQkKCkpOTlZKSov379+vcuXPl2uXn52vr1q3q3r17rRQKAADgjpyGqfT0dLVr105hYWHy9fVVXFycUlNTy7VbvXq1pk2bJj8/v1opFAAAwB15O2tgtVoVGhpqv22xWJSenu7Q5tNPP1VWVpYefvhhbdy4sVoDe3l5KDi40S2WW3NcOXZD5+XlyfZ3M8yJe2Je3A9z4p5cPS9Ow5QzJSUlWrZsmRITE2/pfjZbqXJyvjMdvkohIU0qXVfbY6NywcGN2P5uhjlxT8yL+2FO3FNdzEtVmcLpaT6LxaKsrCz7bavVKovFYr997do1nT17VhMnTlR0dLTOnDmj6dOncxE6AABoEJwemYqIiFBGRoYyMzNlsViUkpKiV1991b6+SZMmOnXqlP32hAkT9OyzzyoiIqJ2KgYAAHAjTsOUt7e3Fi5cqKlTp8pms2nMmDHq3LmzVq9erfDwcA0cOLAu6gQAAHBL1bpmKioqSlFRUQ7LZs2aVWHbN954w7wqAACAOwSfgA4AAGCAMAUAAGCAMAUAAGCAMAUAAGCAMAUAAGCAMAUAAGCAMAUAAGCAMAUAAGCAMAUAAGCAMAUAAGCAMAUAAGCAMAUAAGCAMAUAAGCAMAUAAGCAMAUAAGCAMAUAAGCAMAUAAGCAMAUAAGCAMAUAAGCAMAUAAGCAMAUAAGCAMAUAAGCAMAUAAGCAMAUAAGCAMAUAAGCAMAUAAGCAMAUAAGCAMAUAAGCAMAUAAGCAMAUAAGCAMAUAAGCAMAUAAGCAMAUAAGCAMAUAAGCAMAUAAGCAMAUAAGCAMAUAAGCAMAUAAGCAMAUAAGCAMAUAAGCAMAUAAGCAMAUAAGCAMAUAAGCAMAUAAGCAMAUAAGCAMAUAAGCAMAUAAGCAMAUAAGCAMAUAAGCAMAUAAGCAMAUAAGCAMAUAAGCAMAUAAGCAMAUAAGCAMAUAAGCAMAUAAGCAMAUAAGCAMAUAAGCAMAUAAGCAMAUAAGCgWmEqLS1NsbGxiomJUVJSUrn1O3bs0PDhwzVy5EiNHz9e586dq/FCAQAA3JHTMGWz2ZSQkKDk5GSlpKRo//795cLS8OHDtW/fPu3du1dTp05VYmJirRUMAADgTpyGqfT0dLVr105hYWHy9fVVXFycUlNTHdoEBATYfy8oKJCHh0fNVwoAAOCGvJ01sFqtCg0Ntd+2WCxKT08v12779u3avHmzioqK9Prrrzsd2MvLQ8HBjW6x3JrjyrEbOi8vT7a/m2FO3BPz4n6YE/fk6nlxGqaq6/HHH9fjjz+uffv2af369XrppZeqbG+zlSon57uaGr5CISFNKl1X22OjcsHBjdj+boY5cU/Mi/thTtxTXcxLVZnC6Wk+i8WirKws+22r1SqLxVJp+7i4OB0+fPgWSwQAALgzOQ1TERERysjIUGZmpgoLC5WSkqLo6GiHNhkZGfbfjx49qnbt2tV4oQAAAO7I6Wk+b29vLVy4UFOnTpXNZtOYMWPUuXNnrV69WuHh4Ro4cKC2bdumkydPytvbW4GBgU5P8QEAANQX1bpmKioqSlFRUQ7LZs2aZf/9+eefr9mqAAAA7hB8AjoAAIABwhQAAIABwhQAAIABwhQAAIABwhQAAIABwhQAAIABwhQAAIABwhQAAIABwhQAAIABwhQAAIABwhQAAIABwhQAAIABwhQAAIABwhQAAIABwhQAAIABwhQAAIABwhQAAIABwhQAAIABwhQAAIABwhQAAIABwhQAAIABwhQAAIABwhQAAIABwhQAAIABwhQAAIABwhQAAIABb1cXAAAAUF0Bgf7y93OML9eLbC6q5ibCFAAAuGP4+3mr/bwUh2UZy+L0rYvqkTjNBwAAYIQwBQAAYIAwBQAAYIAwBQAAYIAwBQAAYIAwBQAAYIAwBQAAYIAwBQAAYIAwBQAAYIAwBQAAYIAwBQAAYIAwBQAAYIAwBQAAYIAwBQAAYIAwBQAAYIAwBQAAYIAwBQAAYIAwBQAAYIAwBQAAYIAwBQAAYIAwBQAAYIAwBQAAYIAwBQAAYIAwBQAAYIAwBQAAYIAwBQAAYIAwBQAAYIAwBQAAYIAwBQAAYIAwBQAAYIAwBQAAYIAwBQAAYKBaYSotLU2xsbGKiYlRUlJSufWbN2/W0KFDNXz4cD355JO6ePFijRcKAADgjpyGKZvNpoSEBCUnJyslJUX79+/XuXPnHNp07dpVb7/9tvbt26fY2Fi98sortVYwAACAO3EaptLT09WuXTuFhYXJ19dXcXFxSk1NdWgTGRkpf39/SVKPHj2UlZVVO9UCAAC4Gadhymq1KjQ01H7bYrHIarVW2v6tt97SgAEDaqY6AAAAN+ddk53t3btXn3zyibZt2+a0rZeXh4KDG9Xk8LfElWM3dF5enmx/N8OcuCfmxf0wJ+7LlfPiNExZLBaH03ZWq1UWi6VcuxMnTmjDhg3atm2bfH19nQ5ss5UqJ+e7Wyz31oSENKl0XW2PjcoFBzdi+7sZ5sQ9MS/uhzlxvcpe212ZKZye5ouIiFBGRoYyMzNVWFiolJQURUdHO7T57LPPtHDhQq1fv17Nmzc3rxgAAOAO4fTIlLe3txYuXKipU6fKZrNpzJgx6ty5s1avXq3w8HANHDhQL7/8sr777jvNmjVLknT33Xdrw4YNtV48AACAq1XrmqmoqChFRUU5LCsLTpK0ZcuWGi0KAADgTsEnoAMAABggTAEAABggTAEAABggTAEAABggTAEAABggTAEAABggTAEAABggTAEAABggTAEAABggTAEAABggTAEAABggTAEAABggTAEAABggTAEAABggTAEAABggTAEAABggTAEAABggTAEAABggTAEAABggTAEAABggTAEAABggTAEAABggTAEAABggTAEAABggTAEAABggTAEAABggTAEAABggTAEAABggTAEAABggTAEAABggTAEAABggTAEAABggTAEAABggTAEAABggTAEAABggTAEAABggTAEAABggTAEAABggTAEAABggTAEAABggTAEAABggTAEAABggTAEAABggTAEAABggTAEAABggTAEAABggTAEAABggTAEAABggTAEAABjwdnUBAFCTAgL95e/n+Ket4Eax8vMKXFQRgPqOMAWgXvH381b7eSkOyzKWxSnfRfUAqP84zQcAAGCAMAUAAGCAMAUAAGCAMAUAAGCAMAUAAGCAMAUAAGCAMAUAAGCAMAUAAGCAMAUAAGCgWmEqLS1NsbGxiomJUVJSUrn1p0+f1ujRo3Xvvffq4MGDNV4kAACAu3Iapmw2mxISEpScnKyUlBTt379f586dc2hz9913KzExUcOGDau1QgEAANyR0+/mS09PV7t27RQWFiZJiouLU2pqqjp16mRv06ZNG0mSpydnDQEAQMPiNP1YrVaFhobab1ssFlmt1lotCgAA4E7h9MhUbfHy8lBwcCNXDe/SsRs6Ly9Ptr+baQhzcic+voYwL3ca5sR9uXJenIYpi8WirKws+22r1SqLxWI8sM1Wqpyc74z7qUpISJNK19X22KhccHAjtr+bqU9zUtl+fyc+vvo0L/UFc+J6rtrHq8oUTk/zRUREKCMjQ5mZmSosLFRKSoqio6NrtEAAAIA7ldMw5e3trYULF2rq1KkaOnSohgwZos6dO2v16tVKTU2VdPMi9QEDBujgwYN64YUXFBcXV+uFAwAAuINqXTMVFRWlqKgoh2WzZs2y/37fffcpLS2tZisDAAC4A/BZBgAAAAYIUwAAAAYIUwAAAAYIUwAAAAYIUwAAAAYIUwAAAAYIUwAAAAYIUwAAAAYIUwAAAAYIUwAAAAYIUwAAAAYIUwAAAAYIUwAAAAYIUwAAAAYIUwAAAAYIUwAAAAYIUwAAAAYIUwAAAAYIUwAAAAYIUwAAAAYIUwAAAAYIUwAAAAYIUwAAAAYIUwAAAAYIUwAAAAYIUwAAAAYIUwAAAAYIUwAAAAYIUwAAAAYIUwAAAAa8XV0A0FAFBPrL389xFyy4Uaz8vAIXVQQAuB2EKcBF/P281X5eisOyjGVxyndRPQCA28NpPgAAAAOEKQAAAAOEKQAAAANcMwUAcMCbI4BbQ5gCADjgzRHAreE0HwAAgAHCFAAAgAHCFPXap0AAAAfgSURBVAAAgAHCFAAAgAHCFAAAgAHCFAAAgAHCFAAAgAHCFAAAgAHCFAAAgAHCFAAAgAHCFAAAgAHCFAAAgAHCFAAAgAHCFAAAgAHCFAAAgAFvVxcA1IaAQH/5+zk+vQtuFCs/r8BFFQEA6ivCFOolfz9vtZ+X4rAsY1mc8l1UDwCg/iJMAQDqBY5Iw1UIUwCAeoEj0nAVLkAHAAAwQJgCAAAwQJgCAAAwwDVTAGSTFBLSxGEZF+4CQPVUK0ylpaVpyZIlKikp0bhx4/T00087rC8sLNSzzz6rTz/9VMHBwVq5cqXatGlTKwXjzscLt/u5y8eLC3fdEPsKcGdwGqZsNpsSEhK0efNmWSwWjR07VtHR0erUqZO9za5duxQYGKj33ntPKSkpWr58uVatWlWrhePOxQs3UD3sK+6HgIuKOA1T6enpateuncLCwiRJcXFxSk1NdQhTR44c0S9/+UtJUmxsrBISElRaWioPD49aKht1ic9uAYCbCLioiNMwZbVaFRoaar9tsViUnp5ers3dd999s0NvbzVp0kTZ2dlq1qxZDZeLmlJRQJIqDkkN7bNbbmXb1EXQrGiM60U23eXjdVvLqlvf9SJbuf/AK+qvsj5N6nZVWDepWXKv54i7/cNT3W1bF3W7anvVxt+WO2HuGwKP0tLS0qoaHDx4UMePH9eSJUskSXv27FF6eroWLlxobzNs2DAlJyfbQ9egQYP0xz/+kTAFAADqPacfjWCxWJSVlWW/bbVaZbFYyrW5dOmSJKm4uFjffvutmjZtWsOlAgAAuB+nYSoiIkIZGRnKzMxUYWGhUlJSFB0d7dAmOjpa77zzjiTp0KFDioyM5HopAADQIDg9zSdJx44d09KlS2Wz2TRmzBhNnz5dq1evVnh4uAYOHKgbN27omWee0eeff66goCCtXLnSfsE6AABAfVatMAUAAICK8XUyAAAABghTAAAABuptmEpLS1NsbKxiYmKUlJTk6nIapEuXLmnChAkaOnSo4uLi9Prrr0uScnJyNHnyZA0ePFiTJ09Wbm6uiyttmGw2m0aNGqWf//znkqTMzEyNGzdOMTEx+tWvfqXCwkIXV9iw5OXlKT4+Xo8++qiGDBmijz/+mH3FDWzZskVxcXEaNmyY5syZoxs3brCv1LH58+erb9++GjZsmH1ZZftGaWmpXnzxRcXExGj48OH69NNP66TGehmmyr4CJzk5WSkpKdq/f7/OnTvn6rIaHC8vL82bN08HDhzQzp079eabb+rcuXNKSkpS37599e6776pv376EXRfZunWrOnbsaL+9fPlyTZo0Se+9954CAwP11ltvubC6hmfJkiXq37+/Dh48qL1796pjx47sKy5mtVq1detWvf3229q/f79sNpv9K9PYV+rOT37yEyUnJzssq2zfSEtLU0ZGht59910tXrxYv/nNb+qkxnoZpr7/FTi+vr72r8BB3WrZsqW6desmSQoICFCHDh1ktVqVmpqqUaNGSZJGjRqlw4cPu7LMBikrK0tHjx7V2LFjJd38b+6///u/FRsbK0kaPXo0+0wd+vbbb3X69Gn7fPj6+iowMJB9xQ3YbDZdv35dxcXFun79ukJCQthX6tiDDz6ooKAgh2WV7Rtlyz08PNSjRw/l5eXp8uXLtV5jvQxTFX0FjtVqdWFFuHDhgj7//HN1795dV69eVcuWLSVJISEhunr1qoura3iWLl2qZ555Rp6eN/8EZGdnKzAwUN7eN7+WIjQ0lH2mDl24cEHNmjXT/PnzNWrUKC1YsEDfffcd+4qLWSwWTZkyRY888oj69eungIAAdevWjX3FDVS2b/zw9b+u5qdehim4l2vXrik+Pl7PPfecAgICHNZ5eHjwAa917P3331ezZs0UHh7u6lLw/ykuLtZnn32m8ePHa8+ePfL39y93So99pe7l5uYqNTVVqampOn78uAoKCnT8+HFXl4UfcId9w+kXHd+JqvMVOKgbRUVFio+P1/DhwzV48GBJUvPmzXX58mW1bNlSly9f5jsc69hHH32kI0eOKC0tTTdu3FB+fr6WLFmivLw8FRcXy9vbW1lZWewzdSg0NFShoaHq3r27JOnRRx9VUlIS+4qLnThxQm3atLFv98GDB+ujjz5iX3EDle0bP3z9r6v5qZdHpqrzFTiofaWlpVqwYIE6dOigyZMn25dHR0drz549km5+cfbAgQNdVWKDNHfuXKWlpenIkSNasWKFIiMj9eqrr6pPnz46dOiQJOmdd95hn6lDISEhCg0N1ZdffilJOnnypDp27Mi+4mKtWrXS//zP/6igoEClpaU6efKkOnXqxL7iBirbN8qWl5aW6syZM2rSpIn9dGBtqrefgF7RV+Cgbn3wwQd6/PHHdc8999ivzZkzZ47uu+8+/epXv9KlS5fUqlUrrVq1SsHBwS6utmE6deqUNm3apN/97nfKzMzU7NmzlZubq65du2r58uXy9fV1dYkNxueff64FCxaoqKhIYWFhSkxMVElJCfuKi61Zs0YHDhyQt7e3unbtqiVLlshqtbKv1KE5c+bob3/7m7Kzs9W8eXPNnDlTgwYNqnDfKC0tVUJCgo4fPy5/f38tXbpUERERtV5jvQ1TAAAAdaFenuYDAACoK4QpAAAAA4QpAAAAA4QpAAAAA4QpAAAAA4QpAAAAA4QpAAAAA4QpAAAAA/8P2+H+W0OAvJgAAAAASUVORK5CYII=\n",
            "text/plain": [
              "<Figure size 720x432 with 1 Axes>"
            ]
          },
          "metadata": {}
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "df2 = pd.read_csv(\"s_epiblast.deduplicated.bedGraph\", delimiter='\\t', skiprows=1, header=None)\n",
        "\n",
        "plt.figure(figsize=[10, 6])\n",
        "plt.title(\"Распределение метилирования Epiblast\", fontsize=16)\n",
        "plt.hist(df2[3], bins=100, density=True)\n",
        "plt.show()"
      ],
      "metadata": {
        "id": "N0t3J1Wb5fK1",
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 392
        },
        "outputId": "b3ba4a62-a464-4b42-f40c-06b1074c6506"
      },
      "execution_count": 105,
      "outputs": [
        {
          "output_type": "display_data",
          "data": {
            "image/png": "iVBORw0KGgoAAAANSUhEUgAAAlMAAAF3CAYAAACBlM5VAAAABHNCSVQICAgIfAhkiAAAAAlwSFlzAAALEgAACxIB0t1+/AAAADh0RVh0U29mdHdhcmUAbWF0cGxvdGxpYiB2ZXJzaW9uMy4yLjIsIGh0dHA6Ly9tYXRwbG90bGliLm9yZy+WH4yJAAAgAElEQVR4nO3deXxU1cH/8W92lpCEJUzYlcUUk4C4QJAlGpYAYQkFbBVRoKBFBQQeEYoNEmSpIqsUmkZABJEiCoXQoAQhtPDw4EKpyiPlsdEEYdjDFkgyye8PfpkXQyZMwskyhM/79crrlblz5p5z75k7851z7p3xKCgoKBAAAABui2dlNwAAAOBORpgCAAAwQJgCAAAwQJgCAAAwQJgCAAAwQJgCAAAwQJhCqX388ccKDQ21/7Vr1079+/fXmjVrlJeXV9nNA3CHGDZsmMNryY1/s2bNKvX6lixZotDQUIdloaGhWrBgQYnaMmzYsFLX6UpmZqaWLFmijIyMMl833Id3ZTcAd65FixYpJCREly5dUkpKimbOnKkzZ85o/Pjxld00AHeI0NBQJSQkFFkeHBxc6nUNGTJEXbp0KYtmlZljx47pnXfe0UMPPaQmTZpUdnNQTghTuG2tW7dWs2bNJEmdO3fWjz/+qNWrVxOmAJRYzZo19cADD5TJukJCQhQSElIm6wJKg2k+lJmIiAhdunRJZ86ckSQlJyfrmWeeUWRkpNq1a6e4uDh98sknRR6Xl5enxMRE9enTRxEREYqMjNRvfvMb/d///Z8kaf/+/cVOBYSGhmrDhg32dU2ZMkVdu3bVV199pUGDBikiIkLR0dF6//33i9SbkZGhSZMmKTIyUuHh4RowYIA+++wzp9vWtWtXp3X/+OOPt73O4rbnRtnZ2XrrrbcUHR2t8PBwRUdHa9myZcrPz7eXKdw/e/fudXhsZmam0/1z81TGF198odDQUEVHR5e6bmcK6w0NDdX+/fsd7vv666+L3Xfr169X//79FRERoQ4dOuh3v/udzp8/73J/Ff59/PHHpd7GwvK3Wp90fQpoypQpDo/btGmTQkNDHeoq7Ivt27drypQpeuSRR/Tggw9q0qRJOnfunMPjL126pISEBHXu3Fnh4eGKiYnRqlWrdOOPUtz83G/btq0GDBigbdu2Oazr73//u0aPHq3OnTurbdu26tu3r1asWCGbzeZQLjo6Wv/1X/9VZB8MGzZMTz75ZJF6MzMz7ctyc3PVu3fvIv2am5ur+fPnq3v37goPD7/l8XG7Ctu3Y8cO9e3bV+Hh4erVq1eR/eBsmq/QsmXL1LVrV7Vp00ZDhw7V4cOHb1nntWvXNHv2bPXt21ft2rVTp06d9Nvf/tb+ulTo1KlTevXVV+392LlzZz3//PM6c+aM9u/fr2eeeUaSNGLEiGKPC9z5GJlCmcnMzJSXl5dq1Kgh6XqwiImJ0XPPPSdPT08dOHBAr732mq5everwwj1x4kTt2LFDzz77rDp27KicnBwdOHBAp06dUosWLezlXnvtNUVERNhvnzp1Si+99FKRdly6dEkTJkzQ6NGj1bRpU23btk1vvPGGatasqV/+8peSpOPHj+uJJ55Q3bp1NXXqVNWpU0fbtm3T2LFjtXTpUnXr1q3IeqOiovTCCy9Iknbt2qVly5Y53H876xw8eLCGDBkiSdqwYYM++ugj+315eXn2UDlmzBiFhobq4MGD+uMf/6isrKwib+63w2azacaMGfLy8nJYXhZ1BwUFad26derQoYN92bp16xQUFOQQkiRp3rx5WrlypYYNG6bJkyfLarVq4cKF+ve//60PP/xQXl5eWr9+vb38hg0btHv3br3zzjv2ZU2bNi3VNt4oODjYvq7inlc3unTpkubNm1fsOmfPnq1HH31Ub7/9tn788UfNnz9fJ0+etIf6/Px8Pffcc/ruu+80btw43Xfffdq1a5fmzJmjs2fPauLEiQ7ri4+PV1hYmC5evKgVK1Zo0qRJatmype677z5J14+1jh076umnn5afn5+++eYbLVmyRGfPnnUanm7HqlWrnIajpKQkJSYm6sUXX1RkZKR8fHycHh+34uxcSy8vL3l4eNhv//TTT5o1a5Zeeukl1a1bV+vWrdPEiRNVp04dRUZG3nL9mzZtUoMGDRQfH6+cnBwtWrRIw4cP1/bt2xUUFOT0MTk5Obp8+bLGjBmj4OBgZWVl6YMPPtCvf/1rbdu2zT4NOXnyZP3888+aPHmyGjRooNOnT2vfvn3Kzs5WWFiY4uPjlZCQ4PD61bJlyxLvG9wZCFO4bTabTXl5ebp8+bL+9re/6bPPPtPjjz+u6tWrS5J++9vf2svm5+erffv2OnXqlNatW2cPU/v27dP27ds1bdo0+yc4SerevXuR+lq0aOEwHXDjp+YbXb58WTNnzlRsbKyk66NKVqtVS5Ys0cCBA+Xh4aElS5aooKBA77//vmrXri1J6tKli06cOKHFixcXCT65ubmqV6+evf4ffvihSL2lWWdOTo4kqVGjRvZ17tmzx2F9W7du1Zdffqk1a9bokUcekSR17NhRkrR06VKNHj1adevWdboPSmrNmjXKzMzUwIEDtW/fvjKtOy4uTmvXrtXp06dVr149nTt3TikpKXrqqae0cuVKe7nMzEy9++67evHFFx1CzD333KOnnnpKn3/+ubp37+7Q93v27JGvr2+JpoeK28ZCubm5ql69un1dxT2vbrR48WJ5enqqe/fuRUacpOtvlnPmzLHfDgwM1CuvvKJ9+/apY8eO2r17t7788kvNmTPHHvA7d+6s7OxsrVixQsOHD1edOnUc1lfYvuDgYA0YMEBHjhyxh6kbP5wUFBTo4YcfVm5urlasWKGJEyfK09NsEsJqteqPf/yjnnrqqSKjvAcPHlSrVq00duxY+zJnx0dxvvrqK4WFhRVZvmjRIvXq1ct++/Tp01q/fr19P3Tp0kWxsbFavHixyzB19epVrVixwv5Br02bNvaRwJdfftnpY2rVquVwErzNZlPnzp316KOPKjk5WcOHD5d0ffsnTJig/v3728v27t3b/n9hcLr59QtVC9N8uG29e/dWWFiY2rdvrxkzZqhfv36aPXu2/f709HRNnDhRXbp0UVhYmMLCwrRhwwb95z//sZf5xz/+IQ8PDz3xxBNl1i4vLy/17NnTYVmfPn30888/y2q1Srr+ZhwVFaVatWopLy/P/te5c2f97//+ry5duuTw+OzsbHtILE5p1nnt2jVJkp+f3y3X16hRI7Vr185hfZ06dVJubq4OHjzoUD4/P9+hnKvpuNOnT2vJkiUaM2ZMkfNMSlu3My1btlS7du3s04wff/yx7r//fnsAKLR3717l5+erf//+DnW1bdtWNWvW1IEDB1zWdTvbWOjq1au37Iebff/991q7dq2mTJlif3O+2Y1vppLUq1cveXp66uuvv5YkHThwQJ6enurbt69Duf79+9+yb8+fP68NGzbIz89Pbdu2td9/8uRJxcfH6/HHH1d4eLjCwsK0cOFCXbhwwT7tXqigoMBhP+fl5cnV793PmTNHjRs3dnq1W5MmTZSenq7PP/9c2dnZJXru3egXv/iFPvrooyJ/heG9UIMGDRzCiJeXl3r16qVDhw65rC8qKsqhrxo3bqy2bdu6fB5v27ZNQ4YM0cMPP6z7779fDzzwgK5cueIQFsPDw/Xuu+/qvffe0/fff+9yX6JqYmQKt23p0qWyWCyqWbOmGjVq5PCGdPnyZY0cOVLVqlXTpEmT1LRpU/n4+GjdunXauHGjvdz58+cVGBioatWqlVm7AgIC5OPj47CsXr16kq5/wg4JCdHZs2e1adMmbdq0yek6zp07J39/f0nShQsXlJ2d7fLqotKss3Caq7gphsL1HTt2zOmn9hvXUeg3v/nNLdt3szfffFP16tXT8OHD9ac//cmo7uI8+eSTeuuttzR69GitX79eL774YpHzeArf7Hv06GFUlzO32sYb13+rfrjZzJkz9fDDD6tPnz5KS0tzWqbw+VbI19dXAQEB9jCflZWlwMBA+fr6On1cVlaWw/LCURBJqlatmn7/+9/brwzLz8/XmDFjdPLkSY0dO1bNmzeXn5+fduzYoeXLl9uDe6GtW7dq69atRdr84IMPOt2Wffv2KSUlRe+//77Tac0XX3xRx44d0wsvvFCqEFWoRo0aDtP3xXE2ElqvXj3l5ubq7NmzRfZ5SR7773//u9jH7Ny5UxMmTNDAgQP10ksvqXbt2vLw8NBzzz1nH1mWpIULF+qdd95RUlKSZs+ereDgYP3617/WCy+8YDwiiDsHYQq3rVWrVvar+W528OBBHTt2TGvXrtXDDz9sX75mzRqHcrVr11ZWVpauXr1aZoHqwoULys3NdQhUp0+fliRZLBZJ10PMQw89pNGjRztdR2E5STpy5Iik69t7K6VZZ3p6uqTiz/MpXF/jxo21cOFCp/c3atTI4XZ8fLzatGljv33q1CmNGTPG6WO/+OIL/fWvf9WKFSuKvKHfTt3F6dGjh2bNmqXZs2crKytLvXv3LvJGXhhkVqxYoYCAAKdtuR2utrHQjz/+WOLL6f/617/q4MGD2rx58y3LFT7fCuXk5OjChQv250BgYKCysrKUk5Pj0LbCxwUGBjo8fsaMGQoLC1Nubq6+/PJLzZgxQz4+PoqLi9NPP/2kb775Rm+++aYGDBhgf8znn3/utG1du3bVuHHjHJZNnz7dadm8vDzNnDlTffv21SOPPOJ0CrR27dqaN2+e+vXrp7Zt22rkyJHatWuXw/lsZeHmETbp+v7y8fFxmBItzWNvPCZvlpycrGbNmmnu3Ln2Zbm5uUWCbt26dTV9+nRNnz5dP/zwgzZt2qQlS5aoTp06euqpp1xtFqoIwhTKRXZ2tiQ5BJqsrCylpqY6lOvUqZMSExO1YcOGMvvCPJvNpk8//dR+zpR0fbi+YcOG9hfPLl266Ouvv1arVq1chrhdu3apevXqDidSO3M76wwPD7/l+j799FPVqFHD4UT84tx7770On/CLO/fHZrNp5syZ6tmzpx599NEyqbs4Pj4+Gjx4sJYvX66RI0c6DTWdOnWSp6enfv75Z3Xq1Om267pRSbZRkr755hudOnXKfl7YrVy+fFlvvvmmhg0b5nKf/O1vf9PgwYPtt1NSUpSfn6927dpJktq3b693331XKSkpDufabNmyRT4+PkXOrbmxbx988EElJycrOTlZcXFxunr1qiTHYy03N1dbtmxx2rbAwMAiI0E1a9Z0ehL46tWrZbVatWrVqltu79y5c5WXl6cZM2YoICDgliM+t+v48eM6ePCgfd/YbDalpKSoTZs2LkeAdu/erStXrtin+jIzM/XPf/6z2A8+0vXp35tH4jZv3lxkZPVGzZs318SJE/Xhhx/a90Hhc76wn1A1EaZQLh588EH5+/trxowZGjdunK5cuaJly5apdu3aunjxor1cZGSkYmJiNHfuXB0/flyRkZHKy8vTgQMH9Nhjj7kMMM7UrFlTb731ls6dO6dmzZopOTlZe/fu1dy5c+1XB40bN05DhgzR0KFD9fTTT6tRo0a6cOGCjhw5ooyMDM2ZM0c5OTnasWOH1qxZo/bt2+vo0aP2On766SdJ0uHDh1W3bl35+/uXaJ0nTpzQqlWrtHbtWo0YMeKW52H169dPH3/8sYYPH66RI0fqF7/4hXJycpSRkaGdO3dq6dKlLs/jcubgwYPy8/PT8uXLK6TukSNHqkOHDmrdurXT+5s2barRo0dr5syZ+s9//qP27dvLz89Px48f1z/+8Q8NGTLE5QnGt7ON69ev1/Lly9WoUSOHE52Lk5qaqnr16rm80k+Sjh49qqlTp6pPnz5KT0/XggUL1L59e/t5QF27dtVDDz2k6dOn6+zZs2rVqpV2796tDRs26Pnnny8y0nL06FH5+fnZR6aOHDliH01r3ry5GjVqpAULFsjT01Pe3t567733XLaxJD788ENNnjxZ9evXL7ZMYbuTkpKcjiy6cvnyZafnLgUEBKh58+b22/Xq1dOECRM0duxY1alTR+vWrVN6erpef/11l3VUq1ZNI0eO1KhRo5STk6PFixfL39/fYfr0Zl26dNGOHTs0e/ZsPf744/rXv/6lNWvWOGzjxYsXNXz4cPXr10/NmzeXj4+PUlNTlZWVZf9gcM8998jb21sbN260T+3ee++99il/VA2EKZSLOnXq6J133tEf/vAHjRs3TvXr19czzzyjrKysIsP/8+fP15///Gdt2rRJq1evlr+/vyIiIuxfGVBa/v7+mj9/vmbNmqUjR46oXr16mjZtmgYOHGgv07BhQ23cuFFLlizR/Pnzde7cOQUFBalVq1aKi4uTdP2k3gkTJki6/oaxe/fuInWNHz9eq1evVocOHUq0zr179+rvf/+7pkyZ4nD1ojM+Pj569913lZiYqPXr1yszM1M1atRQkyZN9NhjjxU5L6ykbDabxowZowYNGlRI3YGBgbccHZKufz1G8+bN9cEHH+iDDz6Qh4eHQkJC1LFjR91zzz0lrquQq23My8vT4sWL1aVLF02YMKHYE8lvXuerr76qmjVruiw7bdo0+zk3NptN0dHRmjZtmv1+T09PJSYmav78+UpKStL58+fVqFEjTZ06Vc8++2yR9RV+Q7iPj48sFotGjBhhv3rO19dXS5cuVUJCgl599VUFBgZq0KBBatiwoV577TWXbb2V5s2b6+mnny72/vPnz2vatGn61a9+pc6dO99WHd9//71+9atfFVn+2GOPOZzn1rRpU40aNUoLFixQenq6GjVqpPnz55coaMfFxal69epKSEjQuXPnFBERoQULFtxyCvmJJ57Q8ePHtXHjRq1fv14RERFavny5Q5j28/OzX1jz888/y8PDQ/fee6/mzZtnvyK5du3a+v3vf68///nPGjZsmGw2m/01A1WHRwGXHqAKmTJlivbu3VvsicGlkZmZqW7duun7778vtkyPHj30xhtv8MIISbJ/SePKlStdBkiU3LBhw5SXl6d169ZVdlMAp7jUACiGr6+vw+XnzrRu3ZrhegC4yzHNBxSjfv36+stf/nLLMosXL66g1gAA3BXTfAAAAAaY5gMAADBAmAIAADBQaedM5efny2Yr3xlGLy+Pcq8DpUe/uB/6xD3RL+6HPnFPFdEvPj5Ff06pUKWFKZutQOfPXynXOoKCapR7HSg9+sX90CfuiX5xP/SJe6qIfgkOrlXsfUzzAQAAGCBMAQAAGCBMAQAAGCBMAQAAGCBMAQAAGCBMAQAAGCBMAQAAGCBMAQAAGCBMAQAAGCBMAQAAGCBMAQAAGCBMAQAAGCBMAQAAGPCu7AYAAACUlH9AdVX3c4wvV3NtldSa6whTAADgjlHdz1v3TEl2WJY+N1YXK6k9EtN8AAAARghTAAAABghTAAAABghTAAAABghTAAAABghTAAAABghTAAAABghTAAAABghTAAAABghTAAAABghTAAAABghTAAAABghTAAAABghTAAAABghTAAAABghTAAAABghTAAAABghTAAAABghTAAAABghTAAAABghTAAAABghTAAAABghTAAAABghTAAAABghTAAAABghTAAAABghTAAAABghTAAAABghTAAAABghTAAAABghTAAAABghTAAAABghTAAAABghTAAAABghTAAAABghTAAAABghTAAAABkoUptLS0hQTE6MePXooMTGx2HLbt29XaGio/vWvf5VZAwEAANyZyzBls9mUkJCgpKQkJScna+vWrTp69GiRcpcuXdLq1avVtm3bcmkoAACAO3IZpg4dOqRmzZqpSZMm8vX1VWxsrFJTU4uUW7RokUaPHi0/P79yaSgAAIA7chmmrFarQkJC7LctFousVqtDmW+//VYnTpzQY489VuYNBAAAcGfepivIz8/X3LlzNWfOnFI9zsvLQ0FBNUyrd1GHZ7nXgdKjX9wPfeKe6Bf3Q5+4r8rsF5dhymKx6MSJE/bbVqtVFovFfvvy5cs6cuSInnnmGUnSqVOnNGbMGC1btkwRERHFrtdmK9D581dM2u5SUFCNcq8DpUe/uB/6xD3RL+6HPql8wcG1nC4v734prl6pBGEqIiJC6enpysjIkMViUXJyst5++237/bVq1dL+/fvtt4cNG6bJkyffMkgBAABUFS7DlLe3t+Lj4zVq1CjZbDYNGjRIrVq10qJFixQeHq5u3bpVRDsBAADcUonOmYqKilJUVJTDsvHjxzst+/7775u3CgAA4A7BN6ADAAAYIEwBAAAYIEwBAAAYIEwBAAAYIEwBAAAYIEwBAAAYIEwBAAAYIEwBAAAYIEwBAAAYIEwBAAAYIEwBAAAYIEwBAAAYIEwBAAAYIEwBAAAYIEwBAAAYIEwBAAAYIEwBAAAYIEwBAAAYIEwBAAAYIEwBAAAYIEwBAAAYIEwBAAAYIEwBAAAYIEwBAAAYIEwBAAAYIEwBAAAYIEwBAAAYIEwBAAAYIEwBAAAYIEwBAAAYIEwBAAAYIEwBAAAYIEwBAAAYIEwBAAAYIEwBAAAYIEwBAAAYIEwBAAAYIEwBAAAYIEwBAAAYIEwBAAAYIEwBAAAYIEwBAAAYIEwBAAAYIEwBAAAYIEwBAAAYIEwBAAAYIEwBAAAYIEwBAAAYIEwBAAAYIEwBAAAYIEwBAAAYIEwBAAAYIEwBAAAYIEwBAAAYIEwBAAAYIEwBAAAYIEwBAAAYIEwBAAAYKFGYSktLU0xMjHr06KHExMQi969bt079+vXTgAED9OSTT+ro0aNl3lAAAAB35DJM2Ww2JSQkKCkpScnJydq6dWuRsNSvXz9t2bJFmzdv1qhRozRnzpxyazAAAIA7cRmmDh06pGbNmqlJkyby9fVVbGysUlNTHcr4+/vb/8/OzpaHh0fZtxQAAMANebsqYLVaFRISYr9tsVh06NChIuXWrl2rlStXKjc3V++9917ZthIAAMBNuQxTJTV06FANHTpUW7Zs0bJly/SHP/zhluW9vDwUFFSjrKovpg7Pcq8DpUe/uB/6xD3RL+6HPnFfldkvLsOUxWLRiRMn7LetVqssFkux5WNjY/X666+7rNhmK9D581dK1srbFBRUo9zrQOnRL+6HPnFP9Iv7oU8qX3BwLafLy7tfiqtXKsE5UxEREUpPT1dGRoZycnKUnJys6OhohzLp6en2/3ft2qVmzZrdfmsBAADuIC5Hpry9vRUfH69Ro0bJZrNp0KBBatWqlRYtWqTw8HB169ZNa9as0b59++Tt7a2AgACXU3wAAABVRYnOmYqKilJUVJTDsvHjx9v/f+2118q2VQAAAHcIvgEdAADAAGEKAADAAGEKAADAAGEKAADAAGEKAADAAGEKAADAAGEKAADAAGEKAADAAGEKAADAAGEKAADAAGEKAADAAGEKAADAAGEKAADAAGEKAADAAGEKAADAAGEKAADAAGEKAADAAGEKAADAAGEKAADAAGEKAADAAGEKAADAAGEKAADAAGEKAADAAGEKAADAAGEKAADAAGEKAADAAGEKAADAAGEKAADAAGEKAADAAGEKAADAAGEKAADAAGEKAADAAGEKAADAAGEKAADAAGEKAADAAGEKAADAAGEKAADAAGEKAADAAGEKAADAAGEKAADAAGEKAADAAGEKAADAAGEKAADAAGEKAADAAGEKAADAAGEKAADAAGEKAADAAGEKAADAAGEKAADAAGEKAADAAGEKAADAAGEKAADAAGEKAADAAGEKAADAAGEKAADAAGEKAADAAGEKAADAAGEKAADAQInCVFpammJiYtSjRw8lJiYWuX/lypXq06eP+vXrp2effVbHjh0r84YCAAC4I5dhymazKSEhQUlJSUpOTtbWrVt19OhRhzKtW7fWxo0btWXLFsXExOitt94qtwYDAAC4E5dh6tChQ2rWrJmaNGkiX19fxcbGKjU11aFMZGSkqlevLkl64IEHdOLEifJpLQAAgJtxGaasVqtCQkLsty0Wi6xWa7HlP/roI3Xt2rVsWgcAAODmvMtyZZs3b9Y333yjNWvWuCzr5eWhoKAaZVm9kzo8y70OlB794n7oE/dEv7gf+sR9VWa/uAxTFovFYdrOarXKYrEUKbd3714tX75ca9aska+vr8uKbbYCnT9/pZTNLZ2goBrlXgdKj35xP/SJe6Jf3A99UvmCg2s5XV7e/VJcvVIJpvkiIiKUnp6ujIwM5eTkKDk5WdHR0Q5lvvvuO8XHx2vZsmWqW7eueYsBAADuEC5Hpry9vRUfH69Ro0bJZrNp0KBBatWqlRYtWqTw8HB169ZNb775pq5cuaLx48dLkho0aKDly5eXe+MBAAAqW4nOmYqKilJUVJTDssLgJEmrVq0q00YBAADcKfgGdAAAAAOEKQAAAAOEKQAAAAOEKQAAAAOEKQAAAAOEKQAAAAOEKQAAAAOEKQAAAAOEKQAAAAOEKQAAAAOEKQAAAAOEKQAAAAOEKQAAAAOEKQAAAAOEKQAAAAOEKQAAAAOEKQAAAAOEKQAAAAOEKQAAAAOEKQAAAAOEKQAAAAOEKQAAAAOEKQAAAAOEKQAAAAOEKQAAAAOEKQAAAAOEKQAAAAOEKQAAAAOEKQAAAAOEKQAAAAOEKQAAAAOEKQAAAAOEKQAAAAOEKQAAAAOEKQAAAAOEKQAAAAOEKQAAAAOEKQAAAAOEKQAAAAOEKQAAAAOEKQAAAAOEKQAAAAOEKQAAAAOEKQAAAAOEKQAAAAOEKQAAAAOEKQAAAAPeld2A8mSTFBxcq8jy7Gt5unQhu+IbBAAAqpwqHaaq+XjpninJRZanz43VpUpoDwAAqHqY5gMAADBAmAIAADBAmAIAADBAmAIAADBAmAIAADBAmAIAADBAmAIAADBAmAIAADBAmAIAADBQojCVlpammJgY9ejRQ4mJiUXuP3DggAYOHKj7779fKSkpZd5IAAAAd+UyTNlsNiUkJCgpKUnJycnaunWrjh496lCmQYMGmjNnjvr27VtuDQUAAHBHLn+b79ChQ2rWrJmaNGkiSYqNjVVqaqpatmxpL9O4cWNJkqcns4YAAODu4jL9WK1WhYSE2G9bLBZZrdZybRQAAMCdwkJ4YXcAAApLSURBVOXIVHnx8vJQUFCNyqq+Uuu+23l5ebL/3Qx94p7oF/dDn7ivyuwXl2HKYrHoxIkT9ttWq1UWi8W4YputQOfPXzFez60EB9cq9r7yrhvFCwqqwf53M/SJe6Jf3A99UvmKe2+vzEzhcpovIiJC6enpysjIUE5OjpKTkxUdHV2mDQQAALhTuQxT3t7eio+P16hRo9SnTx/17t1brVq10qJFi5Samirp+knqXbt2VUpKiqZPn67Y2NhybzgAAIA7KNE5U1FRUYqKinJYNn78ePv/bdq0UVpaWtm2DAAA4A7AdxkAAAAYIEwBAAAYIEwBAAAYIEwBAAAYIEwBAAAYIEwBAAAYIEwBAAAYIEwBAAAYIEwBAAAYIEwBAAAYIEwBAAAYIEwBAAAYIEwBAAAYIEwBAAAYIEwBAAAYIEwBAAAYIEwBAAAYIEwBAAAYIEwBAAAYIEwBAAAYIEwBAAAYIEwBAAAYIEwBAAAYIEwBAAAYIEwBAAAYIEwBAAAYIEwBAAAYIEwBAAAYIEwBAAAYIEwBAAAY8K7sBgAA3It/QHVV93N8e8i+lqdLF7IrqUUlc6e2G3c+whQAwEF1P2/dMyXZYVn63FhdqqT2lNSd2m7c+ZjmAwAAMMDIFIAqhakelCebpODgWg7LeH6BMAWgSmGqB+Wpmo8Xzy8UwTQfAACAAcIUAACAAcIUAACAAc6ZAgCgAji7OELiBPaqgDAFAEAFcHZxhMQJ7FUB03wAAAAGCFMAAAAGCFMAAAAGOGcKAAA3wzf531kIUwAAuBm+yf/OwjQfAACAAcIUAACAAcIUAACAAcIUAACAAU5ABwDgDsVVf+6BMAUAwB2Kq/7cA2EKANyUTVJwcC2HZYw6AO6HMAVUEobn4Uo1Hy9GHYA7AGEKqCQMzwMoD1dzbYxoVjDCFACgUjA6Wz5KOqLJ/i87hCkAQKVgdLZysf/LDt8zBQAAYICRKVRJDF+XDleN4W7C6wPKWonCVFpammbNmqX8/HwNGTJEzz33nMP9OTk5mjx5sr799lsFBQVpwYIFaty4cbk0GCgJhq9Lh6vGyg9v3O6H1weUNZdhymazKSEhQStXrpTFYtHgwYMVHR2tli1b2sts2LBBAQEB+uyzz5ScnKx58+Zp4cKF5dpw3LkYBcHdpCLeuAlsQOVyGaYOHTqkZs2aqUmTJpKk2NhYpaamOoSpnTt36qWXXpIkxcTEKCEhQQUFBfLw8CinZqMilfULNaMgdy5nzwWJN+7KdieMtBT33KkKqtJXETjblqu5NlXz8XJYdqduX3lx+cy2Wq0KCQmx37ZYLDp06FCRMg0aNLi+Qm9v1apVS+fOnVOdOnXKuLm4kbMXp5I+6UvzplhZL9SV9Wn7Tg0MFbG/nD0XpKp92XVJjzOp5Ntn8ubr7LElLees3cVtS0mUZt9Icvo6crOyfjM32V8lrcPZB8T/ndmrRPWatrGkSvrY4j7slmT7Svr8qorhzKOgoKDgVgVSUlK0Z88ezZo1S5K0adMmHTp0SPHx8fYyffv2VVJSkj10de/eXX/5y18IUwAAoMpz+dUIFotFJ06csN+2Wq2yWCxFyhw/flySlJeXp4sXL6p27dpl3FQAAAD34zJMRUREKD09XRkZGcrJyVFycrKio6MdykRHR+uTTz6RJG3fvl2RkZGcLwUAAO4KLqf5JGn37t2aPXu2bDabBg0apDFjxmjRokUKDw9Xt27ddO3aNb3yyis6fPiwAgMDtWDBAvsJ6wAAAFVZicIUAAAAnOPnZAAAAAwQpgAAAAxU2TCVlpammJgY9ejRQ4mJiZXdnLvS8ePHNWzYMPXp00exsbF67733JEnnz5/XiBEj1LNnT40YMUJZWVmV3NK7k81mU1xcnJ5//nlJUkZGhoYMGaIePXro5ZdfVk5OTiW38O5y4cIFjRs3Tr169VLv3r319ddfc6y4gVWrVik2NlZ9+/bVxIkTde3aNY6VCjZ16lR17NhRffv2tS8r7tgoKCjQG2+8oR49eqhfv3769ttvK6SNVTJMFf4ETlJSkpKTk7V161YdPXq0spt11/Hy8tKUKVO0bds2rV+/Xh988IGOHj2qxMREdezYUZ9++qk6duxI2K0kq1evVosWLey3582bp+HDh+uzzz5TQECAPvroo0ps3d1n1qxZ6tKli1JSUrR582a1aNGCY6WSWa1WrV69Whs3btTWrVtls9nsP5nGsVJxfvnLXyopKclhWXHHRlpamtLT0/Xpp59q5syZev311yukjVUyTN34Ezi+vr72n8BBxapfv77CwsIkSf7+/mrevLmsVqtSU1MVFxcnSYqLi9OOHTsqs5l3pRMnTmjXrl0aPHiwpOuf5v77v/9bMTExkqSBAwdyzFSgixcv6sCBA/b+8PX1VUBAAMeKG7DZbLp69ary8vJ09epVBQcHc6xUsEceeUSBgYEOy4o7NgqXe3h46IEHHtCFCxd08uTJcm9jlQxTzn4Cx2q1VmKLkJmZqcOHD6tt27Y6c+aM6tevL0kKDg7WmTNnKrl1d5/Zs2frlVdekafn9ZeAc+fOKSAgQN7e138aJCQkhGOmAmVmZqpOnTqaOnWq4uLiNG3aNF25coVjpZJZLBaNHDlSjz/+uDp37ix/f3+FhYVxrLiB4o6Nm9//K6p/qmSYgnu5fPmyxo0bp9/97nfy9/d3uM/Dw4MveK1gn3/+uerUqaPw8PDKbgr+v7y8PH333Xd68skntWnTJlWvXr3IlB7HSsXLyspSamqqUlNTtWfPHmVnZ2vPnj2V3SzcxB2OjSr5E94l+QkcVIzc3FyNGzdO/fr1U8+ePSVJdevW1cmTJ1W/fn2dPHmS33CsYF999ZV27typtLQ0Xbt2TZcuXdKsWbN04cIF5eXlydvbWydOnOCYqUAhISEKCQlR27ZtJUm9evVSYmIix0ol27t3rxo3bmzf7z179tRXX33FseIGijs2bn7/r6j+qZIjUyX5CRyUv4KCAk2bNk3NmzfXiBEj7Mujo6O1adMmSdd/OLtbt26V1cS70qRJk5SWlqadO3dq/vz5ioyM1Ntvv60OHTpo+/btkqRPPvmEY6YCBQcHKyQkRD/88IMkad++fWrRogXHSiVr2LCh/vnPfyo7O1sFBQXat2+fWrZsybHiBoo7NgqXFxQU6ODBg6pVq5Z9OrA8VdlvQHf2EzioWF988YWGDh2q++67z35uzsSJE9WmTRu9/PLLOn78uBo2bKiFCxcqKCioklt7d9q/f79WrFihP/3pT8rIyNCECROUlZWl1q1ba968efL19a3sJt41Dh8+rGnTpik3N1dNmjTRnDlzlJ+fz7FSyRYvXqxt27bJ29tbrVu31qxZs2S1WjlWKtDEiRP1P//zPzp37pzq1q2rsWPHqnv37k6PjYKCAiUkJGjPnj2qXr26Zs+erYiIiHJvY5UNUwAAABWhSk7zAQAAVBTCFAAAgAHCFAAAgAHCFAAAgAHCFAAAgAHCFAAAgAHCFAAAgAHCFAAAgIH/B9OKH6WDvF0ZAAAAAElFTkSuQmCC\n",
            "text/plain": [
              "<Figure size 720x432 with 1 Axes>"
            ]
          },
          "metadata": {}
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "df3 = pd.read_csv(\"s_icm.deduplicated.bedGraph\", delimiter='\\t', skiprows=1, header=None)\n",
        "\n",
        "plt.figure(figsize=[10, 6])\n",
        "plt.title(\"Распределение метилирования ICM\", fontsize=16)\n",
        "plt.hist(df3[3], bins=100, density=True)\n",
        "plt.show()"
      ],
      "metadata": {
        "id": "wXLfnA1V5hMg",
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 392
        },
        "outputId": "5b930221-5cda-4404-c188-6397a0aafc1f"
      },
      "execution_count": 106,
      "outputs": [
        {
          "output_type": "display_data",
          "data": {
            "image/png": "iVBORw0KGgoAAAANSUhEUgAAAlMAAAF3CAYAAACBlM5VAAAABHNCSVQICAgIfAhkiAAAAAlwSFlzAAALEgAACxIB0t1+/AAAADh0RVh0U29mdHdhcmUAbWF0cGxvdGxpYiB2ZXJzaW9uMy4yLjIsIGh0dHA6Ly9tYXRwbG90bGliLm9yZy+WH4yJAAAgAElEQVR4nO3de1xVdb7/8TcXMW+AF9x4IUoljyNENqk4XigUSRDFUTunTFNHm+M06qgns7FDR0xx0sxLjg6H1LxkZpmmmDdMsfQ4dvFwupwcT+GgI1tTQVEM2PL7wx/70ZbLBr/A3urr+XjweLDX+u71/ez1ZW3erO9ibY+SkpISAQAA4JZ4uroAAACA2xlhCgAAwABhCgAAwABhCgAAwABhCgAAwABhCgAAwABhCi6xefNmdezY0f7VpUsXDRo0SOvWrVNxcbGrywNwmxg5cqSefPLJMssvXLig1157TXFxcXrooYcUHh6u+Ph4LViwQGfPnrW3i4qKUseOHTVt2rQKt9+xY8dy+wBKebu6ANzdFi9erMDAQOXn52vnzp2aPXu2zp8/r8mTJ7u6NAC3qRMnTmjs2LEqKSnRyJEjFRYWJkn65ptvtHHjRv3www9atmyZvX2jRo20d+9e5efnq3Hjxvblp0+f1tGjR9WoUaM6fw24vRCm4FKdOnVScHCwJKlXr146efKk1qxZQ5gCcEuKi4s1ceJE1a9fX++8846aN29uX9ejRw8988wzysjIcHhOz5499emnn2r37t369a9/bV++detWtWnTRq1atZLNZquz14DbD9N8cCthYWHKz8/X+fPnJUlpaWkaNWqUIiIi1KVLFyUkJOiDDz4o87zi4mKlpKQoNjZWYWFhioiI0G9+8xv93//9nyTpyJEjDtOKN39t2rTJvq0ZM2aoT58++uKLLzR06FCFhYUpKipKa9euLdNvdna2pk2bpoiICIWGhmrw4MHas2dPua+tT58+5fZ98uTJW95mRa/n5woKCjR//nxFRUUpNDRUUVFRWr58ua5fv25vU7p/Dh065PDcU6dOlbt/Ro4c6dDus88+U8eOHRUVFVXtvstT2m/Hjh115MgRh3Vffvllhftu48aNGjRokMLCwtS9e3f98Y9/VG5urtP9Vfq1efPmar/G0vaVbU+6MV00Y8YMh+dt2bJFHTt2dOirdCx27dqlGTNmqGvXrnr44Yc1bdo0Xbx40eH5+fn5SkpKUq9evRQaGqqYmBitXr1aP/9gi5t/9sPDwzV48GDt2LHDYVuffPKJxo8fr169eik8PFwDBw7UypUry4SIqKgo/du//VuZfXDzdFtpv6dOnbIvKyoq0oABA8qMa1FRkRYuXKh+/fopNDS00uPDmT179uj777/XtGnTHIJUKW9v7zJjWL9+fcXExGjr1q0Oy7du3arBgwfLw8OjWjXg7sOZKbiVU6dOycvLSw0bNpR0I1jExMTo2Weflaenp44ePaqXXnpJ165dc3jjnjp1qvbu3atnnnlGPXr0UGFhoY4ePapz586pffv29nYvvfSS/ZS/JJ07d06///3vy9SRn5+vKVOmaPz48br33nu1Y8cOvfLKK2rUqJH9L9czZ87oiSeeUPPmzfXiiy+qWbNm2rFjhyZOnKhly5apb9++ZbYbGRmp3/3ud5Kk/fv3a/ny5Q7rb2Wbw4YN0/DhwyVJmzZt0nvvvWdfV1xcbA+VEyZMUMeOHXXs2DH9+c9/Vl5eXplf7rfCZrNp1qxZ8vLyclheE337+/trw4YN6t69u33Zhg0b5O/v7xCSJGnBggVatWqVRo4cqenTp8tqtWrRokX629/+pnfeeUdeXl7auHGjvf2mTZt04MABvfHGG/Zl9957b7Ve488FBATYt1XRz9XP5efna8GCBRVuc+7cufrVr36l1157TSdPntTChQt19uxZe6i/fv26nn32WX3zzTeaNGmSHnjgAe3fv1/Jycm6cOGCpk6d6rC9xMREde7cWZcvX9bKlSs1bdo0dejQQQ888ICkG8dajx499PTTT6t+/fr66quvtHTpUl24cKHc8HQrVq9eXW44Sk1NVUpKip577jlFRESoXr165R4fVXHo0CF5eXkpMjKyWs9LSEjQ6NGjlZOTo8DAQB07dkxZWVlKSEjQ0aNHq10H7i6EKbiUzWZTcXGxrly5oo8++kh79uzRY489pgYNGkiS/vVf/9Xe9vr16+rWrZvOnTunDRs22MPU4cOHtWvXLs2cOVOjRo2yt+/Xr1+Z/tq3b6+HHnrI/vjnfzX/3JUrVzR79mzFxcVJunFWyWq1aunSpRoyZIg8PDy0dOlSlZSUaO3atWratKkkqXfv3srJydGSJUvKBJ+ioiK1aNHC3v/3339fpt/qbLOwsFCS1KZNG/s2Dx486LC97du36/PPP9e6devUtWtXSTemOiRp2bJlGj9+fLl/vVfHunXrdOrUKQ0ZMkSHDx+u0b4TEhK0fv16/fjjj2rRooUuXryonTt36qmnntKqVavs7U6dOqU333xTzz33nEOIue+++/TUU0/p448/Vr9+/RzG/uDBg/Lx8XFYVt3XWKqoqEgNGjSwb6uin6ufW7JkiTw9PdWvX78yZ5wkqUOHDkpOTrY/9vPz0/PPP6/Dhw+rR48eOnDggD7//HMlJyfbA36vXr1UUFCglStXavTo0WrWrJnD9krrCwgI0ODBg3X8+HF7mPr5HyclJSV65JFHVFRUpJUrV2rq1Kny9DSbyLBarfrzn/+sp556qsxZ3mPHjikkJEQTJ060Lyvv+KiKM2fOqFmzZvb3kKrq1q2bWrVqpQ8//FDPPvustmzZoi5dulQYsIGfY5oPLjVgwAB17txZ3bp106xZsxQfH6+5c+fa12dlZWnq1Knq3bu3OnfurM6dO2vTpk364Ycf7G0+/fRTeXh46Iknnqixury8vNS/f3+HZbGxsfrHP/4hq9Uq6cYv48jISDVp0kTFxcX2r169eul///d/lZ+f7/D8goICp2/w1dnmTz/9JOnGFEVl22vTpo26dOnisL2ePXuqqKhIx44dc2h//fp1h3bOpuN+/PFHLV26VBMmTFBgYKBR3+Xp0KGDunTpYp9m3Lx5s37xi1/YA0CpQ4cO6fr16xo0aJBDX+Hh4WrUqJHRmYXKXmOpa9euVToON/vuu++0fv16zZgxw34W9mYDBgxwePz444/L09NTX375pSTp6NGj8vT01MCBAx3aDRo0qNKxzc3N1aZNm1S/fn2Fh4fb1589e1aJiYl67LHHFBoaqs6dO2vRokW6dOmSfdq9VElJicN+Li4udphaLE9ycrLatm1bZvpUkoKCgpSVlaWPP/5YBQUFVfrZq2keHh6Kj4/X1q1bVVhYqI8++kgJCQl1WgNuX5yZgkstW7ZMFotFjRo1Ups2bRx+IV25ckVjx47VPffco2nTpunee+9VvXr1tGHDBr3//vv2drm5ufLz89M999xTY3X5+vqqXr16DstatGgh6cZf2IGBgbpw4YK2bNmiLVu2lLuNixcv2v8z6NKlSyooKFBAQECl/VZnm6XTXP7+/pVu7/Tp0+rcuXO562+eKvvNb35TaX03e/XVV9WiRQuNHj1af/nLX4z6rsiTTz6p+fPna/z48dq4caOee+65MtfxlP6yj46ONuqrPJW9xp9vv7JxuNns2bP1yCOPKDY2tszF0KVKf95K+fj4yNfX1x7m8/Ly5OfnJx8fn3Kfl5eX57B89OjR9u/vuece/fu//7uCgoIk3QhaEyZM0NmzZzVx4kS1a9dO9evX1969e7VixQp7cC+1fft2bd++vUzNDz/8cLmv5fDhw9q5c6fWrl1b7rTmc889p9OnT+t3v/udcYhq1aqVDh06VKU/Xm6WkJCgFStWaNmyZbp69apiY2ONasHdgzAFlwoJCbH/N9/Njh07ptOnT2v9+vV65JFH7MvXrVvn0K5p06bKy8vTtWvXaixQXbp0SUVFRQ6B6scff5QkWSwWSTdCzC9/+UuNHz++3G2UtpOk48ePS7rxeitTnW1mZWVJqvg6n9LttW3bVosWLSp3fZs2bRweJyYm6sEHH7Q/PnfunCZMmFDucz/77DN9+OGHWrlyZZlf6LfSd0Wio6M1Z84czZ07V3l5eRowYECZX+SlQWblypXy9fUtt5Zb4ew1ljp58qR69+5dpW1++OGHOnbsWJmLnW9W+vNWqrCwUJcuXbL/DPj5+SkvL0+FhYUOtZU+z8/Pz+H5s2bNUufOnVVUVKTPP/9cs2bNUr169ZSQkKC///3v+uqrr/Tqq69q8ODB9ud8/PHH5dbWp08fTZo0yWHZyy+/XG7b4uJizZ49WwMHDlTXrl3LnQJt2rSpFixYoPj4eIWHh2vs2LHav3+/w/VsVdWjRw+9++67ysjIUExMTLWee//99ys8PFwpKSmKjo4u92cJKA9hCm6roKBAkhwCTV5entLT0x3a9ezZUykpKdq0aVO5Uwi3wmazaffu3fZrpiRpx44dat26tf2XWe/evfXll18qJCTEaYjbv3+/GjRo4HAhdXluZZuhoaGVbm/37t1q2LChw4X4Fbn//vsdLtCv6Nofm82m2bNnq3///vrVr35VI31XpF69eho2bJhWrFihsWPHlhtqevbsKU9PT/3jH/9Qz549b7mvn6vKa5Skr776SufOnbNfF1aZK1eu6NVXX9XIkSOd7pOPPvpIw4YNsz/euXOnrl+/ri5duki6cY3Pm2++qZ07d2rQoEH2dtu2bVO9evXKXAv287F9+OGHlZaWprS0NCUkJOjatWuSHI+1oqIibdu2rdza/Pz8HH5OpBv3airvhrtr1qyR1WrV6tWrK3298+bNU3FxsWbNmiVfX1/97W9/q7R9Rfr376/7779fCxYsUNeuXR2uG5NuhLtPPvlEjz76aLnPHzdunLZs2aKnn376lvrH3YkwBbf18MMPq3Hjxpo1a5YmTZqkq1evavny5WratKkuX75sbxcREaGYmBjNmzdPZ86cUUREhIqLi3X06FE9+uijTgNMeRo1aqT58+fr4sWLCg4OVlpamg4dOqR58+bZ/0160qRJGj58uEaMGKGnn35abdq00aVLl3T8+HFlZ2crOTlZhYWF2rt3r9atW6du3brpxIkT9j7+/ve/S5K+/fZbNW/eXI0bN67SNnNycrR69WqtX79eY8aMqXQqIz4+Xps3b9bo0aM1duxY/dM//ZMKCwuVnZ2tffv2admyZdWeCpFunDWsX7++VqxYUSd9jx07Vt27d1enTp3KXX/vvfdq/Pjxmj17tn744Qd169ZN9evX15kzZ/Tpp59q+PDhioiIqPHXuHHjRq1YsUJt2rTR448/7nSb6enpatGihdP/9JNu3HjyxRdfVGxsrLKysvT666+rW7du9ov4+/Tpo1/+8pd6+eWXdeHCBYWEhOjAgQPatGmTfvvb35YJESdOnFD9+vXtZ6aOHz9uP5vWrl07tWnTRq+//ro8PT3l7e2tt956y2mNVfHOO+9o+vTpatmyZYVtSutOTU01Phvk7e2tN954Q2PGjNHgwYM1atQo+x8c3333nTZu3Kh27dpVGKb69+9f5npJwBnCFNxWs2bN9MYbb+hPf/qTJk2apJYtW2rUqFHKy8src/p/4cKF+s///E9t2bJFa9asUePGjRUWFma/ZUB1NW7cWAsXLtScOXN0/PhxtWjRQjNnztSQIUPsbVq3bq33339fS5cu1cKFC3Xx4kX5+/srJCTEfuHq2bNnNWXKFEk3fmEcOHCgTF+TJ0/WmjVr1L179ypt89ChQ/rkk080Y8YMh/9eLE+9evX05ptvKiUlRRs3btSpU6fUsGFDBQUF6dFHHy1zXVhV2Ww2TZgwQa1ataqTvv38/Co9OyTduD1Gu3bt9Pbbb+vtt9+Wh4eHAgMD1aNHD913331V7quUs9dYXFysJUuWqHfv3poyZUqFF5LfvM0XXnihSnfUnjlzpvbt26cpU6bIZrMpKipKM2fOtK/39PRUSkqKFi5cqNTUVOXm5qpNmzZ68cUX9cwzz5TZXlJSkqQb42KxWDRmzBj7f8/5+Pho2bJlSkpK0gsvvCA/Pz8NHTpUrVu31ksvveS01sq0a9eu0rM8ubm5mjlzpv75n/9ZvXr1MuqrVIcOHbR161atXLlSH3zwgd544w2VlJQoODhY/fv3d3rcANXlUeLsXzCAu8yMGTN06NChCi8Mro5Tp06pb9+++u677ypsEx0drVdeeeWWzqDhznPkyBGNGjVKq1atchogAbgHbo0A1CIfHx+Hfz8vT6dOnRw+DwwAcHthmg+oRS1bttS7775baZslS5bUUTUAgNrANB8AAIABpvkAAAAMEKYAAAAMuOyaqevXr8tmq90ZRi8vj1rvA9XHuLgfxsQ9MS7uhzFxT3UxLvXqlf0opFIuC1M2W4lyc6/Wah/+/g1rvQ9UH+PifhgT98S4uB/GxD3VxbgEBDSpcB3TfAAAAAYIUwAAAAYIUwAAAAYIUwAAAAYIUwAAAAYIUwAAAAYIUwAAAAYIUwAAAAYIUwAAAAYIUwAAAAYIUwAAAAYIUwAAAAYIUwAAAAa8XV1AbbKp/E95LvipWPmXCuq+IAAAcMe5o8PUPfW8dN+MtDLLs+bFKd8F9QAAgDsP03wAAAAGCFMAAAAGCFMAAAAGCFMAAAAGCFMAAAAGCFMAAAAGqhSmMjIyFBMTo+joaKWkpJTbZseOHYqNjVVcXJymTZtWo0UCAAC4K6f3mbLZbEpKStKqVatksVg0bNgwRUVFqUOHDvY2WVlZSklJ0YYNG+Tn56fz58/XatEAAADuwumZqczMTAUHBysoKEg+Pj6Ki4tTenq6Q5t3331XI0aMkJ+fnySpefPmtVMtAACAm3EapqxWqwIDA+2PLRaLrFarQ5usrCz98MMP+pd/+Rc98cQTysjIqPlKAQAA3FCNfJyMzWbTyZMntXbtWuXk5Ojpp5/Wtm3b5OvrW+FzvLw85O/fsCa6vyWu7Ptu5+Xlyf53M4yJe2Jc3A9j4p5cPS5Ow5TFYlFOTo79sdVqlcViKdMmPDxc9erVU1BQkO677z5lZWXpwQcfrHC7NluJcnOvGpTuXHkfclyqtvtGxfz9G7L/3Qxj4p4YF/fDmLinuhiXyjKF02m+sLAwZWVlKTs7W4WFhUpLS1NUVJRDm379+umvf/2rJOnChQvKyspSUFCQYdkAAADuz+mZKW9vbyUmJmrcuHGy2WwaOnSoQkJCtHjxYoWGhqpv377q3bu3Pv30U8XGxsrLy0vTp09X06ZN66J+AAAAl/IoKSkpcUXHRUW2Ojkld9+MtDLLs+bF6dy5y7XaNyrGaXL3w5i4J8bF/TAm7sntp/kAAABQMcIUAACAAcIUAACAAcIUAACAAcIUAACAAcIUAACAAcIUAACAAcIUAACAAcIUAACAAcIUAACAAcIUAACAAcIUAACAAcIUAACAAcIUAACAAcIUAACAAcIUAACAAcIUAACAAcIUAACAAcIUAACAAcIUAACAAcIUAACAAcIUAACAAcIUAACAAcIUAACAAcIUAACAAcIUAACAAcIUAACAAcIUAACAAcIUAACAAcIUAACAAcIUAACAAcIUAACAAcIUAACAAcIUAACAAcIUAACAAcIUAACAAcIUAACAAcIUAACAAcIUAACAAcIUAACAAcIUAACAgSqFqYyMDMXExCg6OlopKSll1m/evFkREREaPHiwBg8erE2bNtV4oQAAAO7I21kDm82mpKQkrVq1ShaLRcOGDVNUVJQ6dOjg0C42NlaJiYm1VigAAIA7cnpmKjMzU8HBwQoKCpKPj4/i4uKUnp5eF7UBAAC4Padhymq1KjAw0P7YYrHIarWWabd7927Fx8dr0qRJOnPmTM1WCQAA4KacTvNVxWOPPaaBAwfKx8dH77zzjl544QWtWbOm0ud4eXnI379hTXR/S1zZ993Oy8uT/e9mGBP3xLi4H8bEPbl6XJyGKYvFopycHPtjq9Uqi8Xi0KZp06b274cPH6758+c77dhmK1Fu7tXq1FptAQFNKlxX232jYv7+Ddn/boYxcU+Mi/thTNxTXYxLZZnC6TRfWFiYsrKylJ2drcLCQqWlpSkqKsqhzdmzZ+3f79u3T+3btzcoFwAA4Pbh9MyUt7e3EhMTNW7cONlsNg0dOlQhISFavHixQkND1bdvX61du1b79u2Tl5eX/Pz8lJycXBe1AwAAuJxHSUlJiSs6Liqy1ckpuftmpJVZnjUvTufOXa7VvlExTpO7H8bEPTEu7ocxcU9uP80HAACAihGmAAAADBCmAAAADBCmAAAADBCmAAAADBCmAAAADBCmAAAADBCmAAAADBCmAAAADBCmAAAADBCmAAAADBCmAAAADBCmAAAADBCmAAAADBCmAAAADBCmAAAADBCmAAAADBCmAAAADBCmAAAADBCmAAAADBCmAAAADBCmAAAADBCmAAAADBCmAAAADBCmAAAADBCmAAAADBCmAAAADBCmAAAADBCmAAAADBCmAAAADBCmAAAADBCmAAAADBCmAAAADBCmAAAADBCmAAAADBCmAAAADBCmAAAADBCmAAAADBCmAAAADBCmAAAADBCmAAAADFQpTGVkZCgmJkbR0dFKSUmpsN2uXbvUsWNH/c///E+NFQgAAODOnIYpm82mpKQkpaamKi0tTdu3b9eJEyfKtMvPz9eaNWsUHh5eK4UCAAC4I6dhKjMzU8HBwQoKCpKPj4/i4uKUnp5ept3ixYs1fvx41a9fv1YKBQAAcEdOw5TValVgYKD9scVikdVqdWjz9ddfKycnR48++miNFwgAAODOvE03cP36dc2bN0/JycnVep6Xl4f8/Ruadn/LXNn33c7Ly5P972YYE/fEuLgfxsQ9uXpcnIYpi8WinJwc+2Or1SqLxWJ/fOXKFR0/flyjRo2SJJ07d04TJkzQ8uXLFRYWVuF2bbYS5eZeNandqYCAJhWuq+2+UTF//4bsfzfDmLgnxsX9MCbuqS7GpbJM4TRMhYWFKSsrS9nZ2bJYLEpLS9Nrr71mX9+kSRMdOXLE/njkyJGaPn16pUEKAADgTuE0THl7eysxMVHjxo2TzWbT0KFDFRISosWLFys0NFR9+/atizoBAADcUpWumYqMjFRkZKTDssmTJ5fbdu3ateZVAQAA3Ca4AzoAAIABwhQAAIABwhQAAIABwhQAAIABwhQAAIABwhQAAIABwhQAAIABwhQAAIABwhQAAIABwhQAAIABwhQAAIABwhQAAIABwhQAAIABwhQAAIABwhQAAIABwhQAAIABwhQAAIABwhQAAIABwhQAAIABwhQAAIABwhQAAIABwhQAAIABwhQAAIABwhQAAIABwhQAAIABwhQAAIABwhQAAIABwhQAAIABwhQAAIABwhQAAIABwhQAAIABwhQAAIABwhQAAIABwhQAAIABwhQAAIABwhQAAIABwhQAAIABwhQAAIABwhQAAIABwhQAAIABwhQAAICBKoWpjIwMxcTEKDo6WikpKWXWb9iwQfHx8Ro8eLCefPJJnThxosYLBQAAcEdOw5TNZlNSUpJSU1OVlpam7du3lwlL8fHx2rZtm7Zu3apx48YpOTm51goGAABwJ07DVGZmpoKDgxUUFCQfHx/FxcUpPT3doU3jxo3t3xcUFMjDw6PmKwUAAHBD3s4aWK1WBQYG2h9bLBZlZmaWabd+/XqtWrVKRUVFeuutt5x27OXlIX//htUst+a4su+7nZeXJ/vfzTAm7olxcT+MiXty9bg4DVNVNWLECI0YMULbtm3T8uXL9ac//anS9jZbiXJzr9ZU9+UKCGhS4bra7hsV8/dvyP53M4yJe2Jc3A9j4p7qYlwqyxROp/ksFotycnLsj61WqywWS4Xt4+LitHfv3mqWCAAAcHtyGqbCwsKUlZWl7OxsFRYWKi0tTVFRUQ5tsrKy7N/v379fwcHBNV4oAACAO3I6zeft7a3ExESNGzdONptNQ4cOVUhIiBYvXqzQ0FD17dtX69at0+HDh+Xt7S1fX1+nU3wAAAB3iipdMxUZGanIyEiHZZMnT7Z//9JLL9VsVQAAALcJ7oAOAABggDAFAABggDAFAABggDAFAABggDAFAABggDAFAABggDAFAABggDAFAABggDAFAABggDAFAABggDAFAABggDAFAABggDAFAABggDAFAABggDAFAABggDAFAABggDAFAABggDAFAABggDAFAABggDAFAABggDAFAABggDAFAABggDAFAABggDAFAABggDAFAABggDAFAABggDAFAABggDAFAABggDAFAABggDAFAABggDAFAABggDAFAABggDAFAABggDAFAABggDAFAABggDAFAABggDAFAABggDAFAABggDAFAABggDAFAABggDAFAABgoEphKiMjQzExMYqOjlZKSkqZ9atWrVJsbKzi4+P1zDPP6PTp0zVeKAAAgDtyGqZsNpuSkpKUmpqqtLQ0bd++XSdOnHBo06lTJ73//vvatm2bYmJiNH/+/ForGAAAwJ04DVOZmZkKDg5WUFCQfHx8FBcXp/T0dIc2ERERatCggSTpoYceUk5OTu1UCwAA4Ga8nTWwWq0KDAy0P7ZYLMrMzKyw/Xvvvac+ffo47djLy0P+/g2rWGbNc2XfdzsvL0/2v5thTNwT4+J+GBP35OpxcRqmqmPr1q366quvtG7dOqdtbbYS5eZercnuywgIaFLhutruGxXz92/I/nczjIl7YlzcD2PinupiXCrLFE7DlMVicZi2s1qtslgsZdodOnRIK1as0Lp16+Tj43OLpQIAANxenF4zFRYWpqysLGVnZ6uwsFBpaWmKiopyaPPNN98oMTFRy5cvV/PmzWutWAAAAHfj9MyUt7e3EhMTNW7cONlsNg0dOlQhISFavHixQkND1bdvX7366qu6evWqJk+eLElq1aqVVqxYUevFAwAAuFqVrpmKjIxUZGSkw7LS4CRJq1evrtGiAAAAbhfcAR0AAMAAYQoAAMAAYQoAAMAAYQoAAMAAYQoAAMAAYQoAAMAAYQoAAMAAYQoAAMAAYQoAAMAAYQoAAMAAYQoAAB2PsPQAAAq9SURBVMAAYQoAAMAAYQoAAMAAYQoAAMAAYQoAAMAAYQoAAMAAYQoAAMAAYQoAAMAAYQoAAMAAYQoAAMAAYQoAAMAAYQoAAMAAYQoAAMAAYQoAAMAAYQoAAMAAYQoAAMAAYQoAAMAAYQoAAMAAYQoAAMAAYQoAAMAAYQoAAMAAYQoAAMAAYQoAAMAAYQoAAMAAYQoAAMAAYQoAAMAAYQoAAMAAYQoAAMAAYQoAAMAAYQoAAMAAYQoAAMBAlcJURkaGYmJiFB0drZSUlDLrjx49qiFDhugXv/iFdu7cWeNFAgAAuCunYcpmsykpKUmpqalKS0vT9u3bdeLECYc2rVq1UnJysgYOHFhrhQIAALgjb2cNMjMzFRwcrKCgIElSXFyc0tPT1aFDB3ubtm3bSpI8PZk1BAAAdxenYcpqtSowMND+2GKxKDMz07hjLy8P+fs3NN7OrXJl33c7Ly9P9r+bYUzcE+PifhgT9+TqcXEapmqLzVai3NyrtdpHQECTCtfVdt+omL9/Q/a/m2FM3BPj4n4YE/dUF+NSWaZwOi9nsViUk5Njf2y1WmWxWGqmMgAAgNuc0zAVFhamrKwsZWdnq7CwUGlpaYqKiqqL2gAAANye0zDl7e2txMREjRs3TrGxsRowYIBCQkK0ePFipaenS7pxkXqfPn20c+dOvfzyy4qLi6v1wgEAANxBla6ZioyMVGRkpMOyyZMn279/8MEHlZGRUbOVAQAA3Aa4lwEAAIABwhQAAIABwhQAAIABl91nCgBqQ2PfBmpQ3/GtreCnYuVfKnBRRQDudIQpAHeUBvW9dd+MNIdlWfPilO+iegDc+ZjmAwAAMECYAgAAMECYAgAAMECYAgAAMECYAgAAMECYAgAAMECYAgAAMECYAgAAMMBNOwEX4U7dAHBnIEwBLsKdugHgzkCYAgAAt43yzupfK7K5qJobCFMAAOC2UdFZ/csuqkfiAnQAAAAjhCkAAAADhCkAAAADhCkAAAADhCkAAAADhCkAAAADhCkAAAAD3GcKdyQ+qgUAUFcIU7gj8VEtAIC6wjQfAACAAcIUAACAAcIUAACAAcIUAACAAcIUAACAAcIUAACAAW6NAABwwH3agOohTKHO2SQFBDRxWMYbNeA+uE8bUD2EKdS5e+p58UYNALhjcM0UAACAAcIUAACAAab5AHAdGwAYIEzBKf6z587HdWwAcOsIU3CK/+wBAKBiVQpTGRkZmjNnjq5fv67hw4fr2WefdVhfWFio6dOn6+uvv5a/v79ef/11tW3btlYKBoC7BdOv1cNZdLiK0zBls9mUlJSkVatWyWKxaNiwYYqKilKHDh3sbTZt2iRfX1/t2bNHaWlpWrBggRYtWlSrhcNMeW86Em88gDth+rV6OIsOV3EapjIzMxUcHKygoCBJUlxcnNLT0x3C1L59+/T73/9ekhQTE6OkpCSVlJTIw8OjlsqGqfLedCTeeCSCZmXYN7jbcbYQ5XEapqxWqwIDA+2PLRaLMjMzy7Rp1arVjQ16e6tJkya6ePGimjVrVsPl3r3K+yV2rcime+p5OV1WnQP9WpGtzBuFSY1V7bu8fmvj9VWlZklVDpruNK1QF7VUJ4S7075xpbrYD7frvr4d6zY5W8gfI3cuj5KSkpLKGuzcuVMHDx7UnDlzJElbtmxRZmamEhMT7W0GDhyo1NRUe+jq16+f3n33XcIUAAC44zm9aafFYlFOTo79sdVqlcViKdPmzJkzkqTi4mJdvnxZTZs2reFSAQAA3I/TMBUWFqasrCxlZ2ersLBQaWlpioqKcmgTFRWlDz74QJK0a9cuRUREcL0UAAC4Kzid5pOkAwcOaO7cubLZbBo6dKgmTJigxYsXKzQ0VH379tVPP/2k559/Xt9++638/Pz0+uuv2y9YBwAAuJNVKUwBAACgfHzQMQAAgAHCFAAAgIE7NkxlZGQoJiZG0dHRSklJcXU5d6UzZ85o5MiRio2NVVxcnN566y1JUm5ursaMGaP+/ftrzJgxysvLc3GldyebzaaEhAT99re/lSRlZ2dr+PDhio6O1h/+8AcVFha6uMK7y6VLlzRp0iQ9/vjjGjBggL788kuOFTewevVqxcXFaeDAgZo6dap++uknjpU69uKLL6pHjx4aOHCgfVlFx0ZJSYleeeUVRUdHKz4+Xl9//XWd1HhHhqnSj8BJTU1VWlqatm/frhMnTri6rLuOl5eXZsyYoR07dmjjxo16++23deLECaWkpKhHjx7avXu3evToQdh1kTVr1qh9+/b2xwsWLNDo0aO1Z88e+fr66r333nNhdXefOXPmqHfv3tq5c6e2bt2q9u3bc6y4mNVq1Zo1a/T+++9r+/btstls9o9M41ipO7/+9a+VmprqsKyiYyMjI0NZWVnavXu3Zs+erf/4j/+okxrvyDD184/A8fHxsX8EDupWy5Yt1blzZ0lS48aN1a5dO1mtVqWnpyshIUGSlJCQoL1797qyzLtSTk6O9u/fr2HDhkm68dfcf/3XfykmJkaSNGTIEI6ZOnT58mUdPXrUPh4+Pj7y9fXlWHEDNptN165dU3Fxsa5du6aAgACOlTrWtWtX+fn5OSyr6NgoXe7h4aGHHnpIly5d0tmzZ2u9xjsyTJX3EThWq9WFFeHUqVP69ttvFR4ervPnz6tly5aSpICAAJ0/f97F1d195s6dq+eff16enjfeAi5evChfX195e9/4qIvAwECOmTp06tQpNWvWTC+++KISEhI0c+ZMXb16lWPFxSwWi8aOHavHHntMvXr1UuPGjdW5c2eOFTdQ0bFx8+//uhqfOzJMwb1cuXJFkyZN0h//+Ec1btzYYZ2Hhwc3eK1jH3/8sZo1a6bQ0FBXl4L/r7i4WN98842efPJJbdmyRQ0aNCgzpcexUvfy8vKUnp6u9PR0HTx4UAUFBTp48KCry8JN3OHYcPpBx7ejqnwEDupGUVGRJk2apPj4ePXv31+S1Lx5c509e1YtW7bU2bNn+QzHOvbFF19o3759ysjI0E8//aT8/HzNmTNHly5dUnFxsby9vZWTk8MxU4cCAwMVGBio8PBwSdLjjz+ulJQUjhUXO3TokNq2bWvf7/3799cXX3zBseIGKjo2bv79X1fjc0eemarKR+Cg9pWUlGjmzJlq166dxowZY18eFRWlLVu2SLrxwdl9+/Z1VYl3pWnTpikjI0P79u3TwoULFRERoddee03du3fXrl27JEkffPABx0wdCggIUGBgoL7//ntJ0uHDh9W+fXuOFRdr3bq1/vu//1sFBQUqKSnR4cOH1aFDB44VN1DRsVG6vKSkRMeOHVOTJk3s04G16Y69A3p5H4GDuvXZZ59pxIgReuCBB+zX5kydOlUPPvig/vCHP+jMmTNq3bq1Fi1aJH9/fxdXe3c6cuSIVq5cqb/85S/Kzs7WlClTlJeXp06dOmnBggXy8fFxdYl3jW+//VYzZ85UUVGRgoKClJycrOvXr3OsuNiSJUu0Y8cOeXt7q1OnTpozZ46sVivHSh2aOnWq/vrXv+rixYtq3ry5Jk6cqH79+pV7bJSUlCgpKUkHDx5UgwYNNHfuXIWFhdV6jXdsmAIAAKgLd+Q0HwAAQF0hTAEAABggTAEAABggTAEAABggTAEAABggTAEAABggTAEAABggTAEAABj4f9iM1c6m2OctAAAAAElFTkSuQmCC\n",
            "text/plain": [
              "<Figure size 720x432 with 1 Axes>"
            ]
          },
          "metadata": {}
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!gzip -d s_8_cell.deduplicated.bismark.cov.gz"
      ],
      "metadata": {
        "id": "DdI3swqTWn82"
      },
      "execution_count": 107,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "!gzip -d s_icm.deduplicated.bismark.cov.gz"
      ],
      "metadata": {
        "id": "jvzfNnc5Womj"
      },
      "execution_count": 108,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "!gzip -d s_epiblast.deduplicated.bismark.cov.gz"
      ],
      "metadata": {
        "id": "cZBg9ZSrWpJe"
      },
      "execution_count": 109,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "! head s_8_cell.deduplicated.bismark.cov"
      ],
      "metadata": {
        "id": "McDe2X10t62u",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "e47d9729-f348-4cd3-c1b6-a3d47b7eef55"
      },
      "execution_count": 110,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "chr11\t3100031\t3100031\t50\t1\t1\n",
            "chr11\t3100048\t3100048\t66.6666666666667\t2\t1\n",
            "chr11\t3100049\t3100049\t50\t1\t1\n",
            "chr11\t3100224\t3100224\t87.5\t7\t1\n",
            "chr11\t3100225\t3100225\t53.3333333333333\t8\t7\n",
            "chr11\t3100286\t3100286\t66.6666666666667\t6\t3\n",
            "chr11\t3100287\t3100287\t33.3333333333333\t5\t10\n",
            "chr11\t3100310\t3100310\t62.5\t5\t3\n",
            "chr11\t3100311\t3100311\t50\t7\t7\n",
            "chr11\t3100349\t3100349\t55.5555555555556\t5\t4\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "! head s_icm.deduplicated.bismark.cov"
      ],
      "metadata": {
        "id": "5awuDrhwuCuX",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "1ac02d21-c6df-4f8f-b533-f28b4daa5438"
      },
      "execution_count": 111,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "chr11\t3100031\t3100031\t33.3333333333333\t1\t2\n",
            "chr11\t3100032\t3100032\t50\t3\t3\n",
            "chr11\t3100048\t3100048\t50\t2\t2\n",
            "chr11\t3100049\t3100049\t57.1428571428571\t4\t3\n",
            "chr11\t3100224\t3100224\t36.3636363636364\t4\t7\n",
            "chr11\t3100225\t3100225\t22.7272727272727\t5\t17\n",
            "chr11\t3100286\t3100286\t25\t2\t6\n",
            "chr11\t3100287\t3100287\t28.5714285714286\t6\t15\n",
            "chr11\t3100310\t3100310\t22.2222222222222\t2\t7\n",
            "chr11\t3100311\t3100311\t41.1764705882353\t7\t10\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "! head s_epiblast.deduplicated.bismark.cov"
      ],
      "metadata": {
        "id": "tBsyzc5XuCzk",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "409cabde-b7ef-48cd-f793-30a7eca95394"
      },
      "execution_count": 112,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "chr11\t3100031\t3100031\t75\t3\t1\n",
            "chr11\t3100032\t3100032\t100\t3\t0\n",
            "chr11\t3100048\t3100048\t100\t5\t0\n",
            "chr11\t3100049\t3100049\t100\t8\t0\n",
            "chr11\t3100224\t3100224\t96.875\t31\t1\n",
            "chr11\t3100225\t3100225\t93.75\t30\t2\n",
            "chr11\t3100286\t3100286\t93.3333333333333\t28\t2\n",
            "chr11\t3100287\t3100287\t92.3076923076923\t36\t3\n",
            "chr11\t3100310\t3100310\t96.7741935483871\t30\t1\n",
            "chr11\t3100311\t3100311\t97.5\t39\t1\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "! pip install pyGenomeTracks"
      ],
      "metadata": {
        "id": "YqqVmTlGuhNk",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "70b925f6-632a-4e45-c54d-2f8e4386ba89"
      },
      "execution_count": 113,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Requirement already satisfied: pyGenomeTracks in /usr/local/lib/python3.7/site-packages (3.6)\n",
            "Requirement already satisfied: numpy>=1.16 in /usr/local/lib/python3.7/site-packages (from pyGenomeTracks) (1.21.5)\n",
            "Requirement already satisfied: pysam>=0.14 in /usr/local/lib/python3.7/site-packages (from pyGenomeTracks) (0.18.0)\n",
            "Requirement already satisfied: pyBigWig>=0.3.16 in /usr/local/lib/python3.7/site-packages (from pyGenomeTracks) (0.3.18)\n",
            "Requirement already satisfied: hicmatrix>=15 in /usr/local/lib/python3.7/site-packages (from pyGenomeTracks) (15)\n",
            "Requirement already satisfied: gffutils>=0.9 in /usr/local/lib/python3.7/site-packages (from pyGenomeTracks) (0.10.1)\n",
            "Requirement already satisfied: pybedtools>=0.8.1 in /usr/local/lib/python3.7/site-packages (from pyGenomeTracks) (0.9.0)\n",
            "Requirement already satisfied: intervaltree>=2.1.0 in /usr/local/lib/python3.7/site-packages (from pyGenomeTracks) (3.1.0)\n",
            "Requirement already satisfied: matplotlib<=3.3.2,>=3.1.1 in /usr/local/lib/python3.7/site-packages (from pyGenomeTracks) (3.3.2)\n",
            "Requirement already satisfied: pytest in /usr/local/lib/python3.7/site-packages (from pyGenomeTracks) (7.0.1)\n",
            "Requirement already satisfied: future>=0.17.0 in /usr/local/lib/python3.7/site-packages (from pyGenomeTracks) (0.18.2)\n",
            "Requirement already satisfied: tqdm>=4.20 in /usr/local/lib/python3.7/site-packages (from pyGenomeTracks) (4.59.0)\n",
            "Requirement already satisfied: six>=1.12.0 in /usr/local/lib/python3.7/site-packages (from gffutils>=0.9->pyGenomeTracks) (1.15.0)\n",
            "Requirement already satisfied: argcomplete>=1.9.4 in /usr/local/lib/python3.7/site-packages (from gffutils>=0.9->pyGenomeTracks) (2.0.0)\n",
            "Requirement already satisfied: argh>=0.26.2 in /usr/local/lib/python3.7/site-packages (from gffutils>=0.9->pyGenomeTracks) (0.26.2)\n",
            "Requirement already satisfied: simplejson in /usr/local/lib/python3.7/site-packages (from gffutils>=0.9->pyGenomeTracks) (3.17.6)\n",
            "Requirement already satisfied: pyfaidx>=0.5.5.2 in /usr/local/lib/python3.7/site-packages (from gffutils>=0.9->pyGenomeTracks) (0.6.4)\n",
            "Requirement already satisfied: importlib-metadata<5,>=0.23 in /usr/local/lib/python3.7/site-packages (from argcomplete>=1.9.4->gffutils>=0.9->pyGenomeTracks) (4.11.1)\n",
            "Requirement already satisfied: scipy>=1.2.* in /usr/local/lib/python3.7/site-packages (from hicmatrix>=15->pyGenomeTracks) (1.7.3)\n",
            "Requirement already satisfied: pandas>=0.25.* in /usr/local/lib/python3.7/site-packages (from hicmatrix>=15->pyGenomeTracks) (1.3.5)\n",
            "Requirement already satisfied: cooler>=0.8.9 in /usr/local/lib/python3.7/site-packages (from hicmatrix>=15->pyGenomeTracks) (0.8.11)\n",
            "Requirement already satisfied: tables>=3.5.* in /usr/local/lib/python3.7/site-packages (from hicmatrix>=15->pyGenomeTracks) (3.7.0)\n",
            "Requirement already satisfied: pypairix in /usr/local/lib/python3.7/site-packages (from cooler>=0.8.9->hicmatrix>=15->pyGenomeTracks) (0.3.7)\n",
            "Requirement already satisfied: cytoolz<0.11 in /usr/local/lib/python3.7/site-packages (from cooler>=0.8.9->hicmatrix>=15->pyGenomeTracks) (0.10.1)\n",
            "Requirement already satisfied: multiprocess in /usr/local/lib/python3.7/site-packages (from cooler>=0.8.9->hicmatrix>=15->pyGenomeTracks) (0.70.12.2)\n",
            "Requirement already satisfied: click>=7 in /usr/local/lib/python3.7/site-packages (from cooler>=0.8.9->hicmatrix>=15->pyGenomeTracks) (8.0.3)\n",
            "Requirement already satisfied: pyyaml in /usr/local/lib/python3.7/site-packages (from cooler>=0.8.9->hicmatrix>=15->pyGenomeTracks) (6.0)\n",
            "Requirement already satisfied: asciitree in /usr/local/lib/python3.7/site-packages (from cooler>=0.8.9->hicmatrix>=15->pyGenomeTracks) (0.3.3)\n",
            "Requirement already satisfied: h5py>=2.5 in /usr/local/lib/python3.7/site-packages (from cooler>=0.8.9->hicmatrix>=15->pyGenomeTracks) (3.6.0)\n",
            "Requirement already satisfied: toolz>=0.8.0 in /usr/local/lib/python3.7/site-packages (from cytoolz<0.11->cooler>=0.8.9->hicmatrix>=15->pyGenomeTracks) (0.11.2)\n",
            "Requirement already satisfied: cached-property in /usr/local/lib/python3.7/site-packages (from h5py>=2.5->cooler>=0.8.9->hicmatrix>=15->pyGenomeTracks) (1.5.2)\n",
            "Requirement already satisfied: zipp>=0.5 in /usr/local/lib/python3.7/site-packages (from importlib-metadata<5,>=0.23->argcomplete>=1.9.4->gffutils>=0.9->pyGenomeTracks) (3.7.0)\n",
            "Requirement already satisfied: typing-extensions>=3.6.4 in /usr/local/lib/python3.7/site-packages (from importlib-metadata<5,>=0.23->argcomplete>=1.9.4->gffutils>=0.9->pyGenomeTracks) (4.1.1)\n",
            "Requirement already satisfied: sortedcontainers<3.0,>=2.0 in /usr/local/lib/python3.7/site-packages (from intervaltree>=2.1.0->pyGenomeTracks) (2.4.0)\n",
            "Requirement already satisfied: pillow>=6.2.0 in /usr/local/lib/python3.7/site-packages (from matplotlib<=3.3.2,>=3.1.1->pyGenomeTracks) (9.0.1)\n",
            "Requirement already satisfied: python-dateutil>=2.1 in /usr/local/lib/python3.7/site-packages (from matplotlib<=3.3.2,>=3.1.1->pyGenomeTracks) (2.8.2)\n",
            "Requirement already satisfied: pyparsing!=2.0.4,!=2.1.2,!=2.1.6,>=2.0.3 in /usr/local/lib/python3.7/site-packages (from matplotlib<=3.3.2,>=3.1.1->pyGenomeTracks) (3.0.7)\n",
            "Requirement already satisfied: certifi>=2020.06.20 in /usr/local/lib/python3.7/site-packages (from matplotlib<=3.3.2,>=3.1.1->pyGenomeTracks) (2021.10.8)\n",
            "Requirement already satisfied: cycler>=0.10 in /usr/local/lib/python3.7/site-packages (from matplotlib<=3.3.2,>=3.1.1->pyGenomeTracks) (0.11.0)\n",
            "Requirement already satisfied: kiwisolver>=1.0.1 in /usr/local/lib/python3.7/site-packages (from matplotlib<=3.3.2,>=3.1.1->pyGenomeTracks) (1.3.2)\n",
            "Requirement already satisfied: pytz>=2017.3 in /usr/local/lib/python3.7/site-packages (from pandas>=0.25.*->hicmatrix>=15->pyGenomeTracks) (2021.3)\n",
            "Requirement already satisfied: setuptools>=0.7 in /usr/local/lib/python3.7/site-packages (from pyfaidx>=0.5.5.2->gffutils>=0.9->pyGenomeTracks) (49.6.0.post20210108)\n",
            "Requirement already satisfied: numexpr>=2.6.2 in /usr/local/lib/python3.7/site-packages (from tables>=3.5.*->hicmatrix>=15->pyGenomeTracks) (2.8.1)\n",
            "Requirement already satisfied: packaging in /usr/local/lib/python3.7/site-packages (from tables>=3.5.*->hicmatrix>=15->pyGenomeTracks) (21.3)\n",
            "Requirement already satisfied: dill>=0.3.4 in /usr/local/lib/python3.7/site-packages (from multiprocess->cooler>=0.8.9->hicmatrix>=15->pyGenomeTracks) (0.3.4)\n",
            "Requirement already satisfied: py>=1.8.2 in /usr/local/lib/python3.7/site-packages (from pytest->pyGenomeTracks) (1.11.0)\n",
            "Requirement already satisfied: tomli>=1.0.0 in /usr/local/lib/python3.7/site-packages (from pytest->pyGenomeTracks) (2.0.1)\n",
            "Requirement already satisfied: pluggy<2.0,>=0.12 in /usr/local/lib/python3.7/site-packages (from pytest->pyGenomeTracks) (1.0.0)\n",
            "Requirement already satisfied: iniconfig in /usr/local/lib/python3.7/site-packages (from pytest->pyGenomeTracks) (1.1.1)\n",
            "Requirement already satisfied: attrs>=19.2.0 in /usr/local/lib/python3.7/site-packages (from pytest->pyGenomeTracks) (21.4.0)\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!pip install -U ipykernel"
      ],
      "metadata": {
        "id": "cbMU4vK3ujfa",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "202190d1-5841-4146-ed3f-aedb4632211f"
      },
      "execution_count": 114,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Requirement already satisfied: ipykernel in /usr/local/lib/python3.7/site-packages (6.9.1)\n",
            "Requirement already satisfied: ipython>=7.23.1 in /usr/local/lib/python3.7/site-packages (from ipykernel) (7.31.1)\n",
            "Requirement already satisfied: jupyter-client<8.0 in /usr/local/lib/python3.7/site-packages (from ipykernel) (7.1.2)\n",
            "Requirement already satisfied: tornado<7.0,>=4.2 in /usr/local/lib/python3.7/site-packages (from ipykernel) (6.1)\n",
            "Requirement already satisfied: traitlets<6.0,>=5.1.0 in /usr/local/lib/python3.7/site-packages (from ipykernel) (5.1.1)\n",
            "Requirement already satisfied: matplotlib-inline<0.2.0,>=0.1.0 in /usr/local/lib/python3.7/site-packages (from ipykernel) (0.1.3)\n",
            "Requirement already satisfied: nest-asyncio in /usr/local/lib/python3.7/site-packages (from ipykernel) (1.5.4)\n",
            "Requirement already satisfied: debugpy<2.0,>=1.0.0 in /usr/local/lib/python3.7/site-packages (from ipykernel) (1.5.1)\n",
            "Requirement already satisfied: jedi>=0.16 in /usr/local/lib/python3.7/site-packages (from ipython>=7.23.1->ipykernel) (0.18.1)\n",
            "Requirement already satisfied: prompt-toolkit!=3.0.0,!=3.0.1,<3.1.0,>=2.0.0 in /usr/local/lib/python3.7/site-packages (from ipython>=7.23.1->ipykernel) (3.0.28)\n",
            "Requirement already satisfied: setuptools>=18.5 in /usr/local/lib/python3.7/site-packages (from ipython>=7.23.1->ipykernel) (49.6.0.post20210108)\n",
            "Requirement already satisfied: decorator in /usr/local/lib/python3.7/site-packages (from ipython>=7.23.1->ipykernel) (5.1.1)\n",
            "Requirement already satisfied: pickleshare in /usr/local/lib/python3.7/site-packages (from ipython>=7.23.1->ipykernel) (0.7.5)\n",
            "Requirement already satisfied: pygments in /usr/local/lib/python3.7/site-packages (from ipython>=7.23.1->ipykernel) (2.11.2)\n",
            "Requirement already satisfied: pexpect>4.3 in /usr/local/lib/python3.7/site-packages (from ipython>=7.23.1->ipykernel) (4.8.0)\n",
            "Requirement already satisfied: backcall in /usr/local/lib/python3.7/site-packages (from ipython>=7.23.1->ipykernel) (0.2.0)\n",
            "Requirement already satisfied: parso<0.9.0,>=0.8.0 in /usr/local/lib/python3.7/site-packages (from jedi>=0.16->ipython>=7.23.1->ipykernel) (0.8.3)\n",
            "Requirement already satisfied: entrypoints in /usr/local/lib/python3.7/site-packages (from jupyter-client<8.0->ipykernel) (0.4)\n",
            "Requirement already satisfied: pyzmq>=13 in /usr/local/lib/python3.7/site-packages (from jupyter-client<8.0->ipykernel) (22.3.0)\n",
            "Requirement already satisfied: jupyter-core>=4.6.0 in /usr/local/lib/python3.7/site-packages (from jupyter-client<8.0->ipykernel) (4.9.2)\n",
            "Requirement already satisfied: python-dateutil>=2.1 in /usr/local/lib/python3.7/site-packages (from jupyter-client<8.0->ipykernel) (2.8.2)\n",
            "Requirement already satisfied: ptyprocess>=0.5 in /usr/local/lib/python3.7/site-packages (from pexpect>4.3->ipython>=7.23.1->ipykernel) (0.7.0)\n",
            "Requirement already satisfied: wcwidth in /usr/local/lib/python3.7/site-packages (from prompt-toolkit!=3.0.0,!=3.0.1,<3.1.0,>=2.0.0->ipython>=7.23.1->ipykernel) (0.2.5)\n",
            "Requirement already satisfied: six>=1.5 in /usr/local/lib/python3.7/site-packages (from python-dateutil>=2.1->jupyter-client<8.0->ipykernel) (1.15.0)\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "! conda install -c bioconda ucsc-fetchchromsizes"
      ],
      "metadata": {
        "id": "85FByJrZumJB",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "31762d00-d275-433d-a090-983b1cb4a6e9"
      },
      "execution_count": 77,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Collecting package metadata (current_repodata.json): - \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\bdone\n",
            "Solving environment: - \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\bdone\n",
            "\n",
            "## Package Plan ##\n",
            "\n",
            "  environment location: /usr/local\n",
            "\n",
            "  added / updated specs:\n",
            "    - ucsc-fetchchromsizes\n",
            "\n",
            "\n",
            "The following packages will be downloaded:\n",
            "\n",
            "    package                    |            build\n",
            "    ---------------------------|-----------------\n",
            "    libpng-1.6.37              |       h21135ba_2         306 KB  conda-forge\n",
            "    libuuid-2.32.1             |    h7f98852_1000          28 KB  conda-forge\n",
            "    mysql-connector-c-6.1.11   |    h6eb9d5d_1007         2.7 MB  conda-forge\n",
            "    ucsc-fetchchromsizes-377   |       h0b8a92a_2           7 KB  bioconda\n",
            "    ------------------------------------------------------------\n",
            "                                           Total:         3.0 MB\n",
            "\n",
            "The following NEW packages will be INSTALLED:\n",
            "\n",
            "  libpng             conda-forge/linux-64::libpng-1.6.37-h21135ba_2\n",
            "  libuuid            conda-forge/linux-64::libuuid-2.32.1-h7f98852_1000\n",
            "  mysql-connector-c  conda-forge/linux-64::mysql-connector-c-6.1.11-h6eb9d5d_1007\n",
            "  ucsc-fetchchromsi~ bioconda/linux-64::ucsc-fetchchromsizes-377-h0b8a92a_2\n",
            "\n",
            "\n",
            "\n",
            "Downloading and Extracting Packages\n",
            "mysql-connector-c-6. | 2.7 MB    | : 100% 1.0/1 [00:00<00:00,  2.06it/s]\n",
            "libuuid-2.32.1       | 28 KB     | : 100% 1.0/1 [00:00<00:00, 13.87it/s]\n",
            "libpng-1.6.37        | 306 KB    | : 100% 1.0/1 [00:00<00:00, 10.34it/s]\n",
            "ucsc-fetchchromsizes | 7 KB      | : 100% 1.0/1 [00:00<00:00, 21.58it/s]\n",
            "Preparing transaction: | \b\bdone\n",
            "Verifying transaction: - \b\b\\ \b\bdone\n",
            "Executing transaction: / \b\bdone\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "! fetchChromSizes mm10 > m.chrom.sizes"
      ],
      "metadata": {
        "id": "MK3M3DmLuoT1",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "0a75b8ce-4907-40ed-b253-3dda5cd94024"
      },
      "execution_count": 115,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "INFO: trying CURL  for database mm10\n",
            "url: http://hgdownload.soe.ucsc.edu/goldenPath/mm10/bigZips/mm10.chrom.sizes\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "! conda install -c bioconda ucsc-bedgraphtobigwig"
      ],
      "metadata": {
        "id": "7f_1-58NuqHN",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "f4cb64f5-9a4f-46bb-aa9c-6771d8fd2a85"
      },
      "execution_count": 116,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Collecting package metadata (current_repodata.json): - \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\bdone\n",
            "Solving environment: \\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\bdone\n",
            "\n",
            "# All requested packages already installed.\n",
            "\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "! bedGraphToBigWig  s_8_cell.deduplicated.bedGraph m.chrom.sizes cell8_methylation.bigWig"
      ],
      "metadata": {
        "id": "Pv8Ij9cRut3X"
      },
      "execution_count": 117,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "! bedGraphToBigWig  s_icm.deduplicated.bedGraph m.chrom.sizes icm_methylation.bigWig"
      ],
      "metadata": {
        "id": "XXzy2cu7vH_f"
      },
      "execution_count": 118,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "! bedGraphToBigWig  s_epiblast.deduplicated.bedGraph m.chrom.sizes epiblast_methylation.bigWig"
      ],
      "metadata": {
        "id": "EOxLShSOvIMD"
      },
      "execution_count": 119,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "! head s_8_cell.deduplicated.bedGraph"
      ],
      "metadata": {
        "id": "cl7vB8N1u5WR",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "071a5ecb-bd58-430d-c1f2-e21eb832285f"
      },
      "execution_count": 120,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "track type=bedGraph\n",
            "chr11\t3100030\t3100031\t50\n",
            "chr11\t3100047\t3100048\t66.6666666666667\n",
            "chr11\t3100048\t3100049\t50\n",
            "chr11\t3100223\t3100224\t87.5\n",
            "chr11\t3100224\t3100225\t53.3333333333333\n",
            "chr11\t3100285\t3100286\t66.6666666666667\n",
            "chr11\t3100286\t3100287\t33.3333333333333\n",
            "chr11\t3100309\t3100310\t62.5\n",
            "chr11\t3100310\t3100311\t50\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "! head s_icm.deduplicated.bedGraph"
      ],
      "metadata": {
        "id": "2Fu0-edEu704",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "87dba74c-90cc-408c-b2d7-1b26f31c4a83"
      },
      "execution_count": 121,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "track type=bedGraph\n",
            "chr11\t3100030\t3100031\t33.3333333333333\n",
            "chr11\t3100031\t3100032\t50\n",
            "chr11\t3100047\t3100048\t50\n",
            "chr11\t3100048\t3100049\t57.1428571428571\n",
            "chr11\t3100223\t3100224\t36.3636363636364\n",
            "chr11\t3100224\t3100225\t22.7272727272727\n",
            "chr11\t3100285\t3100286\t25\n",
            "chr11\t3100286\t3100287\t28.5714285714286\n",
            "chr11\t3100309\t3100310\t22.2222222222222\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "! head s_epiblast.deduplicated.bedGraph"
      ],
      "metadata": {
        "id": "gunIRxSDu7_O",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "8f02531b-85d7-4f57-b9c9-33b2b0282e6a"
      },
      "execution_count": 122,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "track type=bedGraph\n",
            "chr11\t3100030\t3100031\t75\n",
            "chr11\t3100031\t3100032\t100\n",
            "chr11\t3100047\t3100048\t100\n",
            "chr11\t3100048\t3100049\t100\n",
            "chr11\t3100223\t3100224\t96.875\n",
            "chr11\t3100224\t3100225\t93.75\n",
            "chr11\t3100285\t3100286\t93.3333333333333\n",
            "chr11\t3100286\t3100287\t92.3076923076923\n",
            "chr11\t3100309\t3100310\t96.7741935483871\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "! apt-get install bedtools"
      ],
      "metadata": {
        "id": "YlRr5Po3v6kP",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "37674434-56ac-4dc9-8cae-50d548910507"
      },
      "execution_count": 86,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Reading package lists... Done\n",
            "Building dependency tree       \n",
            "Reading state information... Done\n",
            "The following packages were automatically installed and are no longer required:\n",
            "  cuda-command-line-tools-10-0 cuda-command-line-tools-10-1\n",
            "  cuda-command-line-tools-11-0 cuda-compiler-10-0 cuda-compiler-10-1\n",
            "  cuda-compiler-11-0 cuda-cuobjdump-10-0 cuda-cuobjdump-10-1\n",
            "  cuda-cuobjdump-11-0 cuda-cupti-10-0 cuda-cupti-10-1 cuda-cupti-11-0\n",
            "  cuda-cupti-dev-11-0 cuda-documentation-10-0 cuda-documentation-10-1\n",
            "  cuda-documentation-11-0 cuda-documentation-11-1 cuda-gdb-10-0 cuda-gdb-10-1\n",
            "  cuda-gdb-11-0 cuda-gpu-library-advisor-10-0 cuda-gpu-library-advisor-10-1\n",
            "  cuda-libraries-10-0 cuda-libraries-10-1 cuda-libraries-11-0\n",
            "  cuda-memcheck-10-0 cuda-memcheck-10-1 cuda-memcheck-11-0 cuda-nsight-10-0\n",
            "  cuda-nsight-10-1 cuda-nsight-11-0 cuda-nsight-11-1 cuda-nsight-compute-10-0\n",
            "  cuda-nsight-compute-10-1 cuda-nsight-compute-11-0 cuda-nsight-compute-11-1\n",
            "  cuda-nsight-systems-10-1 cuda-nsight-systems-11-0 cuda-nsight-systems-11-1\n",
            "  cuda-nvcc-10-0 cuda-nvcc-10-1 cuda-nvcc-11-0 cuda-nvdisasm-10-0\n",
            "  cuda-nvdisasm-10-1 cuda-nvdisasm-11-0 cuda-nvml-dev-10-0 cuda-nvml-dev-10-1\n",
            "  cuda-nvml-dev-11-0 cuda-nvprof-10-0 cuda-nvprof-10-1 cuda-nvprof-11-0\n",
            "  cuda-nvprune-10-0 cuda-nvprune-10-1 cuda-nvprune-11-0 cuda-nvtx-10-0\n",
            "  cuda-nvtx-10-1 cuda-nvtx-11-0 cuda-nvvp-10-0 cuda-nvvp-10-1 cuda-nvvp-11-0\n",
            "  cuda-nvvp-11-1 cuda-samples-10-0 cuda-samples-10-1 cuda-samples-11-0\n",
            "  cuda-samples-11-1 cuda-sanitizer-11-0 cuda-sanitizer-api-10-1\n",
            "  cuda-toolkit-10-0 cuda-toolkit-10-1 cuda-toolkit-11-0 cuda-toolkit-11-1\n",
            "  cuda-tools-10-0 cuda-tools-10-1 cuda-tools-11-0 cuda-tools-11-1\n",
            "  cuda-visual-tools-10-0 cuda-visual-tools-10-1 cuda-visual-tools-11-0\n",
            "  cuda-visual-tools-11-1 default-jre dkms freeglut3 freeglut3-dev\n",
            "  keyboard-configuration libargon2-0 libcap2 libcryptsetup12\n",
            "  libdevmapper1.02.1 libfontenc1 libidn11 libip4tc0 libjansson4\n",
            "  libnvidia-cfg1-510 libnvidia-common-460 libnvidia-common-510\n",
            "  libnvidia-extra-510 libnvidia-fbc1-510 libnvidia-gl-510 libpam-systemd\n",
            "  libpolkit-agent-1-0 libpolkit-backend-1-0 libpolkit-gobject-1-0 libxfont2\n",
            "  libxi-dev libxkbfile1 libxmu-dev libxmu-headers libxnvctrl0 libxtst6\n",
            "  nsight-compute-2020.2.1 nsight-compute-2022.1.0 nsight-systems-2020.3.2\n",
            "  nsight-systems-2020.3.4 nsight-systems-2021.5.2 nvidia-dkms-510\n",
            "  nvidia-kernel-common-510 nvidia-kernel-source-510 nvidia-modprobe\n",
            "  nvidia-settings openjdk-11-jre policykit-1 policykit-1-gnome python3-xkit\n",
            "  screen-resolution-extra systemd systemd-sysv udev x11-xkb-utils\n",
            "  xserver-common xserver-xorg-core-hwe-18.04 xserver-xorg-video-nvidia-510\n",
            "Use 'apt autoremove' to remove them.\n",
            "The following NEW packages will be installed:\n",
            "  bedtools\n",
            "0 upgraded, 1 newly installed, 0 to remove and 39 not upgraded.\n",
            "Need to get 577 kB of archives.\n",
            "After this operation, 2,040 kB of additional disk space will be used.\n",
            "Get:1 http://archive.ubuntu.com/ubuntu bionic/universe amd64 bedtools amd64 2.26.0+dfsg-5 [577 kB]\n",
            "Fetched 577 kB in 1s (853 kB/s)\n",
            "Selecting previously unselected package bedtools.\n",
            "(Reading database ... 155113 files and directories currently installed.)\n",
            "Preparing to unpack .../bedtools_2.26.0+dfsg-5_amd64.deb ...\n",
            "Unpacking bedtools (2.26.0+dfsg-5) ...\n",
            "Setting up bedtools (2.26.0+dfsg-5) ...\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "! bedtools genomecov -bg -ibam s_8_cell.deduplicated.bam -g m.chromSizes  >  s_8_cell.cover.bedGraph"
      ],
      "metadata": {
        "id": "J430SextwBFm"
      },
      "execution_count": 123,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "! bedtools genomecov -bg -ibam s_icm.deduplicated.bam -g m.chromSizes  >  s_icm.cover.bedGraph"
      ],
      "metadata": {
        "id": "MLATQ15bwBRj"
      },
      "execution_count": 124,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "! bedtools genomecov -bg -ibam s_epiblast.deduplicated.bam -g m.chromSizes  >  s_epiblast.cover.bedGraph"
      ],
      "metadata": {
        "id": "hJ9ew2uhwBZU"
      },
      "execution_count": 125,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "! bedGraphToBigWig   s_8_cell.cover.bedGraph m.chrom.sizes cell8_coverage.bigWig.bw"
      ],
      "metadata": {
        "id": "htJQhjv_wRKf"
      },
      "execution_count": 133,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "! bedGraphToBigWig   s_icm.cover.bedGraph m.chrom.sizes icm_coverage.bigWig.bw"
      ],
      "metadata": {
        "id": "0-ETuKV1wRQa"
      },
      "execution_count": 134,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "! bedGraphToBigWig   s_epiblast.cover.bedGraph m.chrom.sizes epiblast_coverage.bigWig.bw"
      ],
      "metadata": {
        "id": "qi04Qi6bwRWz"
      },
      "execution_count": 135,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "! head s_8_cell.deduplicated.bedGraph"
      ],
      "metadata": {
        "id": "JnKIto6rwbhl",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "e4c60fbc-0a22-4bba-f99b-33b68f740a7e"
      },
      "execution_count": 129,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "track type=bedGraph\n",
            "chr11\t3100030\t3100031\t50\n",
            "chr11\t3100047\t3100048\t66.6666666666667\n",
            "chr11\t3100048\t3100049\t50\n",
            "chr11\t3100223\t3100224\t87.5\n",
            "chr11\t3100224\t3100225\t53.3333333333333\n",
            "chr11\t3100285\t3100286\t66.6666666666667\n",
            "chr11\t3100286\t3100287\t33.3333333333333\n",
            "chr11\t3100309\t3100310\t62.5\n",
            "chr11\t3100310\t3100311\t50\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "! head s_icm.deduplicated.bedGraph"
      ],
      "metadata": {
        "id": "kxSyhAR8wboC",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "5388e742-312c-47eb-c8bf-7655e3df5100"
      },
      "execution_count": 130,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "track type=bedGraph\n",
            "chr11\t3100030\t3100031\t33.3333333333333\n",
            "chr11\t3100031\t3100032\t50\n",
            "chr11\t3100047\t3100048\t50\n",
            "chr11\t3100048\t3100049\t57.1428571428571\n",
            "chr11\t3100223\t3100224\t36.3636363636364\n",
            "chr11\t3100224\t3100225\t22.7272727272727\n",
            "chr11\t3100285\t3100286\t25\n",
            "chr11\t3100286\t3100287\t28.5714285714286\n",
            "chr11\t3100309\t3100310\t22.2222222222222\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "! head s_epiblast.deduplicated.bedGraph"
      ],
      "metadata": {
        "id": "XsBcL3fLwbu3",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "fab93db4-777d-4476-f3bd-caefe843658f"
      },
      "execution_count": 131,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "track type=bedGraph\n",
            "chr11\t3100030\t3100031\t75\n",
            "chr11\t3100031\t3100032\t100\n",
            "chr11\t3100047\t3100048\t100\n",
            "chr11\t3100048\t3100049\t100\n",
            "chr11\t3100223\t3100224\t96.875\n",
            "chr11\t3100224\t3100225\t93.75\n",
            "chr11\t3100285\t3100286\t93.3333333333333\n",
            "chr11\t3100286\t3100287\t92.3076923076923\n",
            "chr11\t3100309\t3100310\t96.7741935483871\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "! make_tracks_file --trackFiles cell8_coverage.bigWig.bw cell8_coverage.bigWig.bw cell8_coverage.bigWig.bw -o tracks.ini.txt"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "IMakqxHJa-qP",
        "outputId": "7c3cdf7c-2f28-43fd-8cb6-dbc8c1285606"
      },
      "execution_count": 136,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Adding bigwig file: cell8_coverage.bigWig.bw\n",
            "Adding bigwig file: cell8_coverage.bigWig.bw\n",
            "Adding bigwig file: cell8_coverage.bigWig.bw\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "! make_tracks_file --trackFiles icm_coverage.bigWig.bw icm_coverage.bigWig.bw icm_coverage.bigWig.bw -o tracks.ini.txt"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "CXvRKLzWa_1d",
        "outputId": "da959047-7ed7-4a5c-83a1-251a51dba9fb"
      },
      "execution_count": 137,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Adding bigwig file: icm_coverage.bigWig.bw\n",
            "Adding bigwig file: icm_coverage.bigWig.bw\n",
            "Adding bigwig file: icm_coverage.bigWig.bw\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "! make_tracks_file --trackFiles epiblast_coverage.bigWig.bw epiblast_coverage.bigWig.bw epiblast_coverage.bigWig.bw -o tracks.ini.txt"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "5wGvodubbAOF",
        "outputId": "4b838f2b-8e31-4b2e-d3ca-03a49d6b0e8a"
      },
      "execution_count": 138,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Adding bigwig file: epiblast_coverage.bigWig.bw\n",
            "Adding bigwig file: epiblast_coverage.bigWig.bw\n",
            "Adding bigwig file: epiblast_coverage.bigWig.bw\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "! pyGenomeTracks --tracks tracks.ini.txt --region chr11:3100030-3500030 -o image_cov.png"
      ],
      "metadata": {
        "id": "Zmc9gJ2jwkXY",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "4f326902-e14f-4bed-a5d8-7056535044b9"
      },
      "execution_count": 139,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "INFO:pygenometracks.tracksClass:initialize 1. [x-axis]\n",
            "INFO:pygenometracks.tracksClass:initialize 2. [spacer]\n",
            "INFO:pygenometracks.tracksClass:initialize 3. [epiblast_coverage.bigWig]\n",
            "INFO:pygenometracks.tracksClass:initialize 4. [epiblast_coverage.bigWig]\n",
            "INFO:pygenometracks.tracksClass:initialize 5. [epiblast_coverage.bigWig]\n",
            "INFO:pygenometracks.tracksClass:time initializing track(s):\n",
            "INFO:pygenometracks.tracksClass:0.0022835731506347656\n",
            "DEBUG:pygenometracks.tracksClass:Figure size in cm is 40 x 8.909574468085108. Dpi is set to 72\n",
            "\n",
            "INFO:pygenometracks.tracksClass:plotting 1. [x-axis]\n",
            "INFO:pygenometracks.tracksClass:plotting 2. [spacer]\n",
            "INFO:pygenometracks.tracksClass:plotting 3. [epiblast_coverage.bigWig]\n",
            "INFO:pygenometracks.tracksClass:plotting 4. [epiblast_coverage.bigWig]\n",
            "INFO:pygenometracks.tracksClass:plotting 5. [epiblast_coverage.bigWig]\n"
          ]
        }
      ]
    }
  ]
}